# Google Summer of Code-2019 : Srijan Verma
### Mentors : [Daniel Zerbino](https://www.ebi.ac.uk/about/people/daniel-zerbino), [Elspeth Bruford](https://www.ebi.ac.uk/about/people/elspeth-bruford), [Ruth Seal](https://www.ebi.ac.uk/about/people/ruth-seal)

### Project Title  
Applying machine learning techniques to characterising and naming lncRNA genes.

![](https://media1.tenor.com/images/8ab51cca650124bdbd211ce2e8fb0714/tenor.gif?itemid=10625836)

## Brief Description
Advances in RNA sequencing technologies have revealed the complexity of our genome.
Long non-coding RNAs (lncRNAs) make up the majority of the non-coding transcriptome.
Understanding the significance of this RNA world is one of the most important challenges
faced in biology today, and the lncRNAs within it represent a gold mine of potential new
biomarkers and drug targets. Its discovery is still at a preliminary stage.

<img src="https://www.researchgate.net/profile/Sayantan_Maji/publication/299381675/figure/fig1/AS:347583198318593@1459881700776/Non-coding-RNA-Non-coding-RNA-comprise-a-much-larger-portion-of-the-human-genome-than.png" width="600" height="300" />

To date, very few lncRNAs have been characterized in detail. However, it is clear that
lncRNAs are important regulators of gene expression, and lncRNAs are thought to have a
wide range of functions in cellular and developmental processes. There are many
specialized lncRNA databases (like RefSeq, GENCODE, Ensembl, SGD, tair). We will use
Machine Learning techniques to highlight and compare two sets of calls (of Ensembl /
GENCODE and RefSeq) and determine which calls are incorrect.

## Specifications of the parent directory
Contains two folders namely: <br/>
1. Ensembl-analysis - Where scripts for making analysis and data collected from Ensembl can be found.
2. RefSeq-analysis - Where scripts for making analysis and data collected from RefSeq can be found.

## Dependencies
### Python
json<br/>
Pandas<br/>
Numpy

## Research papers / References

## Acknowledgements
