{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 36, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_overlap_gene = pd.read_csv('/feature_selection/overlap/Ensembl/all_gene_overlap.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 37, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>lincRNA ID</th>\n", "      <th>all gene ID</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in all gene</th>\n", "      <th>Ensembl-Ensembl overlap ?</th>\n", "      <th>Ensembl-linc_Ensembl overlap?</th>\n", "      <th>same_opp_overlap</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>ENSG00000000460</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>9</td>\n", "      <td>Yes</td>\n", "      <td>No</td>\n", "      <td>0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>ENSG00000007908</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>7</td>\n", "      <td>Yes</td>\n", "      <td>No</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["        lincRNA ID      all gene ID Chr region  No. of transcripts in lincrna  \\\n", "0  ENSG00000230704  ENSG00000000460          1                              1   \n", "1  ENSG00000230704  ENSG00000007908          1                              1   \n", "\n", "   No. of transcripts in all gene Ensembl-Ensembl overlap ?  \\\n", "0                               9                       Yes   \n", "1                               7                       Yes   \n", "\n", "  Ensembl-linc_Ensembl overlap?  same_opp_overlap  \n", "0                            No                 0  \n", "1                            No                 1  "]}, "execution_count": 37, "metadata": {}, "output_type": "execute_result"}], "source": ["df_overlap_gene.head(2)"]}, {"cell_type": "code", "execution_count": 38, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 5509/5509 [00:00<00:00, 13383.86it/s]\n"]}], "source": ["l_id = []\n", "a_id = []\n", "for i in tqdm(range(len(df_overlap_gene['lincRNA ID']))):\n", "    if df_overlap_gene['same_opp_overlap'][i] == 0:\n", "        l_id.append(df_overlap_gene['lincRNA ID'][i])\n", "        a_id.append(df_overlap_gene['all gene ID'][i])"]}, {"cell_type": "code", "execution_count": 39, "metadata": {}, "outputs": [], "source": ["set_lincrna_id = set(l_id)\n", "set_all_gene_id = set(a_id)"]}, {"cell_type": "code", "execution_count": 40, "metadata": {}, "outputs": [{"data": {"text/plain": ["2609"]}, "execution_count": 40, "metadata": {}, "output_type": "execute_result"}], "source": ["len(set_lincrna_id)"]}, {"cell_type": "code", "execution_count": 41, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["/site-packages/IPython/core/interactiveshell.py:3049: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.\n", "  interactivity=interactivity, compiler=compiler, result=result)\n"]}], "source": ["df_all_transcript_data = pd.read_csv('/dataset/Ensembl_numeric/all_gene_data/all_gene_sorted_chr/all_transcript_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 42, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 43, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>97111</td>\n", "      <td>ENSG00000162755</td>\n", "      <td>ENST00000392191</td>\n", "      <td>processed_transcript</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>161098440</td>\n", "      <td>161099735</td>\n", "      <td>737</td>\n", "      <td>4</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>94833</td>\n", "      <td>ENSG00000160789</td>\n", "      <td>ENST00000504687</td>\n", "      <td>protein_coding</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>156126545</td>\n", "      <td>156134960</td>\n", "      <td>546</td>\n", "      <td>4</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID               Biotype  Strand  \\\n", "0  97111  ENSG00000162755  ENST00000392191  processed_transcript       1   \n", "1  94833  ENSG00000160789  ENST00000504687        protein_coding       1   \n", "\n", "  Seq region Name  Transcript Start  Transcript End  Transcript Length  \\\n", "0               1         161098440       161099735                737   \n", "1               1         156126545       156134960                546   \n", "\n", "   No. of Exons  \n", "0             4  \n", "1             4  "]}, "execution_count": 43, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_transcript_data.head(2)"]}, {"cell_type": "code", "execution_count": 44, "metadata": {}, "outputs": [], "source": ["df_all_lincrna_transcript_data = pd.read_csv('/dataset/Ensembl_numeric/all_lincRNA_data/all_lincRNA_sorted_chr/all_lincrna_transcript_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 45, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>4916</td>\n", "      <td>ENSG00000236423</td>\n", "      <td>ENST00000413332</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>3900372</td>\n", "      <td>3905532</td>\n", "      <td>634</td>\n", "      <td>3</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>11412</td>\n", "      <td>ENSG00000270035</td>\n", "      <td>ENST00000602973</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>7698303</td>\n", "      <td>7698872</td>\n", "      <td>570</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0   4916  ENSG00000236423  ENST00000413332  lincRNA       1               1   \n", "1  11412  ENSG00000270035  ENST00000602973  lincRNA      -1               1   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  \n", "0           3900372         3905532                634             3  \n", "1           7698303         7698872                570             1  "]}, "execution_count": 45, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_lincrna_transcript_data.head(2)"]}, {"cell_type": "code", "execution_count": 46, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 13666/13666 [00:00<00:00, 35000.79it/s]\n"]}], "source": ["drop_index = []\n", "for i in tqdm(range(len(df_all_lincrna_transcript_data))):\n", "    if df_all_lincrna_transcript_data['Gene ID'][i] in set_lincrna_id:\n", "        continue\n", "    else:\n", "        drop_index.append(i)"]}, {"cell_type": "code", "execution_count": 47, "metadata": {}, "outputs": [{"data": {"text/plain": ["7661"]}, "execution_count": 47, "metadata": {}, "output_type": "execute_result"}], "source": ["len(drop_index)"]}, {"cell_type": "code", "execution_count": 48, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 4.74 ms, sys: 2.29 ms, total: 7.03 ms\n", "Wall time: 5.68 ms\n"]}], "source": ["%%time\n", "df_all_lincrna_transcript_data.drop(drop_index, axis=0,inplace=True)"]}, {"cell_type": "code", "execution_count": 49, "metadata": {}, "outputs": [], "source": ["df_all_lincrna_transcript_data.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 50, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>11412</td>\n", "      <td>ENSG00000270035</td>\n", "      <td>ENST00000602973</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>7698303</td>\n", "      <td>7698872</td>\n", "      <td>570</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2549</td>\n", "      <td>ENSG00000228397</td>\n", "      <td>ENST00000635097</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>22023997</td>\n", "      <td>22025483</td>\n", "      <td>1023</td>\n", "      <td>3</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0  11412  ENSG00000270035  ENST00000602973  lincRNA      -1               1   \n", "1   2549  ENSG00000228397  ENST00000635097  lincRNA      -1               1   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  \n", "0           7698303         7698872                570             1  \n", "1          22023997        22025483               1023             3  "]}, "execution_count": 50, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_lincrna_transcript_data.head(2)"]}, {"cell_type": "code", "execution_count": 51, "metadata": {}, "outputs": [{"data": {"text/plain": ["6005"]}, "execution_count": 51, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df_all_lincrna_transcript_data)"]}, {"cell_type": "code", "execution_count": 52, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 208423/208423 [00:09<00:00, 22410.81it/s]\n"]}], "source": ["drop_index = []\n", "for i in tqdm(range(len(df_all_transcript_data))):\n", "    if df_all_transcript_data['Gene ID'][i] in set_all_gene_id:\n", "        continue\n", "    else:\n", "        drop_index.append(i)"]}, {"cell_type": "code", "execution_count": 53, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 38.5 ms, sys: 4.25 ms, total: 42.7 ms\n", "Wall time: 44.4 ms\n"]}], "source": ["%%time\n", "df_all_transcript_data.drop(drop_index, axis=0,inplace=True)"]}, {"cell_type": "code", "execution_count": 54, "metadata": {}, "outputs": [], "source": ["df_all_transcript_data.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 55, "metadata": {}, "outputs": [{"data": {"text/plain": ["12839"]}, "execution_count": 55, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df_all_transcript_data)"]}, {"cell_type": "code", "execution_count": 56, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',\n", "       '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X',\n", "       'Y'], dtype=object)"]}, "execution_count": 56, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_lincrna_transcript_data['Seq region Name'].unique()"]}, {"cell_type": "code", "execution_count": 57, "metadata": {}, "outputs": [{"data": {"text/plain": ["array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19,\n", "       20, 21, 22, '22', 'X', 'Y'], dtype=object)"]}, "execution_count": 57, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_transcript_data['Seq region Name'].unique()"]}, {"cell_type": "code", "execution_count": 58, "metadata": {}, "outputs": [], "source": ["df_all_transcript_data['Seq region Name'] = df_all_transcript_data['Seq region Name'].astype(str)"]}, {"cell_type": "code", "execution_count": 59, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',\n", "       '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X',\n", "       'Y'], dtype=object)"]}, "execution_count": 59, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_transcript_data['Seq region Name'].unique()"]}, {"cell_type": "code", "execution_count": 60, "metadata": {}, "outputs": [], "source": ["len_all_gene_chr = df_all_transcript_data['Seq region Name'].value_counts()"]}, {"cell_type": "code", "execution_count": 61, "metadata": {}, "outputs": [], "source": ["# len_all_gene_chr"]}, {"cell_type": "code", "execution_count": 62, "metadata": {}, "outputs": [], "source": ["len_lincrna_gene_chr = df_all_lincrna_transcript_data['Seq region Name'].value_counts()"]}, {"cell_type": "code", "execution_count": 63, "metadata": {}, "outputs": [{"data": {"text/plain": ["719"]}, "execution_count": 63, "metadata": {}, "output_type": "execute_result"}], "source": ["len_lincrna_gene_chr['2']"]}, {"cell_type": "code", "execution_count": 64, "metadata": {}, "outputs": [], "source": ["lincrna_gene_coords = df_all_lincrna_transcript_data[['Transcript Start','Transcript End']].to_numpy()\n", "all_gene_coords = df_all_transcript_data[['Transcript Start','Transcript End']].to_numpy()"]}, {"cell_type": "code", "execution_count": 65, "metadata": {}, "outputs": [], "source": ["a = []\n", "for i in range(0,22,1):\n", "    a.append(str(i+1))"]}, {"cell_type": "code", "execution_count": 66, "metadata": {}, "outputs": [{"data": {"text/plain": ["'1'"]}, "execution_count": 66, "metadata": {}, "output_type": "execute_result"}], "source": ["a.append('X')\n", "a.append('Y')\n", "a[0]"]}, {"cell_type": "code", "execution_count": 70, "metadata": {}, "outputs": [], "source": ["lincrna_gene_id = []\n", "all_gene_id = []\n", "lincrna_transcript_id = []\n", "all_transcript_id = []\n", "no_exons_lincrna = []\n", "no_exons_all_gene = []\n", "chr_region = []\n", "overlap_len = []"]}, {"cell_type": "code", "execution_count": 71, "metadata": {}, "outputs": [], "source": ["lincrna_id = df_all_lincrna_transcript_data['Gene ID']\n", "gene_id = df_all_transcript_data['Gene ID']\n", "l_t_id = df_all_lincrna_transcript_data['Transcript ID']\n", "a_t_id = df_all_transcript_data['Transcript ID']\n", "genes_exon_no = df_all_transcript_data['No. of Exons']\n", "l_exon_no = df_all_lincrna_transcript_data['No. of Exons']"]}, {"cell_type": "code", "execution_count": 72, "metadata": {}, "outputs": [], "source": ["def overlap(A, B):\n", "    if A[0] < B[1] and B[0] < A[1]:\n", "        return True\n", "    else:\n", "        return False"]}, {"cell_type": "code", "execution_count": 73, "metadata": {}, "outputs": [], "source": ["def overlap_length(A, B):\n", "    return(min(A[1], B[1]) - max(A[0], B[0]) + 1) "]}, {"cell_type": "code", "execution_count": 74, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm\n", "def overlap_transcript():\n", "\n", "    for i in tqdm(range(0,24,1)):\n", "        \n", "        if i == 0:\n", "            for j in range(0,len_lincrna_gene_chr[a[i]]):  \n", "                for k in range(0, len_all_gene_chr[a[i]]):\n", "\n", "                    if overlap(lincrna_gene_coords[j], all_gene_coords[k]):\n", "                        \n", "                        overlap_len.append(overlap_length(lincrna_gene_coords[j], all_gene_coords[k]))\n", "                        lincrna_gene_id.append(lincrna_id[j])\n", "                        lincrna_transcript_id.append(l_t_id[j])\n", "                        no_exons_lincrna.append(l_exon_no[j])\n", "                        no_exons_all_gene.append(genes_exon_no[k])\n", "                        all_gene_id.append(gene_id[k])\n", "                        all_transcript_id.append(a_t_id[k])\n", "                        chr_region.append(a[i])\n", "                            \n", "                    else:\n", "                        \n", "                        continue\n", "\n", "        else:\n", "        \n", "            start_index_lincrna = 0\n", "            start_index_all_gene = 0\n", "            \n", "            for l in range(0,i):\n", "                start_index_all_gene = start_index_all_gene + len_all_gene_chr[a[l]]\n", "                start_index_lincrna = start_index_lincrna + len_lincrna_gene_chr[a[l]]\n", "            print('start lincrna {0}'.format(start_index_lincrna))\n", "            print('end lincrna {0}'.format(start_index_lincrna + len_lincrna_gene_chr[a[i]]))\n", "            print('start all_gene {0}'.format(start_index_all_gene))\n", "            print('end all_gene {0}'.format(start_index_all_gene + len_all_gene_chr[a[i]]))\n", "            \n", "            \n", "            for j in range(start_index_lincrna, start_index_lincrna + len_lincrna_gene_chr[a[i]], 1):\n", "                for k in range(start_index_all_gene, start_index_all_gene + len_all_gene_chr[a[i]], 1):\n", "                    \n", "                    if overlap(lincrna_gene_coords[j], all_gene_coords[k]):   \n", "                        \n", "                        overlap_len.append(overlap_length(lincrna_gene_coords[j], all_gene_coords[k]))\n", "                        lincrna_gene_id.append(lincrna_id[j])\n", "                        lincrna_transcript_id.append(l_t_id[j])\n", "                        no_exons_lincrna.append(l_exon_no[j])\n", "                        no_exons_all_gene.append(genes_exon_no[k])\n", "                        all_gene_id.append(gene_id[k])\n", "                        all_transcript_id.append(a_t_id[k])\n", "                        chr_region.append(a[i])\n", "                            \n", "                    else:\n", "                        \n", "                        continue"]}, {"cell_type": "code", "execution_count": 75, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["  4%|\u258d         | 1/24 [00:01<00:39,  1.72s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 476\n", "end lincrna 1195\n", "start all_gene 905\n", "end all_gene 2222\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", "  8%|\u258a         | 2/24 [00:07<01:04,  2.92s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 1195\n", "end lincrna 1492\n", "start all_gene 2222\n", "end all_gene 2989\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 12%|\u2588\u258e        | 3/24 [00:08<00:47,  2.29s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 1492\n", "end lincrna 1688\n", "start all_gene 2989\n", "end all_gene 3478\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 17%|\u2588\u258b        | 4/24 [00:08<00:35,  1.76s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 1688\n", "end lincrna 2011\n", "start all_gene 3478\n", "end all_gene 4229\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 21%|\u2588\u2588        | 5/24 [00:09<00:29,  1.55s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 2011\n", "end lincrna 2279\n", "start all_gene 4229\n", "end all_gene 4877\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 25%|\u2588\u2588\u258c       | 6/24 [00:10<00:22,  1.23s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 2279\n", "end lincrna 2462\n", "start all_gene 4877\n", "end all_gene 5204\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 29%|\u2588\u2588\u2589       | 7/24 [00:10<00:16,  1.03it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 2462\n", "end lincrna 2884\n", "start all_gene 5204\n", "end all_gene 5988\n"]}, {"name": "stderr", "output_type": "stream", "text": [" 38%|\u2588\u2588\u2588\u258a      | 9/24 [00:11<00:10,  1.43it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 2884\n", "end lincrna 3058\n", "start all_gene 5988\n", "end all_gene 6300\n", "start lincrna 3058\n", "end lincrna 3208\n", "start all_gene 6300\n", "end all_gene 6674\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 42%|\u2588\u2588\u2588\u2588\u258f     | 10/24 [00:11<00:07,  1.77it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 3208\n", "end lincrna 3379\n", "start all_gene 6674\n", "end all_gene 7255\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 46%|\u2588\u2588\u2588\u2588\u258c     | 11/24 [00:12<00:06,  2.09it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 3379\n", "end lincrna 3739\n", "start all_gene 7255\n", "end all_gene 8233\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 50%|\u2588\u2588\u2588\u2588\u2588     | 12/24 [00:13<00:06,  1.72it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 3739\n", "end lincrna 3898\n", "start all_gene 8233\n", "end all_gene 8550\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 54%|\u2588\u2588\u2588\u2588\u2588\u258d    | 13/24 [00:13<00:06,  1.83it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 3898\n", "end lincrna 4233\n", "start all_gene 8550\n", "end all_gene 9246\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 58%|\u2588\u2588\u2588\u2588\u2588\u258a    | 14/24 [00:14<00:05,  1.67it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 4233\n", "end lincrna 4515\n", "start all_gene 9246\n", "end all_gene 9980\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 62%|\u2588\u2588\u2588\u2588\u2588\u2588\u258e   | 15/24 [00:15<00:06,  1.46it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 4515\n", "end lincrna 4783\n", "start all_gene 9980\n", "end all_gene 10627\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 67%|\u2588\u2588\u2588\u2588\u2588\u2588\u258b   | 16/24 [00:15<00:05,  1.49it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 4783\n", "end lincrna 4979\n", "start all_gene 10627\n", "end all_gene 11172\n"]}, {"name": "stderr", "output_type": "stream", "text": [" 75%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258c  | 18/24 [00:16<00:02,  2.24it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 4979\n", "end lincrna 5150\n", "start all_gene 11172\n", "end all_gene 11402\n", "start lincrna 5150\n", "end lincrna 5377\n", "start all_gene 11402\n", "end all_gene 11760\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 79%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2589  | 19/24 [00:17<00:02,  1.78it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5377\n", "end lincrna 5513\n", "start all_gene 11760\n", "end all_gene 12040\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 83%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258e | 20/24 [00:17<00:01,  2.06it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5513\n", "end lincrna 5658\n", "start all_gene 12040\n", "end all_gene 12337\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 88%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258a | 21/24 [00:17<00:01,  2.04it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5658\n", "end lincrna 5830\n", "start all_gene 12337\n", "end all_gene 12597\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 92%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258f| 22/24 [00:18<00:01,  1.50it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5830\n", "end lincrna 5952\n", "start all_gene 12597\n", "end all_gene 12796\n"]}, {"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 24/24 [00:19<00:00,  1.21it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5952\n", "end lincrna 6005\n", "start all_gene 12796\n", "end all_gene 12839\n", "CPU times: user 15.3 s, sys: 478 ms, total: 15.8 s\n", "Wall time: 19.9 s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "overlap_transcript()"]}, {"cell_type": "code", "execution_count": 76, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['lincRNA gene ID', 'all gene ID', 'lincRNA trans ID', 'All trans ID', 'No. of exons all gene', 'No. of exons lincRNA', 'Chr region', 'overlap length']\n", "\n", "path = '/all_trans_overlap.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)"]}, {"cell_type": "code", "execution_count": 77, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df2 = pd.read_csv(path)\n", "\n", "df2[df2.columns[0]] = lincrna_gene_id\n", "df2[df2.columns[1]] = all_gene_id\n", "df2[df2.columns[2]] = lincrna_transcript_id\n", "df2[df2.columns[3]] =all_transcript_id\n", "df2[df2.columns[4]] = no_exons_all_gene\n", "df2[df2.columns[5]] = no_exons_lincrna\n", "df2[df2.columns[6]] = chr_region\n", "df2[df2.columns[7]] = overlap_len\n", "\n", "df2.to_csv(path)"]}, {"cell_type": "code", "execution_count": 79, "metadata": {}, "outputs": [{"data": {"text/plain": ["3528"]}, "execution_count": 79, "metadata": {}, "output_type": "execute_result"}], "source": ["len(l_id)\n", "len(a_id)"]}, {"cell_type": "code", "execution_count": 85, "metadata": {}, "outputs": [], "source": ["tot_ov_len = []"]}, {"cell_type": "code", "execution_count": 86, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["\n", "  0%|          | 0/3528 [00:00<?, ?it/s]\u001b[A\n", "  0%|          | 1/3528 [00:02<2:10:05,  2.21s/it]\u001b[A\n", "  0%|          | 2/3528 [00:04<2:06:05,  2.15s/it]\u001b[A\n", "  0%|          | 3/3528 [00:06<2:03:47,  2.11s/it]\u001b[A\n", "  0%|          | 4/3528 [00:08<2:01:19,  2.07s/it]\u001b[A\n", "  0%|          | 5/3528 [00:10<1:59:48,  2.04s/it]\u001b[A\n", "  0%|          | 6/3528 [00:12<1:58:38,  2.02s/it]\u001b[A\n", "  0%|          | 7/3528 [00:14<2:00:33,  2.05s/it]\u001b[A\n", "  0%|          | 8/3528 [00:16<1:59:38,  2.04s/it]\u001b[A\n", "  0%|          | 9/3528 [00:18<1:58:37,  2.02s/it]\u001b[A\n", "  0%|          | 10/3528 [00:20<1:58:10,  2.02s/it]\u001b[A\n", "  0%|          | 11/3528 [00:22<1:57:47,  2.01s/it]\u001b[A\n", "  0%|          | 12/3528 [00:24<1:57:28,  2.00s/it]\u001b[A\n", "  0%|          | 13/3528 [00:26<1:56:09,  1.98s/it]\u001b[A\n", "  0%|          | 14/3528 [00:28<1:53:41,  1.94s/it]\u001b[A\n", "  0%|          | 15/3528 [00:29<1:51:16,  1.90s/it]\u001b[A\n", "  0%|          | 16/3528 [00:31<1:49:32,  1.87s/it]\u001b[A\n", "  0%|          | 17/3528 [00:33<1:49:58,  1.88s/it]\u001b[A\n", "  1%|          | 18/3528 [00:35<1:50:12,  1.88s/it]\u001b[A\n", "  1%|          | 19/3528 [00:37<1:51:09,  1.90s/it]\u001b[A\n", "  1%|          | 20/3528 [00:38<1:42:05,  1.75s/it]\u001b[A\n", "  1%|          | 21/3528 [00:40<1:36:44,  1.66s/it]\u001b[A\n", "  1%|          | 22/3528 [00:41<1:35:24,  1.63s/it]\u001b[A\n", "  1%|          | 23/3528 [00:43<1:37:12,  1.66s/it]\u001b[A\n", "  1%|          | 24/3528 [00:45<1:43:16,  1.77s/it]\u001b[A\n", "  1%|          | 25/3528 [00:47<1:47:29,  1.84s/it]\u001b[A\n", "  1%|          | 26/3528 [00:48<1:37:37,  1.67s/it]\u001b[A\n", "  1%|          | 27/3528 [00:50<1:31:32,  1.57s/it]\u001b[A\n", "  1%|          | 28/3528 [00:51<1:27:19,  1.50s/it]\u001b[A\n", "  1%|          | 29/3528 [00:53<1:44:40,  1.80s/it]\u001b[A\n", "  1%|          | 30/3528 [00:55<1:37:09,  1.67s/it]\u001b[A\n", "  1%|          | 31/3528 [00:56<1:32:56,  1.59s/it]\u001b[A\n", "  1%|          | 32/3528 [00:58<1:29:42,  1.54s/it]\u001b[A\n", "  1%|          | 33/3528 [00:59<1:34:02,  1.61s/it]\u001b[A\n", "  1%|          | 34/3528 [01:01<1:24:26,  1.45s/it]\u001b[A\n", "  1%|          | 35/3528 [01:02<1:17:53,  1.34s/it]\u001b[A\n", "  1%|          | 36/3528 [01:03<1:12:40,  1.25s/it]\u001b[A\n", "  1%|          | 37/3528 [01:04<1:08:53,  1.18s/it]\u001b[A\n", "  1%|          | 38/3528 [01:05<1:05:57,  1.13s/it]\u001b[A\n", "  1%|          | 39/3528 [01:06<1:08:35,  1.18s/it]\u001b[A\n", "  1%|          | 40/3528 [01:07<1:09:06,  1.19s/it]\u001b[A\n", "  1%|          | 41/3528 [01:08<1:08:49,  1.18s/it]\u001b[A\n", "  1%|          | 42/3528 [01:10<1:11:20,  1.23s/it]\u001b[A\n", "  1%|          | 43/3528 [01:11<1:11:00,  1.22s/it]\u001b[A\n", "  1%|          | 44/3528 [01:12<1:08:37,  1.18s/it]\u001b[A\n", "  1%|\u258f         | 45/3528 [01:13<1:07:03,  1.16s/it]\u001b[A\n", "  1%|\u258f         | 46/3528 [01:14<1:09:36,  1.20s/it]\u001b[A\n", "  1%|\u258f         | 47/3528 [01:15<1:07:37,  1.17s/it]\u001b[A\n", "  1%|\u258f         | 48/3528 [01:16<1:04:49,  1.12s/it]\u001b[A\n", "  1%|\u258f         | 49/3528 [01:18<1:03:23,  1.09s/it]\u001b[A\n", "  1%|\u258f         | 50/3528 [01:18<1:00:35,  1.05s/it]\u001b[A\n", "  1%|\u258f         | 51/3528 [01:20<1:04:11,  1.11s/it]\u001b[A\n", "  1%|\u258f         | 52/3528 [01:21<1:03:43,  1.10s/it]\u001b[A\n", "  2%|\u258f         | 53/3528 [01:22<1:02:11,  1.07s/it]\u001b[A\n", "  2%|\u258f         | 54/3528 [01:23<1:02:42,  1.08s/it]\u001b[A\n", "  2%|\u258f         | 55/3528 [01:24<1:01:00,  1.05s/it]\u001b[A\n", "  2%|\u258f         | 56/3528 [01:25<1:03:13,  1.09s/it]\u001b[A\n", "  2%|\u258f         | 57/3528 [01:26<1:04:37,  1.12s/it]\u001b[A\n", "  2%|\u258f         | 58/3528 [01:28<1:08:21,  1.18s/it]\u001b[A\n", "  2%|\u258f         | 59/3528 [01:29<1:09:15,  1.20s/it]\u001b[A\n", "  2%|\u258f         | 60/3528 [01:30<1:08:16,  1.18s/it]\u001b[A\n", "  2%|\u258f         | 61/3528 [01:32<1:16:27,  1.32s/it]\u001b[A\n", "  2%|\u258f         | 62/3528 [01:33<1:19:32,  1.38s/it]\u001b[A\n", "  2%|\u258f         | 63/3528 [01:35<1:21:48,  1.42s/it]\u001b[A\n", "  2%|\u258f         | 64/3528 [01:36<1:29:07,  1.54s/it]\u001b[A\n", "  2%|\u258f         | 65/3528 [01:39<1:39:32,  1.72s/it]\u001b[A\n", "  2%|\u258f         | 66/3528 [01:40<1:34:17,  1.63s/it]\u001b[A\n", "  2%|\u258f         | 67/3528 [01:41<1:30:18,  1.57s/it]\u001b[A\n", "  2%|\u258f         | 68/3528 [01:43<1:26:56,  1.51s/it]\u001b[A\n", "  2%|\u258f         | 69/3528 [01:44<1:21:38,  1.42s/it]\u001b[A\n", "  2%|\u258f         | 70/3528 [01:45<1:17:59,  1.35s/it]\u001b[A\n", "  2%|\u258f         | 71/3528 [01:47<1:32:04,  1.60s/it]\u001b[A\n", "  2%|\u258f         | 72/3528 [01:50<1:43:26,  1.80s/it]\u001b[A\n", "  2%|\u258f         | 73/3528 [01:52<1:51:53,  1.94s/it]\u001b[A\n", "  2%|\u258f         | 74/3528 [01:54<1:48:56,  1.89s/it]\u001b[A\n", "  2%|\u258f         | 75/3528 [01:55<1:46:19,  1.85s/it]\u001b[A\n", "  2%|\u258f         | 76/3528 [01:57<1:40:39,  1.75s/it]\u001b[A\n", "  2%|\u258f         | 77/3528 [01:58<1:34:36,  1.64s/it]\u001b[A\n", "  2%|\u258f         | 78/3528 [02:00<1:38:23,  1.71s/it]\u001b[A\n", "  2%|\u258f         | 79/3528 [02:02<1:41:44,  1.77s/it]\u001b[A\n", "  2%|\u258f         | 80/3528 [02:04<1:42:56,  1.79s/it]\u001b[A\n", "  2%|\u258f         | 81/3528 [02:06<1:41:46,  1.77s/it]\u001b[A\n", "  2%|\u258f         | 82/3528 [02:07<1:33:51,  1.63s/it]\u001b[A\n", "  2%|\u258f         | 83/3528 [02:08<1:28:35,  1.54s/it]\u001b[A\n", "  2%|\u258f         | 84/3528 [02:10<1:27:49,  1.53s/it]\u001b[A\n", "  2%|\u258f         | 85/3528 [02:11<1:29:36,  1.56s/it]\u001b[A\n", "  2%|\u258f         | 86/3528 [02:13<1:26:33,  1.51s/it]\u001b[A\n", "  2%|\u258f         | 87/3528 [02:14<1:20:03,  1.40s/it]\u001b[A\n", "  2%|\u258f         | 88/3528 [02:15<1:15:09,  1.31s/it]\u001b[A\n", "  3%|\u258e         | 89/3528 [02:16<1:10:56,  1.24s/it]\u001b[A\n", "  3%|\u258e         | 90/3528 [02:17<1:07:08,  1.17s/it]\u001b[A\n", "  3%|\u258e         | 91/3528 [02:18<1:06:47,  1.17s/it]\u001b[A\n", "  3%|\u258e         | 92/3528 [02:20<1:10:50,  1.24s/it]\u001b[A\n", "  3%|\u258e         | 93/3528 [02:21<1:11:17,  1.25s/it]\u001b[A\n", "  3%|\u258e         | 94/3528 [02:22<1:05:25,  1.14s/it]\u001b[A\n", "  3%|\u258e         | 95/3528 [02:23<1:01:40,  1.08s/it]\u001b[A\n", "  3%|\u258e         | 96/3528 [02:24<58:40,  1.03s/it]  \u001b[A\n", "  3%|\u258e         | 97/3528 [02:25<56:31,  1.01it/s]\u001b[A\n", "  3%|\u258e         | 98/3528 [02:26<55:09,  1.04it/s]\u001b[A\n", "  3%|\u258e         | 99/3528 [02:26<54:21,  1.05it/s]\u001b[A\n", "  3%|\u258e         | 100/3528 [02:27<53:53,  1.06it/s]\u001b[A\n", "  3%|\u258e         | 101/3528 [02:28<53:01,  1.08it/s]\u001b[A\n", "  3%|\u258e         | 102/3528 [02:29<52:54,  1.08it/s]\u001b[A\n", "  3%|\u258e         | 103/3528 [02:30<52:53,  1.08it/s]\u001b[A\n", "  3%|\u258e         | 104/3528 [02:31<52:50,  1.08it/s]\u001b[A\n", "  3%|\u258e         | 105/3528 [02:32<52:40,  1.08it/s]\u001b[A\n", "  3%|\u258e         | 106/3528 [02:33<52:34,  1.08it/s]\u001b[A\n", "  3%|\u258e         | 107/3528 [02:34<52:30,  1.09it/s]\u001b[A\n", "  3%|\u258e         | 108/3528 [02:35<52:09,  1.09it/s]\u001b[A\n", "  3%|\u258e         | 109/3528 [02:36<52:10,  1.09it/s]\u001b[A\n", "  3%|\u258e         | 110/3528 [02:37<1:03:52,  1.12s/it]\u001b[A\n", "  3%|\u258e         | 111/3528 [02:39<1:07:01,  1.18s/it]\u001b[A\n", "  3%|\u258e         | 112/3528 [02:40<1:09:19,  1.22s/it]\u001b[A\n", "  3%|\u258e         | 113/3528 [02:42<1:16:45,  1.35s/it]\u001b[A\n", "  3%|\u258e         | 114/3528 [02:43<1:16:27,  1.34s/it]\u001b[A\n", "  3%|\u258e         | 115/3528 [02:44<1:16:03,  1.34s/it]\u001b[A\n", "  3%|\u258e         | 116/3528 [02:46<1:17:06,  1.36s/it]\u001b[A\n", "  3%|\u258e         | 117/3528 [02:47<1:16:46,  1.35s/it]\u001b[A\n", "  3%|\u258e         | 118/3528 [02:48<1:16:03,  1.34s/it]\u001b[A\n", "  3%|\u258e         | 119/3528 [02:50<1:16:48,  1.35s/it]\u001b[A\n", "  3%|\u258e         | 120/3528 [02:51<1:16:51,  1.35s/it]\u001b[A\n", "  3%|\u258e         | 121/3528 [02:52<1:19:39,  1.40s/it]\u001b[A\n", "  3%|\u258e         | 122/3528 [02:54<1:23:51,  1.48s/it]\u001b[A\n", "  3%|\u258e         | 123/3528 [02:56<1:24:21,  1.49s/it]\u001b[A\n", "  4%|\u258e         | 124/3528 [02:57<1:25:04,  1.50s/it]\u001b[A\n", "  4%|\u258e         | 125/3528 [02:59<1:22:43,  1.46s/it]\u001b[A\n", "  4%|\u258e         | 126/3528 [03:00<1:21:49,  1.44s/it]\u001b[A\n", "  4%|\u258e         | 127/3528 [03:02<1:24:34,  1.49s/it]\u001b[A\n", "  4%|\u258e         | 128/3528 [03:03<1:23:48,  1.48s/it]\u001b[A\n", "  4%|\u258e         | 129/3528 [03:04<1:22:11,  1.45s/it]\u001b[A\n", "  4%|\u258e         | 130/3528 [03:06<1:20:48,  1.43s/it]\u001b[A\n", "  4%|\u258e         | 131/3528 [03:07<1:22:32,  1.46s/it]\u001b[A\n", "  4%|\u258e         | 132/3528 [03:09<1:21:38,  1.44s/it]\u001b[A\n", "  4%|\u258d         | 133/3528 [03:10<1:20:32,  1.42s/it]\u001b[A\n", "  4%|\u258d         | 134/3528 [03:12<1:21:24,  1.44s/it]\u001b[A\n", "  4%|\u258d         | 135/3528 [03:13<1:20:00,  1.41s/it]\u001b[A\n", "  4%|\u258d         | 136/3528 [03:14<1:19:06,  1.40s/it]\u001b[A\n", "  4%|\u258d         | 137/3528 [03:16<1:19:52,  1.41s/it]\u001b[A\n", "  4%|\u258d         | 138/3528 [03:17<1:22:21,  1.46s/it]\u001b[A\n", "  4%|\u258d         | 139/3528 [03:19<1:18:32,  1.39s/it]\u001b[A\n", "  4%|\u258d         | 140/3528 [03:20<1:16:50,  1.36s/it]\u001b[A\n", "  4%|\u258d         | 141/3528 [03:22<1:24:50,  1.50s/it]\u001b[A\n", "  4%|\u258d         | 142/3528 [03:23<1:27:27,  1.55s/it]\u001b[A\n", "  4%|\u258d         | 143/3528 [03:25<1:27:00,  1.54s/it]\u001b[A\n", "  4%|\u258d         | 144/3528 [03:26<1:27:53,  1.56s/it]\u001b[A\n", "  4%|\u258d         | 145/3528 [03:28<1:27:25,  1.55s/it]\u001b[A\n", "  4%|\u258d         | 146/3528 [03:30<1:40:24,  1.78s/it]\u001b[A\n", "  4%|\u258d         | 147/3528 [03:32<1:46:51,  1.90s/it]\u001b[A\n", "  4%|\u258d         | 148/3528 [03:35<1:58:12,  2.10s/it]\u001b[A\n", "  4%|\u258d         | 149/3528 [03:37<1:54:44,  2.04s/it]\u001b[A\n", "  4%|\u258d         | 150/3528 [03:39<2:01:32,  2.16s/it]\u001b[A\n", "  4%|\u258d         | 151/3528 [03:42<2:02:11,  2.17s/it]\u001b[A\n", "  4%|\u258d         | 152/3528 [03:44<2:00:09,  2.14s/it]\u001b[A\n", "  4%|\u258d         | 153/3528 [03:45<1:53:13,  2.01s/it]\u001b[A\n", "  4%|\u258d         | 154/3528 [03:48<2:10:47,  2.33s/it]\u001b[A\n", "  4%|\u258d         | 155/3528 [03:51<2:17:23,  2.44s/it]\u001b[A\n", "  4%|\u258d         | 156/3528 [03:53<2:15:51,  2.42s/it]\u001b[A\n", "  4%|\u258d         | 157/3528 [03:56<2:10:51,  2.33s/it]\u001b[A\n", "  4%|\u258d         | 158/3528 [03:58<2:08:32,  2.29s/it]\u001b[A\n", "  5%|\u258d         | 159/3528 [04:00<2:09:58,  2.31s/it]\u001b[A"]}, {"ename": "KeyboardInterrupt", "evalue": "", "output_type": "error", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)", "\u001b[0;32m<ipython-input-86-6ec542706f87>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mj\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0ml_id\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mdf2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mj\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0ma_id\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mdf2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mj\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m             \u001b[0mtotal_overlap_len\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtotal_overlap_len\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mdf2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m7\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mj\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;32m~/anaconda3/envs/gsoc_ensembl/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2896\u001b[0m         \u001b[0;31m# shortcut if the key is in columns\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2897\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2898\u001b[0;31m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mis_unique\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2899\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2900\u001b[0m                     \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;32m~/anaconda3/envs/gsoc_ensembl/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36m__contains__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3916\u001b[0m         \"\"\"\n\u001b[1;32m   3917\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3918\u001b[0;31m     \u001b[0;34m@\u001b[0m\u001b[0mAppender\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_index_shared_docs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'contains'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0m_index_doc_kwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3919\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__contains__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3920\u001b[0m         \u001b[0mhash\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;31mKeyboardInterrupt\u001b[0m: "]}], "source": ["for i in tqdm(range(len(l_id))):\n", "    total_overlap_len = 0\n", "    for j in range(len(df2)):\n", "        \n", "        if l_id[i] == df2[df2.columns[0]][j] and a_id[i] == df2[df2.columns[1]][j]:\n", "            total_overlap_len = total_overlap_len + df2[df2.columns[7]][j]\n", "            \n", "    tot_ov_len.append(total_overlap_len)"]}, {"cell_type": "code", "execution_count": 80, "metadata": {}, "outputs": [{"data": {"text/plain": ["'ENSG00000270035'"]}, "execution_count": 80, "metadata": {}, "output_type": "execute_result"}], "source": ["df2[df2.columns[0]][0]"]}, {"cell_type": "code", "execution_count": 87, "metadata": {}, "outputs": [{"data": {"text/plain": ["[1720,\n", " 1710,\n", " 22,\n", " 78224,\n", " 24292,\n", " 20681,\n", " 11670,\n", " 1807,\n", " 520,\n", " 3835,\n", " 164,\n", " 10317,\n", " 10667,\n", " 61372,\n", " 26252,\n", " 46642,\n", " 913,\n", " 14099,\n", " 1635,\n", " 2714,\n", " 164,\n", " 354,\n", " 73,\n", " 2008,\n", " 58723,\n", " 58723,\n", " 513,\n", " 3306,\n", " 737466,\n", " 130171,\n", " 1046,\n", " 5948,\n", " 968,\n", " 4184,\n", " 396,\n", " 282,\n", " 11186,\n", " 12384,\n", " 2116,\n", " 7542,\n", " 1184,\n", " 3410,\n", " 11004,\n", " 312,\n", " 4551,\n", " 10829,\n", " 206,\n", " 19016,\n", " 21468,\n", " 97218,\n", " 111041,\n", " 110,\n", " 1690,\n", " 32035,\n", " 1903,\n", " 9515,\n", " 2384,\n", " 6426,\n", " 174,\n", " 5458,\n", " 1757,\n", " 46642,\n", " 2748,\n", " 1084,\n", " 618,\n", " 4344,\n", " 5182,\n", " 175173,\n", " 19,\n", " 312,\n", " 176077,\n", " 152,\n", " 331,\n", " 12182,\n", " 680,\n", " 1015,\n", " 19016,\n", " 1887,\n", " 607,\n", " 26,\n", " 2256,\n", " 20681,\n", " 62949,\n", " 292,\n", " 9264,\n", " 540104,\n", " 70651,\n", " 3306,\n", " 11472,\n", " 1418,\n", " 1922,\n", " 28956,\n", " 4770,\n", " 24297,\n", " 4357,\n", " 4756,\n", " 8814,\n", " 14816,\n", " 1098,\n", " 1548,\n", " 416,\n", " 388,\n", " 6233,\n", " 731,\n", " 201,\n", " 10829,\n", " 536,\n", " 1137,\n", " 2226,\n", " 2625,\n", " 6233,\n", " 14735,\n", " 8884,\n", " 7092,\n", " 538,\n", " 565,\n", " 164,\n", " 15709,\n", " 1080,\n", " 695,\n", " 1085,\n", " 135938,\n", " 2363,\n", " 123,\n", " 17,\n", " 26252,\n", " 22870,\n", " 1690,\n", " 1070,\n", " 484,\n", " 20766,\n", " 3960,\n", " 5241,\n", " 3960,\n", " 3410,\n", " 4049,\n", " 2322,\n", " 4357,\n", " 4050,\n", " 2430,\n", " 1956,\n", " 898806,\n", " 2582,\n", " 952,\n", " 54406,\n", " 12182,\n", " 746,\n", " 438,\n", " 1940,\n", " 200,\n", " 729,\n", " 1219,\n", " 24297,\n", " 10820,\n", " 1298,\n", " 1295,\n", " 3045,\n", " 288,\n", " 1760]"]}, "execution_count": 87, "metadata": {}, "output_type": "execute_result"}], "source": ["tot_ov_len"]}, {"cell_type": "code", "execution_count": 88, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["\n", "\n", "  0%|          | 0/3528 [00:00<?, ?it/s]\u001b[A\u001b[A\n", "\n", "  0%|          | 1/3528 [00:02<2:04:11,  2.11s/it]\u001b[A\u001b[A\n", "\n", "  0%|          | 2/3528 [00:04<2:00:50,  2.06s/it]\u001b[A\u001b[A\n", "\n", "  0%|          | 3/3528 [00:05<1:57:11,  1.99s/it]\u001b[A\u001b[A"]}, {"ename": "KeyboardInterrupt", "evalue": "", "output_type": "error", "traceback": ["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m", "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)", "\u001b[0;32m<ipython-input-88-15b4cd29ce0d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtqdm\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ml_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mj\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m         \u001b[0;32mif\u001b[0m \u001b[0ml_id\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mdf2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf2\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mj\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m             \u001b[0mnew_index\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;32m~/anaconda3/envs/gsoc_ensembl/lib/python3.6/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m    866\u001b[0m         \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    867\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 868\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_value\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    869\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    870\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;32m~/anaconda3/envs/gsoc_ensembl/lib/python3.6/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_value\u001b[0;34m(self, series, key)\u001b[0m\n\u001b[1;32m   4349\u001b[0m         \u001b[0;31m# use this, e.g. DatetimeIndex\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4350\u001b[0m         \u001b[0;31m# Things like `Series._get_value` (via .at) pass the EA directly here.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 4351\u001b[0;31m         \u001b[0ms\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'_values'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mseries\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   4352\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0ms\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mExtensionArray\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mIndex\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   4353\u001b[0m             \u001b[0;31m# GH 20882, 21257\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;32m~/anaconda3/envs/gsoc_ensembl/lib/python3.6/site-packages/pandas/core/series.py\u001b[0m in \u001b[0;36m_values\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    479\u001b[0m         \u001b[0mReturn\u001b[0m \u001b[0mthe\u001b[0m \u001b[0minternal\u001b[0m \u001b[0mrepr\u001b[0m \u001b[0mof\u001b[0m \u001b[0mthis\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    480\u001b[0m         \"\"\"\n\u001b[0;32m--> 481\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minternal_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    482\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    483\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_formatting_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;32m~/anaconda3/envs/gsoc_ensembl/lib/python3.6/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36minternal_values\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1547\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_block\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexternal_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1548\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1549\u001b[0;31m     \u001b[0;32mdef\u001b[0m \u001b[0minternal_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1550\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_block\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minternal_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1551\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n", "\u001b[0;31mKeyboardInterrupt\u001b[0m: "]}], "source": ["new_index = []\n", "for i in tqdm(range(len(l_id))):\n", "    for j in range(len(df2)):\n", "        if l_id[i] == df2[df2.columns[0]][j]:\n", "            new_index.append(j)"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}