{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_ens_linc_gene = pd.read_csv('/dataset/Ensembl_numeric/all_lincRNA_data/all_lincRNA_sorted_chr/all_lincrna_gene_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',\n", "       '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X',\n", "       'Y'], dtype=object)"]}, "execution_count": 2, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_linc_gene['Seq region Name'].unique()"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [], "source": ["df_ref_lnc_gene = pd.read_csv('/dataset/refseq_numeric/all_type=lncRNA_data/all_lncrna_sorted_chr/all_lncrna_gene_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',\n", "       '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X',\n", "       'Y'], dtype=object)"]}, "execution_count": 4, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_lnc_gene['Seq_id'].unique()"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [], "source": ["len_ens_linc_gene_chr = df_ens_linc_gene['Seq region Name'].value_counts()"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [], "source": ["len_ref_lnc_gene_chr = df_ref_lnc_gene['Seq_id'].value_counts()"]}, {"cell_type": "code", "execution_count": 7, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Display Name</th>\n", "      <th>Biotype</th>\n", "      <th>Start</th>\n", "      <th>End</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>No. of Transcripts</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>6092</td>\n", "      <td>ENSG00000271387</td>\n", "      <td>AL445228.2</td>\n", "      <td>lincRNA</td>\n", "      <td>184385753</td>\n", "      <td>184386704</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2329</td>\n", "      <td>ENSG00000236648</td>\n", "      <td>AL136529.1</td>\n", "      <td>lincRNA</td>\n", "      <td>17717687</td>\n", "      <td>17749978</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["    SNO          Gene ID Display Name  Biotype      Start        End  Strand  \\\n", "0  6092  ENSG00000271387   AL445228.2  lincRNA  184385753  184386704      -1   \n", "1  2329  ENSG00000236648   AL136529.1  lincRNA   17717687   17749978       1   \n", "\n", "  Seq region Name  No. of Transcripts  \n", "0               1                   1  \n", "1               1                   1  "]}, "execution_count": 7, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_linc_gene.head(2)"]}, {"cell_type": "code", "execution_count": 8, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Index</th>\n", "      <th>RefSeq Gene ID</th>\n", "      <th>Dbxref</th>\n", "      <th>Name</th>\n", "      <th>gbkey</th>\n", "      <th>Biotype</th>\n", "      <th>Type</th>\n", "      <th>Seq_id</th>\n", "      <th>Strand</th>\n", "      <th>Start</th>\n", "      <th>End</th>\n", "      <th>No. of Transcripts</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>26</td>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>GeneID:107985730,HGNC:HGNC:52482</td>\n", "      <td>MIR1302-2HG</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>29926</td>\n", "      <td>31295</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>877</td>\n", "      <td>256858</td>\n", "      <td>gene-LINC01142</td>\n", "      <td>GeneID:284688,HGNC:HGNC:49456</td>\n", "      <td>LINC01142</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>1</td>\n", "      <td>-1</td>\n", "      <td>170271405</td>\n", "      <td>170284208</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO   Index    RefSeq Gene ID                            Dbxref  \\\n", "0    1      26  gene-MIR1302-2HG  GeneID:107985730,HGNC:HGNC:52482   \n", "1  877  256858    gene-LINC01142     GeneID:284688,HGNC:HGNC:49456   \n", "\n", "          Name gbkey Biotype  Type Seq_id  Strand      Start        End  \\\n", "0  MIR1302-2HG  Gene  lncRNA  gene      1       1      29926      31295   \n", "1    LINC01142  Gene  lncRNA  gene      1      -1  170271405  170284208   \n", "\n", "   No. of Transcripts  \n", "0                   1  \n", "1                   1  "]}, "execution_count": 8, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_lnc_gene.head(2)"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [{"data": {"text/plain": ["1"]}, "execution_count": 9, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_lnc_gene['Strand'][0]"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [{"data": {"text/plain": ["593"]}, "execution_count": 10, "metadata": {}, "output_type": "execute_result"}], "source": ["len_ens_linc_gene_chr['2']"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [], "source": ["ens_linc_gene_strand = df_ens_linc_gene['Strand']\n", "ref_lnc_gene_strand = df_ref_lnc_gene['Strand']"]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [], "source": ["ens_linc_gene_coords = df_ens_linc_gene[['Start','End']].to_numpy()\n", "ref_lnc_gene_coords = df_ref_lnc_gene[['Start','End']].to_numpy()"]}, {"cell_type": "code", "execution_count": 42, "metadata": {}, "outputs": [], "source": ["a = []\n", "for i in range(0,22,1):\n", "    a.append(str(i+1))"]}, {"cell_type": "code", "execution_count": 43, "metadata": {}, "outputs": [{"data": {"text/plain": ["'1'"]}, "execution_count": 43, "metadata": {}, "output_type": "execute_result"}], "source": ["a.append('X')\n", "a.append('Y')\n", "a[0]"]}, {"cell_type": "code", "execution_count": 44, "metadata": {}, "outputs": [], "source": ["linc_gene_id = []\n", "ref_gene_id = []\n", "same_opp_overlap = []\n", "no_transcripts_linc = []\n", "no_transcripts_ref = []\n", "chr_region = []\n", "ens_ref_overlap = []"]}, {"cell_type": "code", "execution_count": 45, "metadata": {}, "outputs": [], "source": ["for i in range(len(df_ref_lnc_gene)):\n", "    ens_ref_overlap.append('Yes')"]}, {"cell_type": "code", "execution_count": 46, "metadata": {}, "outputs": [], "source": ["lincrna_id = df_ens_linc_gene['Gene ID']\n", "ref_id = df_ref_lnc_gene['RefSeq Gene ID']\n", "linc_trans_no = df_ens_linc_gene['No. of Transcripts']\n", "ref_trans_no = df_ref_lnc_gene['No. of Transcripts']"]}, {"cell_type": "code", "execution_count": 47, "metadata": {}, "outputs": [], "source": ["def overlap(A, B):\n", "    if A[0] < B[1] and B[0] < A[1]:\n", "        return True\n", "    else:\n", "        return False"]}, {"cell_type": "code", "execution_count": 48, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm\n", "def overlap_gene():\n", "\n", "    for i in tqdm(range(0,24,1)):\n", "        \n", "        if i == 0:\n", "            for j in range(0,len_ens_linc_gene_chr[a[i]]):  \n", "                for k in range(0, len_ref_lnc_gene_chr[a[i]]):\n", "\n", "                    if overlap(ens_linc_gene_coords[j], ref_lnc_gene_coords[k]):\n", "                           \n", "                        no_transcripts_linc.append(linc_trans_no[j])\n", "                        no_transcripts_ref.append(ref_trans_no[k])\n", "                        linc_gene_id.append(lincrna_id[j])\n", "                        ref_gene_id.append(ref_id[k])\n", "                        chr_region.append(a[i])\n", "                        \n", "                        if ens_linc_gene_strand[j] == ref_lnc_gene_strand[k]:\n", "                            same_opp_overlap.append(1)\n", "                            \n", "                        elif ens_linc_gene_strand[j] != ref_lnc_gene_strand[k]:\n", "                            same_opp_overlap.append(0)\n", "                            \n", "                    else:\n", "                        continue\n", "\n", "        else:\n", "        \n", "            start_index_lincrna = 0\n", "            start_index_ref = 0\n", "            \n", "            for l in range(0,i):\n", "                start_index_ref = start_index_ref + len_ref_lnc_gene_chr[a[l]]\n", "                start_index_lincrna = start_index_lincrna + len_ens_linc_gene_chr[a[l]]\n", "                \n", "            print('start lincrna {0}'.format(start_index_lincrna))\n", "            print('end lincrna {0}'.format(start_index_lincrna + len_ens_linc_gene_chr[a[i]]))\n", "            print('start ref {0}'.format(start_index_ref))\n", "            print('end ref {0}'.format(start_index_ref + len_ref_lnc_gene_chr[a[i]]))\n", "            \n", "            \n", "            for j in range(start_index_lincrna, start_index_lincrna + len_ens_linc_gene_chr[a[i]], 1):\n", "                for k in range(start_index_ref, start_index_ref + len_ref_lnc_gene_chr[a[i]], 1):\n", "                        \n", "                    if overlap(ens_linc_gene_coords[j], ref_lnc_gene_coords[k]):   \n", "                        \n", "                        no_transcripts_linc.append(linc_trans_no[j])\n", "                        no_transcripts_ref.append(ref_trans_no[k])\n", "                        linc_gene_id.append(lincrna_id[j])\n", "                        ref_gene_id.append(ref_id[k])\n", "                        chr_region.append(a[i])\n", "                        \n", "                        if ens_linc_gene_strand[j] == ref_lnc_gene_strand[k]:\n", "                            same_opp_overlap.append(1)\n", "                            \n", "                        elif ens_linc_gene_strand[j] != ref_lnc_gene_strand[k]:\n", "                            same_opp_overlap.append(0)\n", "                            \n", "                    else:\n", "                        continue"]}, {"cell_type": "code", "execution_count": 49, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["  4%|\u258d         | 1/24 [00:01<00:37,  1.62s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 636\n", "end lincrna 1229\n", "start ref 1307\n", "end ref 2520\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", "  8%|\u258a         | 2/24 [00:02<00:33,  1.54s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 1229\n", "end lincrna 1578\n", "start ref 2520\n", "end ref 3406\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 12%|\u2588\u258e        | 3/24 [00:04<00:29,  1.39s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 1578\n", "end lincrna 1994\n", "start ref 3406\n", "end ref 4212\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 17%|\u2588\u258b        | 4/24 [00:04<00:24,  1.25s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 1994\n", "end lincrna 2484\n", "start ref 4212\n", "end ref 5022\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 21%|\u2588\u2588        | 5/24 [00:06<00:23,  1.24s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 2484\n", "end lincrna 2863\n", "start ref 5022\n", "end ref 5877\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 25%|\u2588\u2588\u258c       | 6/24 [00:06<00:19,  1.09s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 2863\n", "end lincrna 3157\n", "start ref 5877\n", "end ref 6614\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 29%|\u2588\u2588\u2589       | 7/24 [00:07<00:15,  1.06it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 3157\n", "end lincrna 3561\n", "start ref 6614\n", "end ref 7309\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 33%|\u2588\u2588\u2588\u258e      | 8/24 [00:08<00:13,  1.20it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 3561\n", "end lincrna 3841\n", "start ref 7309\n", "end ref 7965\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 38%|\u2588\u2588\u2588\u258a      | 9/24 [00:08<00:11,  1.27it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 3841\n", "end lincrna 4132\n", "start ref 7965\n", "end ref 8678\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 42%|\u2588\u2588\u2588\u2588\u258f     | 10/24 [00:09<00:11,  1.23it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 4132\n", "end lincrna 4446\n", "start ref 8678\n", "end ref 9314\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 46%|\u2588\u2588\u2588\u2588\u258c     | 11/24 [00:10<00:09,  1.36it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 4446\n", "end lincrna 4883\n", "start ref 9314\n", "end ref 10022\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 50%|\u2588\u2588\u2588\u2588\u2588     | 12/24 [00:10<00:09,  1.33it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 4883\n", "end lincrna 5137\n", "start ref 10022\n", "end ref 10540\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 54%|\u2588\u2588\u2588\u2588\u2588\u258d    | 13/24 [00:11<00:08,  1.34it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5137\n", "end lincrna 5438\n", "start ref 10540\n", "end ref 11020\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 58%|\u2588\u2588\u2588\u2588\u2588\u258a    | 14/24 [00:12<00:06,  1.53it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5438\n", "end lincrna 5734\n", "start ref 11020\n", "end ref 11501\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 62%|\u2588\u2588\u2588\u2588\u2588\u2588\u258e   | 15/24 [00:12<00:05,  1.55it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 5734\n", "end lincrna 6093\n", "start ref 11501\n", "end ref 11981\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 67%|\u2588\u2588\u2588\u2588\u2588\u2588\u258b   | 16/24 [00:13<00:05,  1.37it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 6093\n", "end lincrna 6432\n", "start ref 11981\n", "end ref 12510\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 71%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588   | 17/24 [00:14<00:04,  1.53it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 6432\n", "end lincrna 6682\n", "start ref 12510\n", "end ref 12879\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 75%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258c  | 18/24 [00:14<00:03,  1.78it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 6682\n", "end lincrna 6912\n", "start ref 12879\n", "end ref 13250\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 79%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2589  | 19/24 [00:14<00:02,  1.94it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 6912\n", "end lincrna 7135\n", "start ref 13250\n", "end ref 13651\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 83%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258e | 20/24 [00:15<00:02,  1.99it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 7135\n", "end lincrna 7332\n", "start ref 13651\n", "end ref 13929\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 88%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258a | 21/24 [00:15<00:01,  2.17it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 7332\n", "end lincrna 7501\n", "start ref 13929\n", "end ref 14215\n"]}, {"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 24/24 [00:16<00:00,  3.26it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lincrna 7501\n", "end lincrna 7638\n", "start ref 14215\n", "end ref 14525\n", "start lincrna 7638\n", "end lincrna 7690\n", "start ref 14525\n", "end ref 14642\n", "CPU times: user 12.2 s, sys: 291 ms, total: 12.5 s\n", "Wall time: 16.2 s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "overlap_gene()"]}, {"cell_type": "code", "execution_count": 40, "metadata": {}, "outputs": [{"data": {"text/plain": ["6252"]}, "execution_count": 40, "metadata": {}, "output_type": "execute_result"}], "source": ["len(linc_gene_id)"]}, {"cell_type": "code", "execution_count": 50, "metadata": {}, "outputs": [], "source": ["correct_ens_ref_overlap = []\n", "for i in range(len(linc_gene_id)):\n", "    correct_ens_ref_overlap.append(ens_ref_overlap[i])"]}, {"cell_type": "code", "execution_count": 51, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['ENS lincRNA gene ID', 'Ref lncRNA gene ID', 'Chr region', 'No. of transcripts in lincrna', 'No. of transcripts in lncrna', 'ENS-Ref Overlap ?', 'same_opp_overlap']\n", "\n", "path = '/feature_selection/overlap/Ens-Ref/ens_ref_gene_overlap.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)"]}, {"cell_type": "code", "execution_count": 52, "metadata": {}, "outputs": [], "source": ["# linc_gene_id = []\n", "# ref_gene_id = []\n", "# same_opp_overlap = []\n", "# no_transcripts_linc = []\n", "# no_transcripts_ref = []\n", "# chr_region = []\n", "# ens_ref_overlap = []"]}, {"cell_type": "code", "execution_count": 53, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df2 = pd.read_csv(path)\n", "\n", "df2[df2.columns[0]] = linc_gene_id\n", "df2[df2.columns[1]] = ref_gene_id\n", "df2[df2.columns[2]] = chr_region\n", "df2[df2.columns[3]] = no_transcripts_linc\n", "df2[df2.columns[4]] = no_transcripts_ref\n", "df2[df2.columns[5]] = correct_ens_ref_overlap\n", "df2[df2.columns[6]] = same_opp_overlap\n", "\n", "df2.to_csv(path)"]}, {"cell_type": "code", "execution_count": 54, "metadata": {}, "outputs": [{"data": {"text/plain": ["6252"]}, "execution_count": 54, "metadata": {}, "output_type": "execute_result"}], "source": ["len(same_opp_overlap)"]}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}