{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_ref_all_gene = pd.read_csv('/dataset/refseq_numeric/all_gene_data/all_gene_sorted_chr/all_gene_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',\n", "       '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X',\n", "       'Y'], dtype=object)"]}, "execution_count": 2, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_all_gene['Seq_id'].unique()"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [], "source": ["df_lncrna_gene = pd.read_csv('/dataset/refseq_numeric/all_type=lncRNA_data/all_lncrna_sorted_chr/all_lncrna_gene_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Index</th>\n", "      <th>RefSeq Gene ID</th>\n", "      <th>Dbxref</th>\n", "      <th>Name</th>\n", "      <th>gbkey</th>\n", "      <th>Biotype</th>\n", "      <th>Type</th>\n", "      <th>Seq_id</th>\n", "      <th>Strand</th>\n", "      <th>Start</th>\n", "      <th>End</th>\n", "      <th>No. of Transcripts</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>26</td>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>GeneID:107985730,HGNC:HGNC:52482</td>\n", "      <td>MIR1302-2HG</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>29926</td>\n", "      <td>31295</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>877</td>\n", "      <td>256858</td>\n", "      <td>gene-LINC01142</td>\n", "      <td>GeneID:284688,HGNC:HGNC:49456</td>\n", "      <td>LINC01142</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>1</td>\n", "      <td>-1</td>\n", "      <td>170271405</td>\n", "      <td>170284208</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO   Index    RefSeq Gene ID                            Dbxref  \\\n", "0    1      26  gene-MIR1302-2HG  GeneID:107985730,HGNC:HGNC:52482   \n", "1  877  256858    gene-LINC01142     GeneID:284688,HGNC:HGNC:49456   \n", "\n", "          Name gbkey Biotype  Type Seq_id  Strand      Start        End  \\\n", "0  MIR1302-2HG  Gene  lncRNA  gene      1       1      29926      31295   \n", "1    LINC01142  Gene  lncRNA  gene      1      -1  170271405  170284208   \n", "\n", "   No. of Transcripts  \n", "0                   1  \n", "1                   1  "]}, "execution_count": 4, "metadata": {}, "output_type": "execute_result"}], "source": ["df_lncrna_gene.head(2)"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Index</th>\n", "      <th>RefSeq Gene ID</th>\n", "      <th>Dbxref</th>\n", "      <th>Name</th>\n", "      <th>gbkey</th>\n", "      <th>Biotype</th>\n", "      <th>Type</th>\n", "      <th>Seq_id</th>\n", "      <th>Strand</th>\n", "      <th>Start</th>\n", "      <th>End</th>\n", "      <th>No. of Transcripts</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>gene-DDX11L1</td>\n", "      <td>GeneID:100287102,HGNC:HGNC:37102</td>\n", "      <td>DDX11L1</td>\n", "      <td>Gene</td>\n", "      <td>transcribed_pseudogene</td>\n", "      <td>pseudogene</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>11874</td>\n", "      <td>14409</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>3411</td>\n", "      <td>246389</td>\n", "      <td>gene-USF1</td>\n", "      <td>GeneID:7391,HGNC:HGNC:12593,MIM:191523</td>\n", "      <td>USF1</td>\n", "      <td>Gene</td>\n", "      <td>protein_coding</td>\n", "      <td>gene</td>\n", "      <td>1</td>\n", "      <td>-1</td>\n", "      <td>161039251</td>\n", "      <td>161045979</td>\n", "      <td>3</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["    SNO   Index RefSeq Gene ID                                  Dbxref  \\\n", "0     1       1   gene-DDX11L1        GeneID:100287102,HGNC:HGNC:37102   \n", "1  3411  246389      gene-USF1  GeneID:7391,HGNC:HGNC:12593,MIM:191523   \n", "\n", "      Name gbkey                 Biotype        Type Seq_id  Strand  \\\n", "0  DDX11L1  Gene  transcribed_pseudogene  pseudogene      1       1   \n", "1     USF1  Gene          protein_coding        gene      1      -1   \n", "\n", "       Start        End  No. of Transcripts  \n", "0      11874      14409                   1  \n", "1  161039251  161045979                   3  "]}, "execution_count": 5, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_all_gene.head(2)"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [], "source": ["len_all_gene_chr = df_ref_all_gene['Seq_id'].value_counts()"]}, {"cell_type": "code", "execution_count": 7, "metadata": {}, "outputs": [], "source": ["len_lnc_gene_chr = df_lncrna_gene['Seq_id'].value_counts()"]}, {"cell_type": "code", "execution_count": 8, "metadata": {}, "outputs": [], "source": ["lnc_gene_strand = df_lncrna_gene['Strand']\n", "all_gene_strand = df_ref_all_gene['Strand']"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [], "source": ["lncrna_gene_coords = df_lncrna_gene[['Start','End']].to_numpy()\n", "all_gene_coords = df_ref_all_gene[['Start','End']].to_numpy()"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["a = []\n", "for i in range(0,22,1):\n", "    a.append(str(i+1))"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [{"data": {"text/plain": ["'1'"]}, "execution_count": 11, "metadata": {}, "output_type": "execute_result"}], "source": ["a.append('X')\n", "a.append('Y')\n", "a[0]"]}, {"cell_type": "code", "execution_count": 18, "metadata": {}, "outputs": [], "source": ["lncrna_gene_id = []\n", "all_gene_id = []\n", "same_ref = []\n", "same_lncrna_ref = []\n", "same_opp_overlap = []\n", "no_transcripts_lncrna = []\n", "no_transcripts_all_gene = []\n", "chr_region = []\n", "all_gene_biotype = []"]}, {"cell_type": "code", "execution_count": 19, "metadata": {}, "outputs": [], "source": ["for i in range(len(df_ref_all_gene)):\n", "    same_lncrna_ref.append('No')"]}, {"cell_type": "code", "execution_count": 20, "metadata": {}, "outputs": [], "source": ["lncrna_id = df_lncrna_gene['RefSeq Gene ID']\n", "gene_id = df_ref_all_gene['RefSeq Gene ID']\n", "gene_biotype = df_ref_all_gene['Biotype']\n", "genes_trans_no = df_ref_all_gene['No. of Transcripts']\n", "lncrna_trans_no = df_lncrna_gene['No. of Transcripts']"]}, {"cell_type": "code", "execution_count": 21, "metadata": {}, "outputs": [], "source": ["def overlap(A, B):\n", "    if A[0] < B[1] and B[0] < A[1]:\n", "        return True\n", "    else:\n", "        return False"]}, {"cell_type": "code", "execution_count": 22, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm\n", "def overlap_gene():\n", "\n", "    for i in tqdm(range(0,24,1)):\n", "        \n", "        if i == 0:\n", "            for j in range(0,len_lnc_gene_chr[a[i]]):  \n", "                for k in range(0, len_all_gene_chr[a[i]]):\n", "                    \n", "                    if lncrna_id[j] == gene_id[k]:\n", "                        continue\n", "\n", "                    if overlap(lncrna_gene_coords[j], all_gene_coords[k]):\n", "                        \n", "                        if gene_biotype[k] == 'lncRNA':\n", "                            same_lncrna_ref[k] == 'Yes'\n", "                        \n", "                        no_transcripts_lncrna.append(lncrna_trans_no[j])\n", "                        no_transcripts_all_gene.append(genes_trans_no[k])\n", "                        lncrna_gene_id.append(lncrna_id[j])\n", "                        all_gene_id.append(gene_id[k])\n", "                        same_ref.append('Yes')\n", "                        chr_region.append(a[i])\n", "                        all_gene_biotype.append(gene_biotype[k])\n", "                        \n", "                        \n", "                        if lnc_gene_strand[j] == all_gene_strand[k]:\n", "                            same_opp_overlap.append(1)\n", "                            \n", "                        elif lnc_gene_strand[j] != all_gene_strand[k]:\n", "                            same_opp_overlap.append(0)\n", "                            \n", "                    else:\n", "                        continue\n", "\n", "        else:\n", "        \n", "            start_index_lncrna = 0\n", "            start_index_all_gene = 0\n", "            \n", "            for l in range(0,i):\n", "                start_index_all_gene = start_index_all_gene + len_all_gene_chr[a[l]]\n", "                start_index_lncrna = start_index_lncrna + len_lnc_gene_chr[a[l]]\n", "            print('start lncrna {0}'.format(start_index_lncrna))\n", "            print('end lncrna {0}'.format(start_index_lncrna + len_lnc_gene_chr[a[i]]))\n", "            print('start all_gene {0}'.format(start_index_all_gene))\n", "            print('end all_gene {0}'.format(start_index_all_gene + len_all_gene_chr[a[i]]))\n", "            \n", "            \n", "            for j in range(start_index_lncrna, start_index_lncrna + len_lnc_gene_chr[a[i]], 1):\n", "                for k in range(start_index_all_gene, start_index_all_gene + len_all_gene_chr[a[i]], 1):\n", "                    \n", "                    if lncrna_id[j] == gene_id[k]:\n", "                        continue\n", "\n", "                    if overlap(lncrna_gene_coords[j], all_gene_coords[k]):\n", "                        \n", "                        if gene_biotype[k] == 'lncRNA':\n", "                            same_lncrna_ref[k] == 'Yes'\n", "                        \n", "                        no_transcripts_lncrna.append(lncrna_trans_no[j])\n", "                        no_transcripts_all_gene.append(genes_trans_no[k])\n", "                        lncrna_gene_id.append(lncrna_id[j])\n", "                        all_gene_id.append(gene_id[k])\n", "                        same_ref.append('Yes')\n", "                        chr_region.append(a[i])\n", "                        all_gene_biotype.append(gene_biotype[k])\n", "                        \n", "                        \n", "                        if lnc_gene_strand[j] == all_gene_strand[k]:\n", "                            same_opp_overlap.append(1)\n", "                            \n", "                        elif lnc_gene_strand[j] != all_gene_strand[k]:\n", "                            same_opp_overlap.append(0)\n", "                            \n", "                    else:\n", "                        continue\n"]}, {"cell_type": "code", "execution_count": 23, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 24, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["  4%|\u258d         | 1/24 [05:25<2:04:35, 325.04s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 1307\n", "end lncrna 2520\n", "start all_gene 5109\n", "end all_gene 8980\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", "  8%|\u258a         | 2/24 [07:30<1:37:16, 265.28s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 2520\n", "end lncrna 3406\n", "start all_gene 8980\n", "end all_gene 11970\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 12%|\u2588\u258e        | 3/24 [08:41<1:12:25, 206.94s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 3406\n", "end lncrna 4212\n", "start all_gene 11970\n", "end all_gene 14411\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 17%|\u2588\u258b        | 4/24 [09:34<53:32, 160.63s/it]  "]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 4212\n", "end lncrna 5022\n", "start all_gene 14411\n", "end all_gene 17003\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 21%|\u2588\u2588        | 5/24 [10:40<41:54, 132.37s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 5022\n", "end lncrna 5877\n", "start all_gene 17003\n", "end all_gene 20008\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 25%|\u2588\u2588\u258c       | 6/24 [11:52<34:13, 114.08s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 5877\n", "end lncrna 6614\n", "start all_gene 20008\n", "end all_gene 22800\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 29%|\u2588\u2588\u2589       | 7/24 [12:47<27:18, 96.41s/it] "]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 6614\n", "end lncrna 7309\n", "start all_gene 22800\n", "end all_gene 24965\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 33%|\u2588\u2588\u2588\u258e      | 8/24 [13:27<21:14, 79.63s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 7309\n", "end lncrna 7965\n", "start all_gene 24965\n", "end all_gene 27235\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 38%|\u2588\u2588\u2588\u258a      | 9/24 [14:07<16:54, 67.66s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 7965\n", "end lncrna 8678\n", "start all_gene 27235\n", "end all_gene 29414\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 42%|\u2588\u2588\u2588\u2588\u258f     | 10/24 [14:52<14:13, 60.94s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 8678\n", "end lncrna 9314\n", "start all_gene 29414\n", "end all_gene 32338\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 46%|\u2588\u2588\u2588\u2588\u258c     | 11/24 [15:43<12:34, 58.02s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 9314\n", "end lncrna 10022\n", "start all_gene 32338\n", "end all_gene 34864\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 50%|\u2588\u2588\u2588\u2588\u2588     | 12/24 [16:33<11:05, 55.47s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 10022\n", "end lncrna 10540\n", "start all_gene 34864\n", "end all_gene 36249\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 54%|\u2588\u2588\u2588\u2588\u2588\u258d    | 13/24 [16:53<08:12, 44.76s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 10540\n", "end lncrna 11020\n", "start all_gene 36249\n", "end all_gene 38314\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 58%|\u2588\u2588\u2588\u2588\u2588\u258a    | 14/24 [17:21<06:38, 39.81s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 11020\n", "end lncrna 11501\n", "start all_gene 38314\n", "end all_gene 40138\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 62%|\u2588\u2588\u2588\u2588\u2588\u2588\u258e   | 15/24 [17:45<05:16, 35.12s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 11501\n", "end lncrna 11981\n", "start all_gene 40138\n", "end all_gene 42076\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 67%|\u2588\u2588\u2588\u2588\u2588\u2588\u258b   | 16/24 [18:11<04:18, 32.26s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 11981\n", "end lncrna 12510\n", "start all_gene 42076\n", "end all_gene 44526\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 71%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588   | 17/24 [18:46<03:52, 33.25s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 12510\n", "end lncrna 12879\n", "start all_gene 44526\n", "end all_gene 45510\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 75%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258c  | 18/24 [18:56<02:37, 26.26s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 12879\n", "end lncrna 13250\n", "start all_gene 45510\n", "end all_gene 48009\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 79%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2589  | 19/24 [19:22<02:10, 26.03s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 13250\n", "end lncrna 13651\n", "start all_gene 48009\n", "end all_gene 49367\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 83%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258e | 20/24 [19:37<01:30, 22.72s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 13651\n", "end lncrna 13929\n", "start all_gene 49367\n", "end all_gene 50144\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 88%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258a | 21/24 [19:43<00:53, 17.69s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 13929\n", "end lncrna 14215\n", "start all_gene 50144\n", "end all_gene 51333\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 92%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258f| 22/24 [19:52<00:30, 15.18s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 14215\n", "end lncrna 14525\n", "start all_gene 51333\n", "end all_gene 53519\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", " 96%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u258c| 23/24 [20:11<00:16, 16.25s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["start lncrna 14525\n", "end lncrna 14642\n", "start all_gene 53519\n", "end all_gene 54099\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\r", "100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 24/24 [20:13<00:00, 11.93s/it]"]}, {"name": "stdout", "output_type": "stream", "text": ["CPU times: user 19min 4s, sys: 7.39 s, total: 19min 12s\n", "Wall time: 20min 13s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "overlap_gene()"]}, {"cell_type": "code", "execution_count": 25, "metadata": {}, "outputs": [], "source": ["correct_same_lnc_ref = []\n", "for i in range(len(lncrna_gene_id)):\n", "    correct_same_lnc_ref.append(same_lncrna_ref[i])"]}, {"cell_type": "code", "execution_count": 26, "metadata": {}, "outputs": [{"data": {"text/plain": ["11278"]}, "execution_count": 26, "metadata": {}, "output_type": "execute_result"}], "source": ["len(lncrna_gene_id)"]}, {"cell_type": "code", "execution_count": 29, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['lncRNA ID', 'all gene ID', 'Gene Biotype', 'Chr region', 'No. of transcripts in lncrna', 'No. of transcripts in all gene', 'RefSeq-RefSeq overlap ?', 'RefSeq-lnc_RefSeq overlap?', 'same_opp_overlap']\n", "\n", "path = '/feature_selection/overlap/Refseq/all_gene_overlap.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["# lncrna_gene_id = []\n", "# all_gene_id = []\n", "# same_ref = []\n", "# same_lncrna_ref = []\n", "# same_opp_overlap = []\n", "# no_transcripts_lncrna = []\n", "# no_transcripts_all_gene = []\n", "# chr_region = []\n", "# all_gene_biotype = []"]}, {"cell_type": "code", "execution_count": 30, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df2 = pd.read_csv(path)\n", "\n", "df2[df2.columns[0]] = lncrna_gene_id\n", "df2[df2.columns[1]] = all_gene_id\n", "df2[df2.columns[2]] = all_gene_biotype\n", "df2[df2.columns[3]] = chr_region\n", "df2[df2.columns[4]] = no_transcripts_lncrna\n", "df2[df2.columns[5]] = no_transcripts_all_gene\n", "df2[df2.columns[6]] = same_ref\n", "df2[df2.columns[7]] = correct_same_lnc_ref\n", "df2[df2.columns[8]] = same_opp_overlap\n", "\n", "df2.to_csv(path)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Ref(lnc) vs Ref(lnc) overlap [yes/no ?] wrongly calculated. \n", "### Correcting it below"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [], "source": ["df_lnc_all_gene_overlap = pd.read_csv('/1.lncRNA/feature_selection/overlap/Refseq/all_gene_overlap.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>lncRNA ID</th>\n", "      <th>all gene ID</th>\n", "      <th>Gene Biotype</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>No. of transcripts in all gene</th>\n", "      <th>RefSeq-RefSeq overlap ?</th>\n", "      <th>RefSeq-lnc_RefSeq overlap?</th>\n", "      <th>same_opp_overlap</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>gene-MIR1302-2</td>\n", "      <td>miRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>2</td>\n", "      <td>Yes</td>\n", "      <td>No</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>gene-LINC01681</td>\n", "      <td>gene-ISCUP1</td>\n", "      <td>pseudogene</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>0</td>\n", "      <td>Yes</td>\n", "      <td>No</td>\n", "      <td>0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>gene-LOC105371608</td>\n", "      <td>gene-KIFAP3</td>\n", "      <td>protein_coding</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>8</td>\n", "      <td>Yes</td>\n", "      <td>No</td>\n", "      <td>0</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["           lncRNA ID     all gene ID    Gene Biotype Chr region  \\\n", "0   gene-MIR1302-2HG  gene-MIR1302-2           miRNA          1   \n", "1     gene-LINC01681     gene-ISCUP1      pseudogene          1   \n", "2  gene-LOC105371608     gene-KIFAP3  protein_coding          1   \n", "\n", "   No. of transcripts in lncrna  No. of transcripts in all gene  \\\n", "0                             1                               2   \n", "1                             1                               0   \n", "2                             1                               8   \n", "\n", "  RefSeq-RefSeq overlap ? RefSeq-lnc_RefSeq overlap?  same_opp_overlap  \n", "0                     Yes                         No                 1  \n", "1                     Yes                         No                 0  \n", "2                     Yes                         No                 0  "]}, "execution_count": 3, "metadata": {}, "output_type": "execute_result"}], "source": ["df_lnc_all_gene_overlap.head(3)"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [], "source": ["df_lnc_all_gene_overlap.drop('RefSeq-lnc_RefSeq overlap?', axis=1, inplace=True)"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [], "source": ["same_lncrna_ref = []"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [], "source": ["df_ref_all_gene_index = df_ref_all_gene.set_index(['RefSeq Gene ID'])"]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 11278/11278 [00:06<00:00, 1704.62it/s]\n"]}], "source": ["for i in tqdm(range(len(df_lnc_all_gene_overlap))):\n", "    if df_ref_all_gene_index.loc[df_lnc_all_gene_overlap['all gene ID'][i]]['Biotype'] == 'lncRNA':\n", "        same_lncrna_ref.append('Yes')\n", "    else:\n", "        same_lncrna_ref.append('No')"]}, {"cell_type": "code", "execution_count": 13, "metadata": {}, "outputs": [], "source": ["df_lnc_all_gene_overlap['RefSeq-lnc_RefSeq overlap?'] = same_lncrna_ref"]}, {"cell_type": "code", "execution_count": 14, "metadata": {}, "outputs": [], "source": ["df_lnc_all_gene_overlap.to_csv('/1.lncRNA/feature_selection/overlap/Refseq/all_gene_overlap.csv')"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}