{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# Approach :\n", "1. Filter 'all_transcript_data.csv' having gbkey as ncRNA. Save the RefSeq Gene IDs and RefSeq Transcript ID columns.\n", "2. Filter 'all_gene_data.csv' whose gene IDs lie in the 'RefSeq Gene ID' collected in point 1 above.\n", "3. Filter 'all_exon_data.csv' having gbkey as ncRNA."]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv('/all_gene_data/all_transcript_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 41, "metadata": {}, "outputs": [], "source": ["df.head(5)"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [], "source": ["df.drop('SNO', axis=1,inplace=True)"]}, {"cell_type": "code", "execution_count": 42, "metadata": {}, "outputs": [], "source": ["len(df[df.columns[1]])"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [], "source": ["# df[df.columns[5]]\n", "from tqdm import tqdm"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below code for getting all transcript data having gbkey as ncRNA"]}, {"cell_type": "code", "execution_count": 43, "metadata": {}, "outputs": [], "source": ["%%time\n", "refseq_gene_ids = []\n", "ncrna_transcript_index_to_drop = []\n", "for i in tqdm(range(len(df[df.columns[1]]))):\n", "    if df[df.columns[5]][i] != 'ncRNA':\n", "        ncrna_transcript_index_to_drop.append(i)\n", "    elif df[df.columns[5]][i] == 'ncRNA':\n", "        refseq_gene_ids.append(df[df.columns[1]][i])\n", "        "]}, {"cell_type": "code", "execution_count": 7, "metadata": {}, "outputs": [], "source": ["df.drop(ncrna_transcript_index_to_drop, axis=0,inplace=True)"]}, {"cell_type": "code", "execution_count": 8, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(df[df.columns[1]])):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [], "source": ["df['SNO'] = s_no"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["df = df[['SNO','Index', 'RefSeq Gene ID','RefSeq Transcript ID', 'Dbxref', 'Parent', 'gbkey', 'Type', 'Seq_id', 'Strand','Start', 'End','No. of Exons','Transcript Length']]     "]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [], "source": ["df.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 44, "metadata": {}, "outputs": [], "source": ["df.head(5)"]}, {"cell_type": "code", "execution_count": 14, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_gbkey=ncRNA_data/all_ncrna_transcript_data.csv')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# gene data ncrna"]}, {"cell_type": "code", "execution_count": 16, "metadata": {}, "outputs": [], "source": ["df = pd.read_csv('/refseq_numeric/all_gene_data/all_gene_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 17, "metadata": {}, "outputs": [], "source": ["df.drop('SNO', axis=1,inplace=True)"]}, {"cell_type": "code", "execution_count": 45, "metadata": {}, "outputs": [], "source": ["# df.head(5)"]}, {"cell_type": "code", "execution_count": 46, "metadata": {}, "outputs": [], "source": ["%%time\n", "s = set(refseq_gene_ids)\n", "drop_values = []\n", "for i in tqdm(range(len(df[df.columns[1]]))):\n", "    if df[df.columns[1]][i] in s:\n", "        continue\n", "    else:\n", "        drop_values.append(i)"]}, {"cell_type": "code", "execution_count": 21, "metadata": {}, "outputs": [], "source": ["df.drop(drop_values, axis=0,inplace=True)"]}, {"cell_type": "code", "execution_count": 22, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(df[df.columns[1]])):\n", "    s_no.append(i+1)\n", "\n", "df['SNO'] = s_no"]}, {"cell_type": "code", "execution_count": 47, "metadata": {}, "outputs": [], "source": ["# df.head(2)"]}, {"cell_type": "code", "execution_count": 24, "metadata": {}, "outputs": [], "source": ["df = df[['SNO','Index', 'RefSeq Gene ID', 'Dbxref','Name', 'gbkey', 'Biotype', 'Type', 'Seq_id', 'Strand','Start', 'End', 'No. of Transcripts']]     "]}, {"cell_type": "code", "execution_count": 25, "metadata": {}, "outputs": [], "source": ["df.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 48, "metadata": {}, "outputs": [], "source": ["# df.head(5)"]}, {"cell_type": "code", "execution_count": 49, "metadata": {}, "outputs": [], "source": ["# len(df[df.columns[1]])"]}, {"cell_type": "code", "execution_count": 28, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_gbkey=ncRNA_data/all_ncrna_gene_data.csv')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Exon data ncrna below"]}, {"cell_type": "code", "execution_count": 50, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df = pd.read_csv('/all_gene_data/all_exon_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 30, "metadata": {}, "outputs": [], "source": ["df.drop('SNO', axis=1,inplace=True)"]}, {"cell_type": "code", "execution_count": 51, "metadata": {}, "outputs": [], "source": ["# df.head(5)"]}, {"cell_type": "code", "execution_count": 52, "metadata": {}, "outputs": [], "source": ["ncrna_exon_index_to_drop = []\n", "\n", "for i in tqdm(range(len(df[df.columns[1]]))):\n", "    \n", "    if df[df.columns[6]][i] != 'ncRNA':\n", "        \n", "        ncrna_exon_index_to_drop.append(i)"]}, {"cell_type": "code", "execution_count": 33, "metadata": {}, "outputs": [], "source": ["df.drop(ncrna_exon_index_to_drop, axis=0,inplace=True)"]}, {"cell_type": "code", "execution_count": 53, "metadata": {}, "outputs": [], "source": ["# len(df[df.columns[1]])"]}, {"cell_type": "code", "execution_count": 35, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(df[df.columns[1]])):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 36, "metadata": {}, "outputs": [], "source": ["df['SNO'] = s_no"]}, {"cell_type": "code", "execution_count": 37, "metadata": {}, "outputs": [], "source": ["df = df[['SNO','Index', 'RefSeq Gene ID','RefSeq Transcript ID','RefSeq Exon ID','Dbxref', 'Parent', 'gbkey', 'Type', 'Seq_id', 'Strand','Start', 'End','Exon Length']]     "]}, {"cell_type": "code", "execution_count": 38, "metadata": {}, "outputs": [], "source": ["df.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 54, "metadata": {}, "outputs": [], "source": ["df.head(5)"]}, {"cell_type": "code", "execution_count": 40, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_gbkey=ncRNA_data/all_ncrna_exon_data.csv')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# Once you load the csv files, do not forget to reset the index values ! [ df.reset_index(drop=True, inplace=True) ]. Default index values are not sequential (Since they got dropped)"]}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}