{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Extracting all numeric genes data from gff3 file, RefSeq\n", "### Approach to be followed :\n", "1. Between two consecutive gene_index of gff3 file lies transcript data (no. of transcripts, etc) and exon data.\n", "2. Between two consecutive transcript_index of gff3 file lies exon data."]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [], "source": ["import gffpandas.gffpandas as gffpd\n", "path = '/gff/GCF_000001405.38_GRCh38.p12_genomic.gff'"]}, {"cell_type": "code", "execution_count": 194, "metadata": {}, "outputs": [], "source": ["%%time\n", "annotation = gffpd.read_gff3(path)"]}, {"cell_type": "code", "execution_count": 447, "metadata": {}, "outputs": [], "source": ["# print(annotation.header)"]}, {"cell_type": "code", "execution_count": 448, "metadata": {}, "outputs": [], "source": ["# print(annotation.df)"]}, {"cell_type": "code", "execution_count": 449, "metadata": {}, "outputs": [], "source": ["# print(annotation.df['type'].unique())"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [], "source": ["%%time\n", "from tqdm import tqdm\n", "index = []\n", "count = 0\n", "for i in tqdm(range(len(annotation.df['type']))):\n", "    if annotation.df['type'][i] == 'lnc_RNA':\n", "        #print(annotation.df['attributes'][i])\n", "        index.append(str(i))\n", "        count = count + 1\n", "#         gene_ids.append(annotation.df['attributes'][i][8:23])\n", "print(count)        "]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [], "source": ["# index"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Extracting all genes data from RefSeq. \n", "#### Data collected are :\n", "a) From Attributes column of gff3 file\n", "1. Gene refseq IDs\n", "2. Gene dbxref\n", "3. Gene Name\n", "4. Gene biotype\n", "\n", "b) From different columns of gff3 file\n", "5. Gene start\n", "6. Gene End\n", "7. Gene Sequence ID\n", "8. Gene Strand\n", "9. Gene Index (Index where the gene lies in the gff3 file)\n", "10. Gene Type"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [], "source": ["gene_refseq_ids = [] #from attributes\n", "gene_dbxref = [] #from attributes\n", "gene_name = [] #from attributes\n", "gene_gbkey = [] #from attributes\n", "gene_biotype = [] #from attributes\n", "gene_start = []\n", "gene_end = []\n", "gene_seqid = []\n", "gene_strand = []\n", "gene_index = []\n", "gene_type = []"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [], "source": ["if annotation.df['attributes'][26][0:7] == 'ID=gene':\n", "    print(True)"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [], "source": ["annotation.df['attributes'][26][0:7]"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_refseq_gene_ids()] to get refseq gene data from gff3 file"]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 13, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 6 \u00b5s, sys: 3 \u00b5s, total: 9 \u00b5s\n", "Wall time: 13.1 \u00b5s\n"]}], "source": ["%%time\n", "def get_refseq_gene_ids():\n", "    for i in tqdm(range(len(annotation.df['attributes']))):\n", "        if annotation.df['attributes'][i][0:7] == 'ID=gene':\n", "            gene_index.append(i)\n", "            gene_start.append(annotation.df['start'][i])\n", "            gene_end.append(annotation.df['end'][i])\n", "            gene_seqid.append(annotation.df['seq_id'][i])\n", "            gene_strand.append(annotation.df['strand'][i])\n", "            gene_type.append(annotation.df['type'][i])\n", "            \n", "############################_DATA_COLLECTED_FROM_ATTRIBUTES_GIVEN_BELOW_#####################################################################            \n", "            \n", "            gene_biotype.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('gene_biotype=')+len('gene_biotype='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('gene_biotype=')+len('gene_biotype=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('gene_biotype=')+len('gene_biotype=')))])\n", "            gene_refseq_ids.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('ID=')+len('ID='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('ID=')+len('ID=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('ID=')+len('ID=')))])\n", "            gene_dbxref.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('Dbxref=')+len('Dbxref='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Dbxref=')+len('Dbxref=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Dbxref=')+len('Dbxref=')))])\n", "            gene_name.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('Name=')+len('Name='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')))])\n", "            gene_gbkey.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('gbkey=')+len('gbkey='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('gbkey=')+len('gbkey=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('gbkey=')+len('gbkey=')))])\n", "            \n", "            \n", "#             gene_refseq_ids.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('ID=')+3:annotation.df['attributes'][i].find(\";\",annotation.df['attributes'][i].find('ID=')+3)]) \n", "#             gene_dbxref.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('Dbxref=')+7:annotation.df['attributes'][i].find(\";\",annotation.df['attributes'][i].find('Dbxref=')+7)])     \n", "#             gene_name.append(annotation.df['attributes'][i][annotation.df['attributes'][i].find('Name=')+5:annotation.df['attributes'][i].find(\";\",annotation.df['attributes'][i].find('Name=')+5)])  "]}, {"cell_type": "code", "execution_count": 16, "metadata": {}, "outputs": [], "source": ["%%time\n", "get_refseq_gene_ids()"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [], "source": ["print(len(gene_refseq_ids)) \n", "print(len(gene_dbxref))\n", "print(len(gene_name))\n", "print(len(gene_gbkey))\n", "print(len(gene_biotype))\n", "print(len(gene_start))\n", "print(len(gene_end))\n", "print(len(gene_seqid))\n", "print(len(gene_strand))\n", "print(len(gene_index))\n", "print(len(gene_type))"]}, {"cell_type": "markdown", "metadata": {}, "source": ["#### converting '+' to '1' and '-' to '-1' so as to match it with Ensembl data (just for consistency)"]}, {"cell_type": "code", "execution_count": 180, "metadata": {}, "outputs": [], "source": ["changed_gene_strand = []\n", "for i in range(len(gene_strand)):\n", "    if gene_strand[i] == '+':\n", "        changed_gene_strand.append(1)\n", "    else:\n", "        changed_gene_strand.append(-1)"]}, {"cell_type": "code", "execution_count": 185, "metadata": {}, "outputs": [], "source": ["gene_strand = []\n", "gene_strand = changed_gene_strand"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Exporting gene data to all_gene_data.csv file"]}, {"cell_type": "code", "execution_count": 186, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['SNO','Index', 'RefSeq Gene ID', 'Dbxref', 'Name', 'gbkey', 'Biotype', 'Type', 'Seq_id', 'Strand','Start', 'End']\n", "\n", "path = '/all_gene_data/all_gene_data.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)"]}, {"cell_type": "code", "execution_count": 187, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(gene_refseq_ids)):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 188, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv(path)\n", "\n", "df[df.columns[0]] = s_no\n", "df[df.columns[1]] = gene_index\n", "df[df.columns[2]] = gene_refseq_ids\n", "df[df.columns[3]] = gene_dbxref\n", "df[df.columns[4]] = gene_name\n", "df[df.columns[5]] = gene_gbkey\n", "df[df.columns[6]] = gene_biotype\n", "df[df.columns[7]] = gene_type\n", "df[df.columns[8]] = gene_seqid\n", "df[df.columns[9]] = gene_strand\n", "df[df.columns[10]] = gene_start\n", "df[df.columns[11]] = gene_end\n", "\n", "df.to_csv(path)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Collecting transcript data below"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["no_of_transcripts = []\n", "transcript_index = []\n", "gene_refseq_ids_for_transcripts = []"]}, {"cell_type": "code", "execution_count": 317, "metadata": {}, "outputs": [], "source": ["transcript_refseq_ids = [] #from attributes\n", "transcript_dbxref = [] #from attributes\n", "#transcript_name = [] #from attributes\n", "transcript_gbkey = [] #from attributes\n", "transcript_parent = [] #from attributes\n", "transcript_start = []\n", "transcript_end = []\n", "transcript_seqid = []\n", "transcript_strand = []\n", "transcript_type = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_transcript_data()] to get transcript data of a particular gene"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [], "source": ["%%time\n", "def get_transcript_data():\n", "    #for i in range(len(gene_index)):\n", "    for i in tqdm(range(0,len(gene_index),1)):\n", "        \n", "        if i == len(gene_index)-1: # OR if i == gene_index[-1]\n", "            transcript_count = 0  #For dealing with the last case (i.e. gene_index[60958] element)\n", "            for k in range(gene_index[i],len(annotation.df['attributes'])-1,1):\n", "                if annotation.df['attributes'][k][0:6] == 'ID=rna':\n", "                    transcript_count = transcript_count + 1\n", "                    gene_refseq_ids_for_transcripts.append(gene_refseq_ids[i])\n", "                    transcript_index.append(k)                \n", "\n", "##########################_TRANSCRIPT_DATA_COLLECTION_BELOW_#################################################################\n", "                                \n", "                    transcript_start.append(annotation.df['start'][k])\n", "                    transcript_end.append(annotation.df['end'][k])\n", "                    transcript_seqid.append(annotation.df['seq_id'][k])\n", "                    transcript_strand.append(annotation.df['strand'][k])\n", "                    transcript_type.append(annotation.df['type'][k])\n", "            \n", "############################_DATA_COLLECTED_FROM_ATTRIBUTES_GIVEN_BELOW_#####################################################################            \n", "            \n", "                    transcript_parent.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('Parent=')+len('Parent='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Parent=')+len('Parent=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Parent=')+len('Parent=')))])\n", "                    transcript_refseq_ids.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('ID=')+len('ID='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('ID=')+len('ID=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('ID=')+len('ID=')))])\n", "                    transcript_dbxref.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('Dbxref=')+len('Dbxref='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Dbxref=')+len('Dbxref=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Dbxref=')+len('Dbxref=')))])\n", "#                     transcript_name.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Name=')+len('Name='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')))])\n", "                    transcript_gbkey.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('gbkey=')+len('gbkey='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('gbkey=')+len('gbkey=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('gbkey=')+len('gbkey=')))])\n", "    \n", "            no_of_transcripts.append(transcript_count)      \n", "            \n", "        else:\n", "            transcript_count = 0\n", "            for j in range(gene_index[i]+1,gene_index[i+1],1): #Taking gene IDs in pairs and then fetching transcript data \n", "                if annotation.df['attributes'][j][0:6] == 'ID=rna':\n", "                    transcript_count = transcript_count + 1\n", "                    gene_refseq_ids_for_transcripts.append(gene_refseq_ids[i])\n", "                    transcript_index.append(j)\n", "                    \n", "##########################_TRANSCRIPT_DATA_COLLECTION_BELOW_#################################################################\n", "                                \n", "                    transcript_start.append(annotation.df['start'][j])\n", "                    transcript_end.append(annotation.df['end'][j])\n", "                    transcript_seqid.append(annotation.df['seq_id'][j])\n", "                    transcript_strand.append(annotation.df['strand'][j])\n", "                    transcript_type.append(annotation.df['type'][j])\n", "            \n", "############################_DATA_COLLECTED_FROM_ATTRIBUTES_GIVEN_BELOW_#####################################################################            \n", "            \n", "                    transcript_parent.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Parent=')+len('Parent='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Parent=')+len('Parent=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Parent=')+len('Parent=')))])\n", "                    transcript_refseq_ids.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('ID=')+len('ID='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('ID=')+len('ID=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('ID=')+len('ID=')))])\n", "                    transcript_dbxref.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Dbxref=')+len('Dbxref='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Dbxref=')+len('Dbxref=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Dbxref=')+len('Dbxref=')))])\n", "#                     transcript_name.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Name=')+len('Name='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')))])\n", "                    transcript_gbkey.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('gbkey=')+len('gbkey='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('gbkey=')+len('gbkey=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('gbkey=')+len('gbkey=')))])\n", "\n", "            \n", "            no_of_transcripts.append(transcript_count)      "]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["%%time\n", "get_transcript_data()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["#### converting '+' to '1' and '-' to '-1' so as to match it with Ensembl data (just for consistency)"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [], "source": ["%%time\n", "\n", "changed_transcript_strand = []\n", "for i in range(len(transcript_strand)):\n", "    if transcript_strand[i] == '+':\n", "        changed_transcript_strand.append(1)\n", "    else:\n", "        changed_transcript_strand.append(-1)\n", "\n", "transcript_strand = []\n", "transcript_strand = changed_transcript_strand"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Exporting transcript data to all_transcript_data.csv file"]}, {"cell_type": "code", "execution_count": 330, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['SNO','Index', 'RefSeq Gene ID','RefSeq Transcript ID', 'Dbxref', 'Parent', 'gbkey', 'Type', 'Seq_id', 'Strand','Start', 'End']\n", "\n", "path = '/refseq_numeric/all_gene_data/all_transcript_data.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)"]}, {"cell_type": "code", "execution_count": 331, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(transcript_refseq_ids)):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [], "source": ["%%time\n", "import pandas as pd\n", "\n", "df = pd.read_csv(path)\n", "\n", "df[df.columns[0]] = s_no\n", "df[df.columns[1]] = transcript_index\n", "df[df.columns[2]] = gene_refseq_ids_for_transcripts\n", "df[df.columns[3]] = transcript_refseq_ids\n", "df[df.columns[4]] = transcript_dbxref\n", "df[df.columns[5]] = transcript_parent\n", "df[df.columns[6]] = transcript_gbkey\n", "df[df.columns[7]] = transcript_type\n", "df[df.columns[8]] = transcript_seqid\n", "df[df.columns[9]] = transcript_strand\n", "df[df.columns[10]] = transcript_start\n", "df[df.columns[11]] = transcript_end\n", "\n", "df.to_csv(path)"]}, {"cell_type": "code", "execution_count": 333, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv('/refseq_numeric/all_gene_data/all_gene_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 334, "metadata": {}, "outputs": [], "source": ["df['No. of Transcripts'] = no_of_transcripts"]}, {"cell_type": "code", "execution_count": 335, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_gene_data/all_gene_data.csv')"]}, {"cell_type": "code", "execution_count": 13, "metadata": {}, "outputs": [], "source": ["print(len(no_of_transcripts))\n", "print(len(gene_refseq_ids_for_transcripts))\n", "print(len(transcript_dbxref))\n", "print(len(transcript_gbkey))\n", "print(len(transcript_index))\n", "print(len(transcript_parent))\n", "print(len(transcript_seqid))\n", "print(len(transcript_start))\n", "print(len(transcript_end))\n", "print(len(transcript_type))\n", "print(len(transcript_refseq_ids))\n", "print(len(transcript_strand))\n", "#print(len(no_of_transcripts))"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Below - to get exon data from gff3 file"]}, {"cell_type": "code", "execution_count": 363, "metadata": {}, "outputs": [], "source": ["no_of_exons = []\n", "gene_refseq_ids_for_exons = []\n", "transcript_refseq_ids_for_exons = []\n", "exon_index = []\n", "exon_refseq_ids = [] #from attributes\n", "exon_dbxref = [] #from attributes\n", "exon_gbkey = [] #from attributes\n", "exon_parent = [] #from attributes\n", "exon_start = []\n", "exon_end = []\n", "exon_seqid = []\n", "exon_strand = []\n", "exon_type = []\n", "transcript_len = []\n", "exon_len = []\n", "# exon_name = [] #from attributes (Exon name not present for all; check for index 50)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_exon_data()] to get exon data of a particular transcript. This function is also used for calculating transcript length and exon length\n", "### Small bug in below function : - It counts genes as exons as well and will get data for all the gene (due to this, exon length and transcript length were wrongly calculated). \n", "#### Above bug has been resolved in the later cells"]}, {"cell_type": "code", "execution_count": 15, "metadata": {}, "outputs": [], "source": ["%%time\n", "def get_exon_data():\n", "    #for i in range(len(gene_index)):\n", "    for i in tqdm(range(0,len(transcript_index),1)):\n", "    # Hard coding for the last index    \n", "        if i == len(transcript_index)-1: # OR if i == transcript_index[-1]\n", "            exon_count = 0  #For dealing with the last case (i.e. gene_index[60958] element)\n", "            total_exon_len = 0\n", "            for k in range(transcript_index[i]+1,len(annotation.df['attributes'])-1,1):\n", "#                 if annotation.df['attributes'][k][0:6] == 'ID=rna':\n", "                exon_count = exon_count + 1\n", "#                 gene_refseq_ids_for_exons.append(gene_refseq_ids[i])\n", "                exon_index.append(k)                \n", "                transcript_refseq_ids_for_exons.append(transcript_refseq_ids[i])\n", "        \n", "##########################_TRANSCRIPT_DATA_COLLECTION_BELOW_#################################################################\n", "                                \n", "                exon_start.append(annotation.df['start'][k])\n", "                exon_end.append(annotation.df['end'][k])\n", "                exon_seqid.append(annotation.df['seq_id'][k])\n", "                exon_strand.append(annotation.df['strand'][k])\n", "                exon_type.append(annotation.df['type'][k])\n", "            \n", "############################_DATA_COLLECTED_FROM_ATTRIBUTES_GIVEN_BELOW_#####################################################################            \n", "            \n", "                exon_parent.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('Parent=')+len('Parent='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Parent=')+len('Parent=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Parent=')+len('Parent=')))])\n", "                exon_refseq_ids.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('ID=')+len('ID='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('ID=')+len('ID=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('ID=')+len('ID=')))])\n", "                exon_dbxref.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('Dbxref=')+len('Dbxref='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Dbxref=')+len('Dbxref=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('Dbxref=')+len('Dbxref=')))])\n", "#                     transcript_name.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Name=')+len('Name='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')))])\n", "                exon_gbkey.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('gbkey=')+len('gbkey='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('gbkey=')+len('gbkey=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('gbkey=')+len('gbkey=')))])\n", "                gene_refseq_ids_for_exons.append(annotation.df['attributes'][k][annotation.df['attributes'][k].find('gene=')+len('gene='):annotation.df['attributes'][k].find('', len(annotation.df['attributes'][k]) if annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('gene=')+len('gene=')) == -1 else annotation.df['attributes'][k].find(';', annotation.df['attributes'][k].find('gene=')+len('gene=')))])\n", "        \n", "############################_CALCULATING_TRANSCRIPT_&_EXON_LENGTH_BELOW_#####################################################################            \n", "        \n", "                trans_len = 0\n", "                start = 0\n", "                end = 0\n", "#                 total_exon_len = 0\n", "                exon_len_sum = 0\n", "                start = int(annotation.df['start'][k])\n", "                end = int(annotation.df['end'][k])\n", "                total_exon_len = total_exon_len + (end - start + 1)\n", "                exon_len_sum = (end - start + 1)\n", "                exon_len.append(exon_len_sum)\n", "                \n", "            transcript_len.append(total_exon_len)\n", "            no_of_exons.append(exon_count)      \n", "            \n", "        else:\n", "            exon_count = 0\n", "            total_exon_len = 0\n", "            for j in range(transcript_index[i]+1,transcript_index[i+1],1): #Taking transcript IDs in pairs and then fetching exon data \n", "#                 if annotation.df['attributes'][j][0:6] == 'ID=rna':\n", "                exon_count = exon_count + 1\n", "                #gene_refseq_ids_for_exons.append(gene_refseq_ids[i])\n", "                exon_index.append(j)\n", "                transcript_refseq_ids_for_exons.append(transcript_refseq_ids[i])\n", "                    \n", "##########################_EXON_DATA_COLLECTION_BELOW_#################################################################\n", "                                \n", "                exon_start.append(annotation.df['start'][j])\n", "                exon_end.append(annotation.df['end'][j])\n", "                exon_seqid.append(annotation.df['seq_id'][j])\n", "                exon_strand.append(annotation.df['strand'][j])\n", "                exon_type.append(annotation.df['type'][j])\n", "            \n", "############################_DATA_COLLECTED_FROM_ATTRIBUTES_GIVEN_BELOW_#####################################################################            \n", "            \n", "                exon_parent.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Parent=')+len('Parent='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Parent=')+len('Parent=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Parent=')+len('Parent=')))])\n", "                exon_refseq_ids.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('ID=')+len('ID='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('ID=')+len('ID=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('ID=')+len('ID=')))])\n", "                exon_dbxref.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Dbxref=')+len('Dbxref='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Dbxref=')+len('Dbxref=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('Dbxref=')+len('Dbxref=')))])\n", "#                     transcript_name.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('Name=')+len('Name='):annotation.df['attributes'][i].find('', len(annotation.df['attributes'][i]) if annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')) == -1 else annotation.df['attributes'][i].find(';', annotation.df['attributes'][i].find('Name=')+len('Name=')))])\n", "                exon_gbkey.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('gbkey=')+len('gbkey='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('gbkey=')+len('gbkey=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('gbkey=')+len('gbkey=')))])\n", "                gene_refseq_ids_for_exons.append(annotation.df['attributes'][j][annotation.df['attributes'][j].find('gene=')+len('gene='):annotation.df['attributes'][j].find('', len(annotation.df['attributes'][j]) if annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('gene=')+len('gene=')) == -1 else annotation.df['attributes'][j].find(';', annotation.df['attributes'][j].find('gene=')+len('gene=')))])\n", "\n", "############################_CALCULATING_TRANSCRIPT_&_EXON_LENGTH_BELOW_#####################################################################            \n", "        \n", "                trans_len = 0\n", "                start = 0\n", "                end = 0\n", "#                 total_exon_len = 0\n", "                exon_len_sum = 0\n", "                start = int(annotation.df['start'][j])\n", "                end = int(annotation.df['end'][j])\n", "                total_exon_len = total_exon_len + (end - start + 1)\n", "                exon_len_sum = (end - start + 1)\n", "                exon_len.append(exon_len_sum)\n", "                \n", "            transcript_len.append(total_exon_len)\n", "            no_of_exons.append(exon_count)      "]}, {"cell_type": "code", "execution_count": 16, "metadata": {}, "outputs": [], "source": ["%%time\n", "get_exon_data()"]}, {"cell_type": "code", "execution_count": 17, "metadata": {}, "outputs": [], "source": ["print(len(no_of_exons))\n", "print(len(gene_refseq_ids_for_exons))\n", "print(len(exon_dbxref))\n", "print(len(exon_gbkey))\n", "print(len(exon_index))\n", "print(len(exon_parent))\n", "print(len(exon_seqid))\n", "print(len(exon_start))\n", "print(len(exon_end))\n", "print(len(exon_type))\n", "print(len(exon_refseq_ids))\n", "print(len(exon_strand))\n", "print(len(exon_len))\n", "print(len(transcript_len))\n", "print(len(transcript_refseq_ids_for_exons))"]}, {"cell_type": "markdown", "metadata": {}, "source": ["#### converting '+' to '1' and '-' to '-1' so as to match it with Ensembl data (just for consistency)"]}, {"cell_type": "code", "execution_count": 18, "metadata": {}, "outputs": [], "source": ["%%time\n", "# exon_strand\n", "changed_exon_strand = []\n", "for i in range(len(exon_strand)):\n", "    if exon_strand[i] == '+':\n", "        changed_exon_strand.append(1)\n", "    else:\n", "        changed_exon_strand.append(-1)\n", "\n", "exon_strand = []\n", "exon_strand = changed_exon_strand"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Exporting exon data to all_exon_data.csv file"]}, {"cell_type": "code", "execution_count": 370, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['SNO','Index', 'RefSeq Gene ID','RefSeq Transcript ID', 'RefSeq Exon ID', 'Dbxref', 'Parent', 'gbkey', 'Type', 'Seq_id', 'Strand','Start', 'End','Exon Length']\n", "\n", "path = '/all_gene_data/all_exon_data.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)"]}, {"cell_type": "code", "execution_count": 372, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(exon_refseq_ids)):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 19, "metadata": {}, "outputs": [], "source": ["%%time\n", "import pandas as pd\n", "\n", "df = pd.read_csv(path)\n", "\n", "df[df.columns[0]] = s_no\n", "df[df.columns[1]] = exon_index\n", "df[df.columns[2]] = gene_refseq_ids_for_exons\n", "df[df.columns[3]] = transcript_refseq_ids_for_exons\n", "df[df.columns[4]] = exon_refseq_ids\n", "df[df.columns[5]] = exon_dbxref\n", "df[df.columns[6]] = exon_parent\n", "df[df.columns[7]] = exon_gbkey\n", "df[df.columns[8]] = exon_type\n", "df[df.columns[9]] = exon_seqid\n", "df[df.columns[10]] = exon_strand\n", "df[df.columns[11]] = exon_start\n", "df[df.columns[12]] = exon_end\n", "df[df.columns[13]] = exon_len\n", "\n", "df.to_csv(path)"]}, {"cell_type": "code", "execution_count": 375, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv('/refseq_numeric/all_gene_data/all_transcript_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 376, "metadata": {}, "outputs": [], "source": ["df['No. of Exons'] = no_of_exons\n", "df['Transcript Length'] = transcript_len\n", "\n", "df.to_csv('/all_gene_data/all_transcript_data.csv')"]}, {"cell_type": "code", "execution_count": 20, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv('/all_gene_data/all_exon_test_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 411, "metadata": {}, "outputs": [], "source": ["df.drop('Exon Length', axis=1,inplace=True)"]}, {"cell_type": "code", "execution_count": 21, "metadata": {}, "outputs": [], "source": ["df.head(5)"]}, {"cell_type": "code", "execution_count": 413, "metadata": {}, "outputs": [{"data": {"text/plain": ["3523384"]}, "execution_count": 413, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df[df.columns[1]])"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Below cells to drop all those exon data which have index lying in 'gene_index'"]}, {"cell_type": "code", "execution_count": 22, "metadata": {}, "outputs": [], "source": ["%%time\n", "c = 0\n", "drop_values = []\n", "s = set(gene_index)\n", "for i in tqdm(range(len(df[df.columns[1]]))):\n", "    if df[df.columns[1]][i] in s:\n", "        c = c+1\n", "        drop_values.append(i)\n", "#         df.drop(i, axis=0,inplace=True)"]}, {"cell_type": "code", "execution_count": 23, "metadata": {}, "outputs": [], "source": ["print(c)\n", "print(len(drop_values))"]}, {"cell_type": "code", "execution_count": 24, "metadata": {}, "outputs": [], "source": ["%%time\n", "df.drop(drop_values, axis=0,inplace=True)"]}, {"cell_type": "code", "execution_count": 418, "metadata": {}, "outputs": [], "source": ["df.drop('SNO', axis=1,inplace=True)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### All exon data = 3462426 rows"]}, {"cell_type": "code", "execution_count": 420, "metadata": {}, "outputs": [{"data": {"text/plain": ["3462426"]}, "execution_count": 420, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df[df.columns[1]])"]}, {"cell_type": "code", "execution_count": 421, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(df[df.columns[1]])):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 422, "metadata": {}, "outputs": [], "source": ["df['SNO'] = s_no"]}, {"cell_type": "code", "execution_count": 423, "metadata": {}, "outputs": [], "source": ["df = df[['SNO','Index', 'RefSeq Gene ID','RefSeq Transcript ID', 'RefSeq Exon ID', 'Dbxref', 'Parent', 'gbkey', 'Type', 'Seq_id', 'Strand','Start', 'End']]     "]}, {"cell_type": "code", "execution_count": 25, "metadata": {}, "outputs": [], "source": ["df.head(5)"]}, {"cell_type": "code", "execution_count": 121, "metadata": {}, "outputs": [], "source": ["exon_len = []\n", "no_of_exons = []\n", "transcript_len = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Below function for correctly calculating exon and transcript length"]}, {"cell_type": "code", "execution_count": 193, "metadata": {}, "outputs": [], "source": ["%%time\n", "def get_exon_transcript_length():\n", "    #for i in range(len(gene_index)):\n", "    count = 0\n", "    dropit = []\n", "    s = set(gene_index)\n", "    for i in tqdm(range(0,len(transcript_index),1)):\n", "    \n", "        if i == len(transcript_index)-1: # OR if i == transcript_index[-1]\n", "            exon_count = 0  #For dealing with the last case (i.e. gene_index[60958] element)\n", "            total_exon_len = 0\n", "            for k in range(transcript_index[i]+1,len(annotation.df['attributes'])-1,1):\n", "#                 if annotation.df['attributes'][k][0:6] == 'ID=rna':\n", "                exon_count = exon_count + 1\n", "                trans_len = 0\n", "                start = 0\n", "                end = 0\n", "                exon_len_sum = 0\n", "                start = int(annotation.df['start'][k])\n", "                end = int(annotation.df['end'][k])\n", "                total_exon_len = total_exon_len + (end - start + 1)\n", "                exon_len_sum = (end - start + 1)\n", "                exon_len.append(exon_len_sum)\n", "                \n", "            transcript_len.append(total_exon_len)\n", "            no_of_exons.append(exon_count)      \n", "            \n", "        else:\n", "            exon_count = 0\n", "            total_exon_len = 0\n", "            for j in range(transcript_index[i]+1,transcript_index[i+1],1): #Taking transcript IDs in pairs and then fetching exon data \n", "#                 if annotation.df['attributes'][j][0:7] == 'ID=exon':\n", "                    \n", "#                     exon_count = exon_count + 1\n", "#                     trans_len = 0\n", "#                     start = 0\n", "#                     end = 0\n", "#                     exon_len_sum = 0\n", "#                     start = int(annotation.df['start'][j])\n", "#                     end = int(annotation.df['end'][j])\n", "#                     total_exon_len = total_exon_len + (end - start + 1)\n", "#                     exon_len_sum = (end - start + 1)\n", "#                     exon_len.append(exon_len_sum)\n", "\n", "#             transcript_len.append(total_exon_len)\n", "#             no_of_exons.append(exon_count)   \n", "\n", "                if j in s:\n", "                    dropit.append(j)\n", "                    count = count + 1\n", "                    break\n", "                    \n", "                elif annotation.df['attributes'][j][0:7] == 'ID=exon':\n", "                    \n", "                    if annotation.df['attributes'][j][0:10] == 'ID=exon-id':\n", "                        continue\n", "                        \n", "                    else: \n", "                        exon_count = exon_count + 1\n", "                        trans_len = 0\n", "                        start = 0\n", "                        end = 0\n", "                        exon_len_sum = 0\n", "                        start = int(annotation.df['start'][j])\n", "                        end = int(annotation.df['end'][j])\n", "                        total_exon_len = total_exon_len + (end - start + 1)\n", "                        exon_len_sum = (end - start + 1)\n", "                        exon_len.append(exon_len_sum)\n", "\n", "            transcript_len.append(total_exon_len)\n", "            no_of_exons.append(exon_count)      "]}, {"cell_type": "code", "execution_count": 192, "metadata": {}, "outputs": [], "source": ["get_exon_transcript_length()"]}, {"cell_type": "code", "execution_count": 191, "metadata": {}, "outputs": [], "source": ["print(len(transcript_len))\n", "print(len(exon_len))\n", "print(len(no_of_exons))"]}, {"cell_type": "code", "execution_count": 440, "metadata": {}, "outputs": [], "source": ["df['Exon Length'] = exon_len"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_gene_data/all_exon_data.csv')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## reset index column "]}, {"cell_type": "code", "execution_count": 29, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv('/refseq_numeric/all_gene_data/all_exon_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 451, "metadata": {}, "outputs": [], "source": ["df.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 453, "metadata": {}, "outputs": [], "source": ["# df.head(5)"]}, {"cell_type": "code", "execution_count": 190, "metadata": {}, "outputs": [], "source": ["# len(df['SNO'])"]}, {"cell_type": "code", "execution_count": 455, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_gene_data/all_exon_data.csv')"]}, {"cell_type": "code", "execution_count": 176, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv('/refseq_numeric/all_gene_data/all_transcript_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 189, "metadata": {}, "outputs": [], "source": ["# df.head(5)"]}, {"cell_type": "code", "execution_count": 178, "metadata": {}, "outputs": [], "source": ["df.drop(['No. of Exons','Transcript Length'], axis=1,inplace=True)"]}, {"cell_type": "code", "execution_count": 179, "metadata": {}, "outputs": [], "source": ["df['No. of Exons'] = no_of_exons\n", "df['Transcript Length'] = transcript_len"]}, {"cell_type": "code", "execution_count": 180, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_gene_data/all_transcript_data_correct.csv')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Note: - \n", "1. gff3 refseq file has 36,94,584 rows out of which first and second rows are not needed. Therefore data to be collected was = 36,94,582\n", "2. Gene data collected has rows = 60,959\n", "3. Transcript data collected has rows = 1,71,197\n", "4. Exon data collected has rows = 34,62,426\n", "5. Gene data [60,959] + Trasnscript data [1,71,197] + Exon data [34,62,426] = required data from gff3 [36,94,582]. \n", "\n", "### Data collected is correct !"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# Correcting no. of exons and transcript length below"]}, {"cell_type": "code", "execution_count": 112, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv('/refseq_numeric/all_gene_data/all_transcript_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 19, "metadata": {}, "outputs": [], "source": ["gene_index = []\n", "transcript_index = []"]}, {"cell_type": "code", "execution_count": 20, "metadata": {}, "outputs": [], "source": ["transcript_index = df['Index'].tolist()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## after this, run the 'get_exon_transcript_length()' function given above"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}