{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Remove NT-NW (below code)"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import pandas as pd"]}, {"cell_type": "code", "execution_count": 33, "metadata": {}, "outputs": [], "source": ["df_all_gene = pd.read_csv('/refseq_numeric/all_type=lncRNA_data/all_lncrna_gene_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 34, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Index</th>\n", "      <th>RefSeq Gene ID</th>\n", "      <th>Dbxref</th>\n", "      <th>Name</th>\n", "      <th>gbkey</th>\n", "      <th>Biotype</th>\n", "      <th>Type</th>\n", "      <th>Seq_id</th>\n", "      <th>Strand</th>\n", "      <th>Start</th>\n", "      <th>End</th>\n", "      <th>No. of Transcripts</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>26</td>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>GeneID:107985730,HGNC:HGNC:52482</td>\n", "      <td>MIR1302-2HG</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>NC_000001.11</td>\n", "      <td>1</td>\n", "      <td>29926</td>\n", "      <td>31295</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>36</td>\n", "      <td>gene-FAM138A</td>\n", "      <td>GeneID:645520,HGNC:HGNC:32334</td>\n", "      <td>FAM138A</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>NC_000001.11</td>\n", "      <td>-1</td>\n", "      <td>34611</td>\n", "      <td>36081</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO  Index    RefSeq Gene ID                            Dbxref  \\\n", "0    1     26  gene-MIR1302-2HG  GeneID:107985730,HGNC:HGNC:52482   \n", "1    2     36      gene-FAM138A     GeneID:645520,HGNC:HGNC:32334   \n", "\n", "          Name gbkey Biotype  Type        Seq_id  Strand  Start    End  \\\n", "0  MIR1302-2HG  Gene  lncRNA  gene  NC_000001.11       1  29926  31295   \n", "1      FAM138A  Gene  lncRNA  gene  NC_000001.11      -1  34611  36081   \n", "\n", "   No. of Transcripts  \n", "0                   1  \n", "1                   1  "]}, "execution_count": 34, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_gene.head(2)"]}, {"cell_type": "code", "execution_count": 35, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['NC_000001.11', 'NT_187361.1', 'NT_187367.1', 'NT_187368.1',\n", "       'NC_000002.12', 'NC_000003.12', 'NT_167215.1', 'NC_000004.12',\n", "       'NC_000005.10', 'NC_000006.12', 'NC_000007.14', 'NC_000008.11',\n", "       'NC_000009.12', 'NT_187373.1', 'NT_187374.1', 'NC_000010.11',\n", "       'NC_000011.10', 'NT_187376.1', 'NC_000012.12', 'NC_000013.11',\n", "       'NC_000014.9', 'NT_113796.3', 'NT_113888.1', 'NT_187381.1',\n", "       'NC_000015.10', 'NT_187382.1', 'NC_000016.10', 'NT_187383.1',\n", "       'NC_000017.11', 'NT_113930.2', 'NC_000018.10', 'NC_000019.10',\n", "       'NC_000020.11', 'NC_000021.9', 'NC_000022.11', 'NT_187386.1',\n", "       'NT_187388.1', 'NT_187389.1', 'NC_000023.11', 'NC_000024.10',\n", "       'NT_187420.1', 'NT_113901.1', 'NT_167213.1', 'NT_167214.1',\n", "       'NT_167218.1', 'NT_187497.1', 'NT_167208.1', 'NT_187498.1',\n", "       'NT_187499.1', 'NT_187500.1', 'NT_187501.1', 'NT_187505.1',\n", "       'NT_187506.1', 'NT_187508.1', 'NT_187509.1', 'NT_187510.1',\n", "       'NT_187513.1', 'NT_113889.1', 'NW_012132914.1', 'NW_011332688.1',\n", "       'NW_014040926.1', 'NW_009646194.1', 'NW_017852928.1',\n", "       'NW_009646196.1', 'NW_011332687.1', 'NW_018654708.1',\n", "       'NW_012132915.1', 'NW_015495299.1', 'NW_018654710.1',\n", "       'NW_011332689.1', 'NW_017363813.1', 'NW_009646197.1',\n", "       'NW_012132916.1', 'NW_011332691.1', 'NW_018654711.1',\n", "       'NW_009646198.1', 'NW_019805491.1', 'NW_013171799.1',\n", "       'NW_013171800.1', 'NW_013171801.1', 'NW_017363814.1',\n", "       'NW_015495300.1', 'NW_015495301.1', 'NW_018654712.1',\n", "       'NW_009646199.1', 'NW_016107297.1', 'NW_016107298.1',\n", "       'NW_018654713.1', 'NW_013171802.1', 'NW_017852929.1',\n", "       'NW_018654714.1', 'NW_018654715.1', 'NW_012132919.1',\n", "       'NW_018654717.1', 'NW_017852932.1', 'NW_017852931.1',\n", "       'NW_018654716.1', 'NW_009646201.1', 'NW_013171806.1',\n", "       'NW_009646202.1', 'NW_011332693.1', 'NW_011332695.1',\n", "       'NW_017363816.1', 'NW_009646203.1', 'NW_018654718.1',\n", "       'NW_011332696.1', 'NW_018654720.1', 'NW_015148967.1',\n", "       'NW_011332697.1', 'NW_019805499.1', 'NW_011332699.1',\n", "       'NW_011332700.1', 'NW_011332698.1', 'NW_018654722.1',\n", "       'NW_011332701.1', 'NW_012132920.1', 'NW_013171812.1',\n", "       'NW_019805500.1', 'NW_017852933.1', 'NW_013171813.1',\n", "       'NW_018654723.1', 'NW_017363817.1', 'NW_016107299.1',\n", "       'NW_017363819.1', 'NW_017363818.1', 'NW_019805503.1',\n", "       'NW_019805502.1', 'NW_013171814.1', 'NW_018654724.1',\n", "       'NW_014040929.1', 'NW_016107300.1', 'NW_016107301.1',\n", "       'NW_016107302.1', 'NW_016107303.1', 'NW_016107304.1',\n", "       'NW_016107305.1', 'NW_016107306.1', 'NW_016107307.1',\n", "       'NW_016107308.1', 'NW_016107309.1', 'NW_016107310.1',\n", "       'NW_016107311.1', 'NW_016107313.1', 'NW_009646207.1',\n", "       'NW_014040930.1', 'NW_014040931.1', 'NW_009646208.1',\n", "       'NW_015148968.1', 'NW_015148969.1', 'NW_018654726.1',\n", "       'NW_009646209.1', 'NT_187515.1', 'NT_187514.1', 'NT_187520.1',\n", "       'NW_003315907.2', 'NT_187519.1', 'NT_187516.1', 'NT_187518.1',\n", "       'NT_187525.1', 'NW_003315908.1', 'NT_187527.1', 'NT_187523.1',\n", "       'NT_187535.1', 'NT_187537.1', 'NW_003315913.1', 'NT_187533.1',\n", "       'NT_187538.1', 'NT_187532.1', 'NT_187534.1', 'NT_187539.1',\n", "       'NT_187540.1', 'NW_003315915.1', 'NW_003315914.1', 'NT_187542.1',\n", "       'NT_187545.1', 'NT_187543.1', 'NT_187550.1', 'NT_187548.1',\n", "       'NT_187547.1', 'NW_003315920.1', 'NW_003571036.1',\n", "       'NW_003315917.2', 'NW_003315918.1', 'NT_187549.1',\n", "       'NW_003315919.1', 'NT_187546.1', 'NT_167244.2', 'NW_003315921.1',\n", "       'NT_187556.1', 'NT_187557.1', 'NW_004166862.2', 'NT_187552.1',\n", "       'NT_187553.1', 'NT_187558.1', 'NT_187561.1', 'NT_187559.1',\n", "       'NW_003315922.2', 'NT_187562.1', 'NT_187560.1', 'NT_187572.1',\n", "       'NT_187565.1', 'NT_187576.1', 'NT_187570.1', 'NT_187577.1',\n", "       'NT_187566.1', 'NT_187567.1', 'NT_187574.1', 'NT_187573.1',\n", "       'NT_187571.1', 'NT_187569.1', 'NW_003315929.1', 'NW_003315930.1',\n", "       'NT_187578.1', 'NW_003315934.1', 'NT_187579.1', 'NW_003315935.1',\n", "       'NT_187580.1', 'NT_187586.1', 'NT_187584.1', 'NT_187585.1',\n", "       'NT_187583.1', 'NW_003871074.1', 'NT_187582.1', 'NT_187581.1',\n", "       'NW_003571049.1', 'NW_003571050.1', 'NT_187587.1',\n", "       'NW_003315939.2', 'NW_003315941.1', 'NW_003315940.1',\n", "       'NT_187589.1', 'NT_187594.1', 'NT_187592.1', 'NT_187596.1',\n", "       'NT_187598.1', 'NT_187601.1', 'NT_187600.1', 'NT_187602.1',\n", "       'NT_187604.1', 'NW_003315943.1', 'NT_187605.1', 'NW_003315944.2',\n", "       'NT_187606.1', 'NT_187610.1', 'NT_187609.1', 'NT_187608.1',\n", "       'NT_187607.1', 'NW_003315952.3', 'NT_187613.1', 'NT_187611.1',\n", "       'NT_187614.1', 'NW_003871091.1', 'NW_003315953.2', 'NT_167251.2',\n", "       'NW_003315955.1', 'NW_003315956.1', 'NW_003315958.1',\n", "       'NW_003315961.1', 'NT_187617.1', 'NW_003315962.1',\n", "       'NW_003315965.1', 'NT_187620.1', 'NW_003571054.1',\n", "       'NW_003315966.2', 'NT_187625.1', 'NT_187624.1', 'NT_187628.1',\n", "       'NT_187627.1', 'NW_003315968.2', 'NW_003315969.2', 'NT_187629.1',\n", "       'NT_187632.1', 'NT_187633.1', 'NT_187631.1', 'NW_003315972.2',\n", "       'NW_003315971.2', 'NT_187634.1', 'NT_187646.1', 'NT_187648.1',\n", "       'NT_187647.1', 'NT_187649.1', 'NT_187650.1', 'NT_187651.1',\n", "       'NT_187652.1', 'NT_113891.3', 'NT_187653.1', 'NT_187655.1',\n", "       'NT_187654.1', 'NT_187657.1', 'NT_187658.1', 'NT_187660.1',\n", "       'NT_187664.1', 'NT_187661.1', 'NT_187663.1', 'NT_187665.1',\n", "       'NT_187666.1', 'NW_003571055.2', 'NW_004504305.1', 'NT_187667.1',\n", "       'NT_187678.1', 'NT_187679.1', 'NT_167245.2', 'NT_187680.1',\n", "       'NT_187681.1', 'NW_003571056.2', 'NT_187682.1', 'NT_187688.1',\n", "       'NT_167246.2', 'NW_003571057.2', 'NT_187689.1', 'NT_167247.2',\n", "       'NW_003571058.2', 'NT_187690.1', 'NT_167248.2', 'NW_003571059.2',\n", "       'NT_187691.1', 'NT_167249.2', 'NW_003571060.1', 'NT_187692.1',\n", "       'NW_003571061.2', 'NT_187693.1', 'NT_187636.1', 'NT_187637.1',\n", "       'NT_187639.1', 'NT_187640.1', 'NT_187641.1', 'NT_187643.1',\n", "       'NT_187645.1', 'NT_187668.1', 'NT_187669.1', 'NT_187671.1',\n", "       'NT_187672.1', 'NT_187673.1', 'NT_187674.1', 'NT_187675.1',\n", "       'NT_187676.1', 'NT_187683.1', 'NT_187684.1', 'NT_187685.1',\n", "       'NT_187686.1', 'NT_187687.1'], dtype=object)"]}, "execution_count": 35, "metadata": {}, "output_type": "execute_result"}], "source": ["df_all_gene['Seq_id'].unique()"]}, {"cell_type": "code", "execution_count": 36, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 37, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 6 \u00b5s, sys: 0 ns, total: 6 \u00b5s\n", "Wall time: 7.87 \u00b5s\n"]}], "source": ["%%time\n", "def drop_other_seq_region_name(path):\n", "    c = 0\n", "    drop_index = []\n", "    df = pd.read_csv(path, index_col=0)\n", "#     df['Seq_id'] = df['Seq_id'].astype(str)\n", "    for i in tqdm(range(len(df['Seq_id']))):\n", "        if (df['Seq_id'][i]).startswith(('NT', 'NW', 'NC_0129')):\n", "            drop_index.append(i)\n", "            c = c + 1\n", "        else:\n", "            continue\n", "    df.drop(drop_index, axis=0,inplace=True)\n", "    df.reset_index(drop=True, inplace=True)\n", "    return df, c, drop_index"]}, {"cell_type": "code", "execution_count": 50, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 27991/27991 [00:00<00:00, 38102.95it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["CPU times: user 1.61 s, sys: 175 ms, total: 1.78 s\n", "Wall time: 1.93 s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "path = '/all_type=lncRNA_data/all_lncrna_transcript_data_with_sequences_corrected.csv'\n", "df, c, drop_index = drop_other_seq_region_name(path)"]}, {"cell_type": "code", "execution_count": 51, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['NC_000001.11', 'NC_000002.12', 'NC_000003.12', 'NC_000004.12',\n", "       'NC_000005.10', 'NC_000006.12', 'NC_000007.14', 'NC_000008.11',\n", "       'NC_000009.12', 'NC_000010.11', 'NC_000011.10', 'NC_000012.12',\n", "       'NC_000013.11', 'NC_000014.9', 'NC_000015.10', 'NC_000016.10',\n", "       'NC_000017.11', 'NC_000018.10', 'NC_000019.10', 'NC_000020.11',\n", "       'NC_000021.9', 'NC_000022.11', 'NC_000023.11', 'NC_000024.10'],\n", "      dtype=object)"]}, "execution_count": 51, "metadata": {}, "output_type": "execute_result"}], "source": ["df['Seq_id'].unique()"]}, {"cell_type": "code", "execution_count": 52, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_type=lncRNA_data/without_NW-NT-NC0129_regions/all_lncrna_transcript_data_sequences_without_NT-NW_seq.csv')"]}, {"cell_type": "code", "execution_count": 53, "metadata": {}, "outputs": [{"data": {"text/plain": ["25715"]}, "execution_count": 53, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## convert NC to numeric and sort via chr (below cells)"]}, {"cell_type": "code", "execution_count": 54, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 6 \u00b5s, sys: 1 \u00b5s, total: 7 \u00b5s\n", "Wall time: 10 \u00b5s\n"]}], "source": ["%%time\n", "def sort_file_via_chr(path):\n", "    df = pd.read_csv(path, index_col=0)\n", "    di1 = {'NC_000001.11': 1, 'NC_000002.12': 2, 'NC_000003.12': 3, 'NC_000004.12': 4, 'NC_000005.10': 5, 'NC_000006.12': 6, 'NC_000007.14': 7, 'NC_000008.11': 8, 'NC_000009.12': 9, 'NC_000010.11': 10, 'NC_000011.10': 11, 'NC_000012.12': 12, 'NC_000013.11': 13, 'NC_000014.9': 14, 'NC_000015.10': 15, 'NC_000016.10': 16, 'NC_000017.11': 17, 'NC_000018.10': 18, 'NC_000019.10': 19, 'NC_000020.11': 20, 'NC_000021.9': 21, 'NC_000022.11': 22, 'NC_000023.11': 23, 'NC_000024.10': 24}\n", "    df['Seq_id'].replace(di1, inplace=True)\n", "#     df['Seq region Name'] = df['Seq region Name'].map({'1': 1, '2': 2, '3': 3, '4': 4, '5': 5, '6': 6, '7': 7, '8': 8, '9': 9, '10': 10, '11': 11, '12': 12, '13': 13, '14': 14, '15': 15, '16': 16, '17': 17, '18': 18, '19': 19, '20': 20, '21': 21, '22': 22, 'X': 23, 'Y': 24 })\n", "    df_sorted = df.sort_values(by=[\"Seq_id\"], ascending=True)\n", "    df_sorted['Seq_id'] = df_sorted['Seq_id'].astype(str)\n", "    di = {'23': \"X\", '24': \"Y\"}\n", "    df_sorted['Seq_id'].replace(di, inplace=True)\n", "    df_sorted.reset_index(drop=True, inplace=True)\n", "    return df_sorted"]}, {"cell_type": "code", "execution_count": 73, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 1 s, sys: 150 ms, total: 1.15 s\n", "Wall time: 1.32 s\n"]}], "source": ["%%time\n", "path = '/refseq_numeric/all_type=lncRNA_data/without_NW-NT-NC0129_regions/all_lncrna_transcript_data_sequences_without_NT-NW_seq.csv'\n", "df_sorted = sort_file_via_chr(path)"]}, {"cell_type": "code", "execution_count": 74, "metadata": {}, "outputs": [{"data": {"text/plain": ["array(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12',\n", "       '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X',\n", "       'Y'], dtype=object)"]}, "execution_count": 74, "metadata": {}, "output_type": "execute_result"}], "source": ["df_sorted['Seq_id'].unique()"]}, {"cell_type": "code", "execution_count": 75, "metadata": {}, "outputs": [], "source": ["df_sorted.to_csv('/dataset/refseq_numeric/all_type=lncRNA_data/all_lncrna_sorted_chr/all_lncrna_transcript_data_sequence_sorted_chr.csv')"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}