{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Loading all gene IDs from the csv file"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import csv \n", "import pandas as pd\n", "import numpy as np\n", "import json"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [], "source": ["path = '/all_gene_data/all_gene_ids.csv'\n", "\n", "df = pd.read_csv(path)\n", "gene_ids = df['All gene ID'].tolist()\n", "cleaned_ids = []\n", "cleaned_ids = gene_ids"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## calculating time taken to open 1.05GB json file"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## 'read_content' variable (given in below cell) contains the json response received"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 12.9 s, sys: 5.86 s, total: 18.8 s\n", "Wall time: 21.7 s\n"]}], "source": ["%%time\n", "with open(\"/merged_all_gene_data(dict).json\") as access_json:\n", "    read_content = json.load(access_json)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Get all gene IDs having biotype as 'lincRNA'"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [], "source": ["lincrna_ids = []"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [], "source": ["def get_all_lincrna_ids():\n", "    for i in range(len(cleaned_ids)):\n", "        if read_content[cleaned_ids[i]]['biotype'] == 'lincRNA':\n", "            lincrna_ids.append(cleaned_ids[i])\n", "        else:\n", "            continue\n"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [{"data": {"text/plain": ["7690"]}, "execution_count": 6, "metadata": {}, "output_type": "execute_result"}], "source": ["get_all_lincrna_ids()\n", "len(lincrna_ids)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Saving lincRNA gene IDs in a csv file"]}, {"cell_type": "code", "execution_count": 32, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['SNO', 'All lincRNA ID']\n", "\n", "path = '/all_lincRNA_data/all_lincrna_ids.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)\n"]}, {"cell_type": "code", "execution_count": 33, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(lincrna_ids)):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 34, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv(path)\n", "\n", "df[df.columns[0]] = s_no\n", "df[df.columns[1]] = lincrna_ids\n", "\n", "df.to_csv(path)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Loading all lincRNA IDs from csv file to a list"]}, {"cell_type": "code", "execution_count": 7, "metadata": {}, "outputs": [], "source": ["cleaned_ids = []\n", "path = '/all_lincRNA_data/all_lincrna_ids.csv'"]}, {"cell_type": "code", "execution_count": 8, "metadata": {}, "outputs": [], "source": ["df = pd.read_csv(path)\n", "cleaned_ids = df['All lincRNA ID'].tolist()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Total lincRNA gene IDs = 7690"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [{"data": {"text/plain": ["7690"]}, "execution_count": 9, "metadata": {}, "output_type": "execute_result"}], "source": ["len(cleaned_ids)"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["gene_display_name = []\n", "gene_start = []\n", "gene_end = []\n", "gene_strand = []\n", "gene_seq_region_name = []\n", "null_gene_ids = []\n", "gene_ids = []\n", "gene_biotype = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_gene_data() ] to extract 'gene' data. Data Extracted are :\n", "1. gene display_name\n", "2. gene start\n", "3. gene end\n", "4. gene strand\n", "5. gene seq_region_name\n", "6. gene biotype"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [], "source": ["def get_gene_data():\n", "    \n", "    count = 0\n", "    for i in range(len(cleaned_ids)):\n", "        if read_content[cleaned_ids[i]] == None:\n", "            null_gene_ids.append([cleaned_ids[i]])\n", "            #del cleaned_ids[i]\n", "        else:\n", "            \n", "            gene_ids.append(cleaned_ids[i])\n", "            gene_display_name.append(read_content[cleaned_ids[i]]['display_name'])\n", "            gene_start.append(read_content[cleaned_ids[i]]['start'])\n", "            gene_end.append(read_content[cleaned_ids[i]]['end'])\n", "            gene_strand.append(read_content[cleaned_ids[i]]['strand'])\n", "            gene_seq_region_name.append(read_content[cleaned_ids[i]]['seq_region_name'])\n", "            gene_biotype.append(read_content[cleaned_ids[i]]['biotype'])\n", "            if cleaned_ids[i] in read_content:\n", "                count = count + 1\n", "        \n", "    #print(count) "]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [], "source": ["get_gene_data()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## 'null_gene_ids' variable contains the IDs having null values"]}, {"cell_type": "code", "execution_count": 13, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["No. of Null IDs are 0\n", "Null IDs are :\n"]}], "source": ["print('No. of Null IDs are {0}'.format(len(null_gene_ids)))\n", "print('Null IDs are :')\n", "for i in range(len(null_gene_ids)):\n", "    \n", "    print(null_gene_ids[i])"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["print('No. of contents of gene_start is {0}'.format(len(gene_start)))\n", "print('No. of contents of gene_end is {0}'.format(len(gene_end)))\n", "print('No. of contents of gene_strand is {0}'.format(len(gene_strand)))\n", "print('No. of contents of gene_seq_region_name is {0}'.format(len(gene_seq_region_name)))\n", "print('No. of contents of gene_display_name is {0}'.format(len(gene_display_name)))"]}, {"cell_type": "code", "execution_count": 14, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["7690\n"]}], "source": ["cleaned_ids = []\n", "cleaned_ids = gene_ids\n", "print(len(cleaned_ids))"]}, {"cell_type": "code", "execution_count": 15, "metadata": {}, "outputs": [], "source": ["no_of_transcripts = []\n", "gene_ids_for_transcripts = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [ get_no_of_transcripts() ] to calculate no. of transcripts in a particular gene"]}, {"cell_type": "code", "execution_count": 16, "metadata": {}, "outputs": [], "source": ["def get_no_of_transcripts():\n", "    for i in range(len(cleaned_ids)):\n", "        no_of_transcripts.append(len(read_content[cleaned_ids[i]]['Transcript']))\n", "        \n", "        for k in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "            gene_ids_for_transcripts.append(cleaned_ids[i])\n", "            \n", "#     for j in range(len(cleaned_ids)):\n", "#         print('No. of transcripts in gene \"{0}\" are {1}'.format(cleaned_ids[j],no_of_transcripts[j]))"]}, {"cell_type": "code", "execution_count": 17, "metadata": {}, "outputs": [], "source": ["get_no_of_transcripts()"]}, {"cell_type": "code", "execution_count": 18, "metadata": {}, "outputs": [], "source": ["transcript_id = []\n", "transcript_start = []\n", "transcript_end = []\n", "transcript_biotype = []\n", "transcript_strand = []\n", "transcript_seq_region_name = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_transcript_data() ] to extract 'transcript' data. Data Extracted are :\n", "1. transcript id\n", "2. transcript start\n", "3. transcript end\n", "4. transcript biotype\n", "5. transcript strand\n", "6. transcript seq_region_name"]}, {"cell_type": "code", "execution_count": 19, "metadata": {}, "outputs": [], "source": ["def get_transcript_data():\n", "    for i in range(len(cleaned_ids)):\n", "        \n", "        for j in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "            transcript_id.append(read_content[cleaned_ids[i]]['Transcript'][j]['id'])\n", "            transcript_start.append(read_content[cleaned_ids[i]]['Transcript'][j]['start'])\n", "            transcript_end.append(read_content[cleaned_ids[i]]['Transcript'][j]['end'])\n", "            transcript_biotype.append(read_content[cleaned_ids[i]]['Transcript'][j]['biotype'])\n", "            transcript_strand.append(read_content[cleaned_ids[i]]['Transcript'][j]['strand'])\n", "            transcript_seq_region_name.append(read_content[cleaned_ids[i]]['Transcript'][j]['seq_region_name'])\n", "            \n", "#     for k in range(len(gene_ids_for_transcripts)):\n", "#         print('Transcript \"{0}\" of gene ID \"{1}\" has start and end as : \"{2}\" & \"{3}\"'.format(transcript_id[k],gene_ids_for_transcripts[k],transcript_start[k],transcript_end[k]))"]}, {"cell_type": "code", "execution_count": 20, "metadata": {}, "outputs": [], "source": ["get_transcript_data()"]}, {"cell_type": "code", "execution_count": 21, "metadata": {}, "outputs": [], "source": ["print(len(transcript_id))\n", "print(len(transcript_start))\n", "print(len(transcript_end))\n", "print(len(gene_ids_for_transcripts))"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["len(read_content[cleaned_ids[0]]['Transcript'][0][\"Exon\"])"]}, {"cell_type": "code", "execution_count": 41, "metadata": {}, "outputs": [], "source": ["no_of_exons = []\n", "transcript_ids_for_exons = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [ get_no_of_exons() ] to calculate no. of exons for a particular transcript"]}, {"cell_type": "code", "execution_count": 42, "metadata": {}, "outputs": [], "source": ["def get_no_of_exons():\n", "    for i in range(len(cleaned_ids)):\n", "        for j in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "            no_of_exons.append(len(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"]))\n", "        \n", "            for k in range(len(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"])):\n", "                transcript_ids_for_exons.append(read_content[cleaned_ids[i]]['Transcript'][j]['id'])\n", "            \n", "#     for l in range(len(cleaned_ids)):\n", "#         print('No. of exons in transcript \"{0}\" are {1}'.format(transcript_id[l],no_of_exons[l]))"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["len(read_content[cleaned_ids[0]]['Transcript'][0][\"Exon\"])"]}, {"cell_type": "code", "execution_count": 43, "metadata": {}, "outputs": [], "source": ["get_no_of_exons()"]}, {"cell_type": "code", "execution_count": 24, "metadata": {}, "outputs": [{"data": {"text/plain": ["43789"]}, "execution_count": 24, "metadata": {}, "output_type": "execute_result"}], "source": ["sum(no_of_exons)"]}, {"cell_type": "code", "execution_count": 44, "metadata": {}, "outputs": [{"data": {"text/plain": ["43789"]}, "execution_count": 44, "metadata": {}, "output_type": "execute_result"}], "source": ["len(transcript_ids_for_exons)"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["read_content[cleaned_ids[0]]['Transcript'][0][\"Exon\"][0]"]}, {"cell_type": "code", "execution_count": 26, "metadata": {}, "outputs": [], "source": ["exon_id = []\n", "exon_start = []\n", "exon_end = []\n", "gene_ids_for_exons = []\n", "exon_strand = []\n", "exon_seq_region_name = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_exon_data() ] to extract 'exon' data. Data Extracted are :\n", "1. exon id\n", "2. exon start\n", "3. exon end\n", "4. Exon strand\n", "5. Exon seq region name"]}, {"cell_type": "code", "execution_count": 27, "metadata": {}, "outputs": [], "source": ["def get_exon_data():\n", "    for i in range(len(cleaned_ids)):\n", "        \n", "        for j in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "            for k in range(len(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"])):\n", "                \n", "                \n", "                exon_id.append(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['id'])\n", "                exon_start.append(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['start'])\n", "                exon_end.append(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['end'])\n", "                exon_strand.append(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['strand'])\n", "                exon_seq_region_name.append(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['seq_region_name'])\n", "                gene_ids_for_exons.append(cleaned_ids[i])\n", "        \n", "            \n", "#     for l in range(len(transcript_ids_for_exons)):\n", "        \n", "#         print('Exon \"{0}\" of Transcript ID \"{1}\" having gene ID \"{2}\" has start and end as : \"{3}\" & \"{4}\"'.format(exon_id[l],transcript_ids_for_exons[l],gene_ids_for_exons[l],exon_start[l],exon_end[l]))\n", "    "]}, {"cell_type": "code", "execution_count": 28, "metadata": {}, "outputs": [], "source": ["get_exon_data()"]}, {"cell_type": "code", "execution_count": 30, "metadata": {}, "outputs": [], "source": ["len(exon_id)"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["len(gene_ids_for_exons)"]}, {"cell_type": "code", "execution_count": 29, "metadata": {}, "outputs": [], "source": ["transcript_len = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function[ get_transcript_length() ] to calculate length of transcript"]}, {"cell_type": "code", "execution_count": 30, "metadata": {}, "outputs": [], "source": ["def get_transcript_length():\n", " #   for i in range(transcript_id):\n", "  #      for j in range(exon)\n", "\n", "    for i in range(len(cleaned_ids)):\n", "        \n", "        for j in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "            trans_len = 0\n", "            start = 0\n", "            end = 0\n", "            total_exon_len = 0\n", "            for k in range(len(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"])):\n", "                start = read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['start']\n", "                end = read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['end']\n", "                total_exon_len = total_exon_len + (end - start + 1)\n", "\n", "            transcript_len.append(total_exon_len)\n", "    \n", "#     for k in range(len(transcript_id)):\n", "#         print('Transcript ID \"{0}\" has length of {1} bps'.format(transcript_id[k], transcript_len[k]))\n", "                \n", "        "]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["len(transcript_id)"]}, {"cell_type": "code", "execution_count": 31, "metadata": {}, "outputs": [], "source": ["get_transcript_length()"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["len(transcript_len)"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["transcript_len[-1]"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["transcript_id[-1]"]}, {"cell_type": "code", "execution_count": 32, "metadata": {}, "outputs": [], "source": ["exon_len = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function[ get_exon_length() ] to calculate length of exon"]}, {"cell_type": "code", "execution_count": 33, "metadata": {}, "outputs": [], "source": ["def get_exon_length():\n", " #   for i in range(transcript_id):\n", "  #      for j in range(exon)\n", "#exon_id\n", "    for i in range(len(cleaned_ids)):\n", "        \n", "        for j in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "           # exon_len = 0\n", "           # start = 0\n", "           # end = 0\n", "           # exon_len = 0\n", "            for k in range(len(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"])):\n", "                start = 0\n", "                end = 0\n", "                exon_len_sum = 0\n", "                start = read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['start']\n", "                end = read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['end']\n", "                exon_len_sum = (end - start + 1)\n", "\n", "                exon_len.append(exon_len_sum)\n", "    \n", "#     for k in range(len(exon_id)):\n", "#         print('Exon ID \"{0}\" has length of {1} bps'.format(exon_id[k], exon_len[k]))\n", "                \n", "        "]}, {"cell_type": "code", "execution_count": 34, "metadata": {}, "outputs": [], "source": ["get_exon_length()"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["len(exon_len)"]}, {"cell_type": "code", "execution_count": 48, "metadata": {}, "outputs": [{"data": {"text/plain": ["1279073"]}, "execution_count": 48, "metadata": {}, "output_type": "execute_result"}], "source": ["len(exon_id)"]}, {"cell_type": "code", "execution_count": 46, "metadata": {}, "outputs": [{"data": {"text/plain": ["'ENSG00000182586'"]}, "execution_count": 46, "metadata": {}, "output_type": "execute_result"}], "source": ["gene_ids_for_exons[904]"]}, {"cell_type": "code", "execution_count": 45, "metadata": {}, "outputs": [{"data": {"text/plain": ["'ENST00000584169'"]}, "execution_count": 45, "metadata": {}, "output_type": "execute_result"}], "source": ["transcript_ids_for_exons[904]"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Exporting gene data to all_lincrna_gene_data.csv file"]}, {"cell_type": "code", "execution_count": 63, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['SNO', 'Gene ID', 'Display Name', 'Biotype', 'Start', 'End', 'Strand', 'Seq region Name', 'No. of Transcripts']\n", "\n", "path = '/all_lincRNA_data/all_lincrna_gene_data.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)\n"]}, {"cell_type": "code", "execution_count": 64, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(cleaned_ids)):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 65, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv(path)\n", "\n", "df[df.columns[0]] = s_no\n", "df[df.columns[1]] = cleaned_ids\n", "df[df.columns[2]] = gene_display_name\n", "df[df.columns[3]] = gene_biotype\n", "df[df.columns[4]] = gene_start\n", "df[df.columns[5]] = gene_end\n", "df[df.columns[6]] = gene_strand\n", "df[df.columns[7]] = gene_seq_region_name\n", "df[df.columns[8]] = no_of_transcripts\n", "\n", "\n", "df.to_csv(path)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Exporting transcript data to all_lincrna_transcript_data.csv file"]}, {"cell_type": "code", "execution_count": 66, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['SNO', 'Gene ID', 'Transcript ID', 'Biotype', 'Strand', 'Seq region Name', 'Transcript Start', 'Transcript End', 'Transcript Length','No. of Exons']\n", "\n", "path = '/all_lincRNA_data/all_lincrna_transcript_data.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)\n"]}, {"cell_type": "code", "execution_count": 67, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(transcript_id)):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 68, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv(path)\n", "\n", "df[df.columns[0]] = s_no\n", "df[df.columns[1]] = gene_ids_for_transcripts\n", "df[df.columns[2]] = transcript_id\n", "df[df.columns[3]] = transcript_biotype\n", "df[df.columns[4]] = transcript_strand\n", "df[df.columns[5]] = transcript_seq_region_name\n", "df[df.columns[6]] = transcript_start\n", "df[df.columns[7]] = transcript_end\n", "df[df.columns[8]] = transcript_len\n", "df[df.columns[9]] = no_of_exons\n", "\n", "df.to_csv(path)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Exporting exon data to all_lincrna_exon_data.csv file"]}, {"cell_type": "code", "execution_count": 47, "metadata": {}, "outputs": [], "source": ["import csv \n", "header = ['SNO', 'Gene ID', 'Transcript ID', 'Exon ID', 'Strand','Seq region Name','Exon Start', 'Exon End', 'Exon Length']\n", "\n", "path = '/all_lincRNA_data/all_lincrna_exon_data.csv'\n", "\n", "with open(path, 'wt', newline ='') as file:\n", "    writer = csv.writer(file, delimiter=',')\n", "    writer.writerow(i for i in header)\n"]}, {"cell_type": "code", "execution_count": 48, "metadata": {}, "outputs": [], "source": ["s_no = []\n", "for i in range(len(exon_id)):\n", "    s_no.append(i+1)"]}, {"cell_type": "code", "execution_count": 49, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "\n", "df = pd.read_csv(path)\n", "\n", "df[df.columns[0]] = s_no\n", "df[df.columns[1]] = gene_ids_for_exons\n", "df[df.columns[2]] = transcript_ids_for_exons\n", "df[df.columns[3]] = exon_id\n", "df[df.columns[4]] = exon_strand\n", "df[df.columns[5]] = exon_seq_region_name\n", "df[df.columns[6]] = exon_start\n", "df[df.columns[7]] = exon_end\n", "df[df.columns[8]] = exon_len\n", "\n", "df.to_csv(path)"]}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}