{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Loading all gene IDs from the csv file"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import csv \n", "import pandas as pd\n", "import numpy as np\n", "import json"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [], "source": ["path = '/all_gene_data/all_gene_ids.csv'\n", "\n", "df = pd.read_csv(path)\n", "gene_ids = df['All gene ID'].tolist()\n", "cleaned_ids = []\n", "cleaned_ids = gene_ids"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## calculating time taken to open 1.05GB json file"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## 'read_content' variable (given in below cell) contains the json response received"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 14.2 s, sys: 5.61 s, total: 19.8 s\n", "Wall time: 24.2 s\n"]}], "source": ["%%time\n", "with open(\"/merged_all_gene_data(dict).json\") as access_json:\n", "    read_content = json.load(access_json)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Get all gene IDs having biotype as 'lincRNA'"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [], "source": ["lincrna_ids = []"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [], "source": ["def get_all_lincrna_ids():\n", "    for i in range(len(cleaned_ids)):\n", "        if read_content[cleaned_ids[i]]['biotype'] == 'lincRNA':\n", "            lincrna_ids.append(cleaned_ids[i])\n", "        else:\n", "            continue\n"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [{"data": {"text/plain": ["7690"]}, "execution_count": 6, "metadata": {}, "output_type": "execute_result"}], "source": ["get_all_lincrna_ids()\n", "len(lincrna_ids)"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Loading all lincRNA IDs from csv file to a list"]}, {"cell_type": "code", "execution_count": 7, "metadata": {}, "outputs": [], "source": ["cleaned_ids = []\n", "path = '/all_lincRNA_data/all_lincrna_ids.csv'"]}, {"cell_type": "code", "execution_count": 8, "metadata": {}, "outputs": [], "source": ["df = pd.read_csv(path)\n", "cleaned_ids = df['All lincRNA ID'].tolist()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Total lincRNA gene IDs = 7690"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [{"data": {"text/plain": ["7690"]}, "execution_count": 9, "metadata": {}, "output_type": "execute_result"}], "source": ["len(cleaned_ids)"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm\n", "gene_ids = []\n", "null_gene_ids = []"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [], "source": ["def get_gene_data():\n", "    \n", "    count = 0\n", "    for i in tqdm(range(len(cleaned_ids))):\n", "        if read_content[cleaned_ids[i]] == None:\n", "            null_gene_ids.append([cleaned_ids[i]])\n", "            #del cleaned_ids[i]\n", "        else:\n", "            \n", "            gene_ids.append(cleaned_ids[i])\n", "#             gene_display_name.append(read_content[cleaned_ids[i]]['display_name'])\n", "#             gene_start.append(read_content[cleaned_ids[i]]['start'])\n", "#             gene_end.append(read_content[cleaned_ids[i]]['end'])\n", "#             gene_strand.append(read_content[cleaned_ids[i]]['strand'])\n", "#             gene_seq_region_name.append(read_content[cleaned_ids[i]]['seq_region_name'])\n", "#             gene_biotype.append(read_content[cleaned_ids[i]]['biotype'])\n", "            if cleaned_ids[i] in read_content:\n", "                count = count + 1\n", "        \n", "    #print(count) "]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 7690/7690 [00:00<00:00, 508124.17it/s]\n"]}], "source": ["get_gene_data()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## 'null_gene_ids' variable contains the IDs having null values"]}, {"cell_type": "code", "execution_count": 13, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["No. of Null IDs are 0\n", "Null IDs are :\n"]}], "source": ["print('No. of Null IDs are {0}'.format(len(null_gene_ids)))\n", "print('Null IDs are :')\n", "for i in range(len(null_gene_ids)):\n", "    \n", "    print(null_gene_ids[i])"]}, {"cell_type": "code", "execution_count": 14, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["7690\n"]}], "source": ["cleaned_ids = []\n", "cleaned_ids = gene_ids\n", "print(len(cleaned_ids))"]}, {"cell_type": "code", "execution_count": 17, "metadata": {}, "outputs": [], "source": ["transcript_strand = []\n", "transcript_seq_region_name = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_transcript_data() ] to extract 'transcript' data. Data Extracted are :\n", "1. transcript id\n", "2. transcript start\n", "3. transcript end\n", "4. transcript biotype"]}, {"cell_type": "code", "execution_count": 18, "metadata": {}, "outputs": [], "source": ["def get_transcript_data():\n", "    for i in range(len(cleaned_ids)):\n", "        \n", "        for j in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "            transcript_strand.append(read_content[cleaned_ids[i]]['Transcript'][j]['strand'])\n", "            transcript_seq_region_name.append(read_content[cleaned_ids[i]]['Transcript'][j]['seq_region_name'])\n", "            \n", "#     for k in range(len(gene_ids_for_transcripts)):\n", "#         print('Transcript \"{0}\" of gene ID \"{1}\" has start and end as : \"{2}\" & \"{3}\"'.format(transcript_id[k],gene_ids_for_transcripts[k],transcript_start[k],transcript_end[k]))"]}, {"cell_type": "code", "execution_count": 19, "metadata": {}, "outputs": [], "source": ["get_transcript_data()"]}, {"cell_type": "code", "execution_count": 20, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["13666\n", "13666\n"]}], "source": ["print(len(transcript_strand))\n", "print(len(transcript_seq_region_name))"]}, {"cell_type": "code", "execution_count": 23, "metadata": {}, "outputs": [], "source": ["exon_strand = []\n", "exon_seq_region_name = []"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Below function [get_exon_data() ] to extract 'exon' data. Data Extracted are :\n", "1. exon id\n", "2. exon start\n", "3. exon end"]}, {"cell_type": "code", "execution_count": 24, "metadata": {}, "outputs": [], "source": ["def get_exon_data():\n", "    for i in tqdm(range(len(cleaned_ids))):\n", "        \n", "        for j in range(len(read_content[cleaned_ids[i]]['Transcript'])):\n", "            for k in range(len(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"])):\n", "                \n", "                exon_strand.append(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['strand'])\n", "                exon_seq_region_name.append(read_content[cleaned_ids[i]]['Transcript'][j][\"Exon\"][k]['seq_region_name'])\n", "                \n", "            \n", "#     for l in range(len(transcript_ids_for_exons)):\n", "        \n", "#         print('Exon \"{0}\" of Transcript ID \"{1}\" having gene ID \"{2}\" has start and end as : \"{3}\" & \"{4}\"'.format(exon_id[l],transcript_ids_for_exons[l],gene_ids_for_exons[l],exon_start[l],exon_end[l]))\n", "    "]}, {"cell_type": "code", "execution_count": 25, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 7690/7690 [00:00<00:00, 55789.79it/s]\n"]}], "source": ["get_exon_data()"]}, {"cell_type": "code", "execution_count": 26, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["43789\n", "43789\n"]}], "source": ["print(len(exon_seq_region_name))\n", "print(len(exon_strand))"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# Transcript data correction"]}, {"cell_type": "code", "execution_count": 27, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df = pd.read_csv('/all_lincRNA_data/all_lincrna_transcript_data.csv', index_col=0)\n"]}, {"cell_type": "code", "execution_count": 28, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>lincRNA</td>\n", "      <td>23181334</td>\n", "      <td>23251499</td>\n", "      <td>1329</td>\n", "      <td>2</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000400540</td>\n", "      <td>lincRNA</td>\n", "      <td>29935905</td>\n", "      <td>29950488</td>\n", "      <td>1031</td>\n", "      <td>6</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000481738</td>\n", "      <td>lincRNA</td>\n", "      <td>29937906</td>\n", "      <td>29942567</td>\n", "      <td>575</td>\n", "      <td>2</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID  Biotype  Transcript Start  \\\n", "0    1  ENSG00000115934  ENST00000538297  lincRNA          23181334   \n", "1    2  ENSG00000122043  ENST00000400540  lincRNA          29935905   \n", "2    3  ENSG00000122043  ENST00000481738  lincRNA          29937906   \n", "\n", "   Transcript End  Transcript Length  No. of Exons  \n", "0        23251499               1329             2  \n", "1        29950488               1031             6  \n", "2        29942567                575             2  "]}, "execution_count": 28, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(3)"]}, {"cell_type": "code", "execution_count": 29, "metadata": {}, "outputs": [], "source": ["df['Strand'] = transcript_strand\n", "df['Seq region Name'] = transcript_seq_region_name"]}, {"cell_type": "code", "execution_count": 30, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>lincRNA</td>\n", "      <td>23181334</td>\n", "      <td>23251499</td>\n", "      <td>1329</td>\n", "      <td>2</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000400540</td>\n", "      <td>lincRNA</td>\n", "      <td>29935905</td>\n", "      <td>29950488</td>\n", "      <td>1031</td>\n", "      <td>6</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000481738</td>\n", "      <td>lincRNA</td>\n", "      <td>29937906</td>\n", "      <td>29942567</td>\n", "      <td>575</td>\n", "      <td>2</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "    </tr>\n", "    <tr>\n", "      <th>3</th>\n", "      <td>4</td>\n", "      <td>ENSG00000122548</td>\n", "      <td>ENST00000242109</td>\n", "      <td>lincRNA</td>\n", "      <td>26533121</td>\n", "      <td>26538788</td>\n", "      <td>4283</td>\n", "      <td>2</td>\n", "      <td>-1</td>\n", "      <td>7</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID  Biotype  Transcript Start  \\\n", "0    1  ENSG00000115934  ENST00000538297  lincRNA          23181334   \n", "1    2  ENSG00000122043  ENST00000400540  lincRNA          29935905   \n", "2    3  ENSG00000122043  ENST00000481738  lincRNA          29937906   \n", "3    4  ENSG00000122548  ENST00000242109  lincRNA          26533121   \n", "\n", "   Transcript End  Transcript Length  No. of Exons  Strand Seq region Name  \n", "0        23251499               1329             2      -1              12  \n", "1        29950488               1031             6       1              13  \n", "2        29942567                575             2       1              13  \n", "3        26538788               4283             2      -1               7  "]}, "execution_count": 30, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(4)"]}, {"cell_type": "code", "execution_count": 31, "metadata": {}, "outputs": [], "source": ["df = df[['SNO', 'Gene ID', 'Transcript ID', 'Biotype', 'Strand', 'Seq region Name', 'Transcript Start', 'Transcript End', 'Transcript Length','No. of Exons']]     "]}, {"cell_type": "code", "execution_count": 32, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "      <td>23181334</td>\n", "      <td>23251499</td>\n", "      <td>1329</td>\n", "      <td>2</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000400540</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "      <td>29935905</td>\n", "      <td>29950488</td>\n", "      <td>1031</td>\n", "      <td>6</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000481738</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "      <td>29937906</td>\n", "      <td>29942567</td>\n", "      <td>575</td>\n", "      <td>2</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0    1  ENSG00000115934  ENST00000538297  lincRNA      -1              12   \n", "1    2  ENSG00000122043  ENST00000400540  lincRNA       1              13   \n", "2    3  ENSG00000122043  ENST00000481738  lincRNA       1              13   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  \n", "0          23181334        23251499               1329             2  \n", "1          29935905        29950488               1031             6  \n", "2          29937906        29942567                575             2  "]}, "execution_count": 32, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(3)"]}, {"cell_type": "code", "execution_count": 33, "metadata": {}, "outputs": [{"data": {"text/plain": ["13666"]}, "execution_count": 33, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df['SNO'])"]}, {"cell_type": "code", "execution_count": 34, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_lincRNA_data/all_lincrna_transcript_data.csv')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# Exon data correction below"]}, {"cell_type": "code", "execution_count": 35, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df = pd.read_csv('/all_lincRNA_data/all_lincrna_exon_data.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 36, "metadata": {}, "outputs": [{"data": {"text/plain": ["43789"]}, "execution_count": 36, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df)"]}, {"cell_type": "code", "execution_count": 37, "metadata": {}, "outputs": [], "source": ["df['Strand'] = exon_strand\n", "df['Seq region Name'] = exon_seq_region_name"]}, {"cell_type": "code", "execution_count": 38, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Exon ID</th>\n", "      <th>Exon Start</th>\n", "      <th>Exon End</th>\n", "      <th>Exon Length</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>ENSE00002222490</td>\n", "      <td>23251461</td>\n", "      <td>23251499</td>\n", "      <td>39</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>ENSE00002300099</td>\n", "      <td>23181334</td>\n", "      <td>23182623</td>\n", "      <td>1290</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000400540</td>\n", "      <td>ENSE00001543395</td>\n", "      <td>29935905</td>\n", "      <td>29936224</td>\n", "      <td>320</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID          Exon ID  Exon Start  \\\n", "0    1  ENSG00000115934  ENST00000538297  ENSE00002222490    23251461   \n", "1    2  ENSG00000115934  ENST00000538297  ENSE00002300099    23181334   \n", "2    3  ENSG00000122043  ENST00000400540  ENSE00001543395    29935905   \n", "\n", "   Exon End  Exon Length  Strand Seq region Name  \n", "0  23251499           39      -1              12  \n", "1  23182623         1290      -1              12  \n", "2  29936224          320       1              13  "]}, "execution_count": 38, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(3)"]}, {"cell_type": "code", "execution_count": 39, "metadata": {}, "outputs": [], "source": ["df = df[['SNO', 'Gene ID', 'Transcript ID', 'Exon ID', 'Strand','Seq region Name','Exon Start', 'Exon End', 'Exon Length']]     "]}, {"cell_type": "code", "execution_count": 40, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Exon ID</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Exon Start</th>\n", "      <th>Exon End</th>\n", "      <th>Exon Length</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>ENSE00002222490</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "      <td>23251461</td>\n", "      <td>23251499</td>\n", "      <td>39</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>ENSE00002300099</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "      <td>23181334</td>\n", "      <td>23182623</td>\n", "      <td>1290</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID          Exon ID  Strand  \\\n", "0    1  ENSG00000115934  ENST00000538297  ENSE00002222490      -1   \n", "1    2  ENSG00000115934  ENST00000538297  ENSE00002300099      -1   \n", "\n", "  Seq region Name  Exon Start  Exon End  Exon Length  \n", "0              12    23251461  23251499           39  \n", "1              12    23181334  23182623         1290  "]}, "execution_count": 40, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(2)"]}, {"cell_type": "code", "execution_count": 41, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_lincRNA_data/all_lincrna_exon_data.csv')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["# Correcting sequence data"]}, {"cell_type": "code", "execution_count": 42, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df = pd.read_csv('/all_lincRNA_data/all_lincrna_transcript_data_with_sequences.csv', index_col=0)\n"]}, {"cell_type": "code", "execution_count": 43, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>Length of Sequences</th>\n", "      <th>| Seq_len - Trans_len |</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>lincRNA</td>\n", "      <td>23181334</td>\n", "      <td>23251499</td>\n", "      <td>1329</td>\n", "      <td>2</td>\n", "      <td>1329</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>CCCGCCATGATCGTGAGGCCTCCCCAGCGATGTGGAACTGCTGGAG...</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000400540</td>\n", "      <td>lincRNA</td>\n", "      <td>29935905</td>\n", "      <td>29950488</td>\n", "      <td>1031</td>\n", "      <td>6</td>\n", "      <td>1031</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>ATAGAGGTGTTTGGGGTGTCAGCTCTGAAATGCTAAAAATAAAACC...</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000481738</td>\n", "      <td>lincRNA</td>\n", "      <td>29937906</td>\n", "      <td>29942567</td>\n", "      <td>575</td>\n", "      <td>2</td>\n", "      <td>575</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GCAGAGAACAAAGATTGGTGGCTTCCTCCTGAGCACACTGGGATGT...</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID  Biotype  Transcript Start  \\\n", "0    1  ENSG00000115934  ENST00000538297  lincRNA          23181334   \n", "1    2  ENSG00000122043  ENST00000400540  lincRNA          29935905   \n", "2    3  ENSG00000122043  ENST00000481738  lincRNA          29937906   \n", "\n", "   Transcript End  Transcript Length  No. of Exons  Length of Sequences  \\\n", "0        23251499               1329             2                 1329   \n", "1        29950488               1031             6                 1031   \n", "2        29942567                575             2                  575   \n", "\n", "   | Seq_len - Trans_len | Match ?  \\\n", "0                        0     YES   \n", "1                        0     YES   \n", "2                        0     YES   \n", "\n", "                                           Sequences  \n", "0  CCCGCCATGATCGTGAGGCCTCCCCAGCGATGTGGAACTGCTGGAG...  \n", "1  ATAGAGGTGTTTGGGGTGTCAGCTCTGAAATGCTAAAAATAAAACC...  \n", "2  GCAGAGAACAAAGATTGGTGGCTTCCTCCTGAGCACACTGGGATGT...  "]}, "execution_count": 43, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(3)"]}, {"cell_type": "code", "execution_count": 44, "metadata": {}, "outputs": [], "source": ["df['Strand'] = transcript_strand\n", "df['Seq region Name'] = transcript_seq_region_name"]}, {"cell_type": "code", "execution_count": 45, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>Length of Sequences</th>\n", "      <th>| Seq_len - Trans_len |</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>lincRNA</td>\n", "      <td>23181334</td>\n", "      <td>23251499</td>\n", "      <td>1329</td>\n", "      <td>2</td>\n", "      <td>1329</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>CCCGCCATGATCGTGAGGCCTCCCCAGCGATGTGGAACTGCTGGAG...</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000400540</td>\n", "      <td>lincRNA</td>\n", "      <td>29935905</td>\n", "      <td>29950488</td>\n", "      <td>1031</td>\n", "      <td>6</td>\n", "      <td>1031</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>ATAGAGGTGTTTGGGGTGTCAGCTCTGAAATGCTAAAAATAAAACC...</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>3</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000481738</td>\n", "      <td>lincRNA</td>\n", "      <td>29937906</td>\n", "      <td>29942567</td>\n", "      <td>575</td>\n", "      <td>2</td>\n", "      <td>575</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GCAGAGAACAAAGATTGGTGGCTTCCTCCTGAGCACACTGGGATGT...</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID  Biotype  Transcript Start  \\\n", "0    1  ENSG00000115934  ENST00000538297  lincRNA          23181334   \n", "1    2  ENSG00000122043  ENST00000400540  lincRNA          29935905   \n", "2    3  ENSG00000122043  ENST00000481738  lincRNA          29937906   \n", "\n", "   Transcript End  Transcript Length  No. of Exons  Length of Sequences  \\\n", "0        23251499               1329             2                 1329   \n", "1        29950488               1031             6                 1031   \n", "2        29942567                575             2                  575   \n", "\n", "   | Seq_len - Trans_len | Match ?  \\\n", "0                        0     YES   \n", "1                        0     YES   \n", "2                        0     YES   \n", "\n", "                                           Sequences  Strand Seq region Name  \n", "0  CCCGCCATGATCGTGAGGCCTCCCCAGCGATGTGGAACTGCTGGAG...      -1              12  \n", "1  ATAGAGGTGTTTGGGGTGTCAGCTCTGAAATGCTAAAAATAAAACC...       1              13  \n", "2  GCAGAGAACAAAGATTGGTGGCTTCCTCCTGAGCACACTGGGATGT...       1              13  "]}, "execution_count": 45, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(3)"]}, {"cell_type": "code", "execution_count": 46, "metadata": {}, "outputs": [], "source": ["df = df[['SNO', 'Gene ID', 'Transcript ID', 'Biotype', 'Strand', 'Seq region Name', 'Transcript Start', 'Transcript End', 'Transcript Length','No. of Exons', 'Length of Sequences', '| Seq_len - Trans_len |','Match ?', 'Sequences']]     "]}, {"cell_type": "code", "execution_count": 47, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>Length of Sequences</th>\n", "      <th>| Seq_len - Trans_len |</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>ENSG00000115934</td>\n", "      <td>ENST00000538297</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>12</td>\n", "      <td>23181334</td>\n", "      <td>23251499</td>\n", "      <td>1329</td>\n", "      <td>2</td>\n", "      <td>1329</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>CCCGCCATGATCGTGAGGCCTCCCCAGCGATGTGGAACTGCTGGAG...</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2</td>\n", "      <td>ENSG00000122043</td>\n", "      <td>ENST00000400540</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>13</td>\n", "      <td>29935905</td>\n", "      <td>29950488</td>\n", "      <td>1031</td>\n", "      <td>6</td>\n", "      <td>1031</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>ATAGAGGTGTTTGGGGTGTCAGCTCTGAAATGCTAAAAATAAAACC...</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0    1  ENSG00000115934  ENST00000538297  lincRNA      -1              12   \n", "1    2  ENSG00000122043  ENST00000400540  lincRNA       1              13   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  \\\n", "0          23181334        23251499               1329             2   \n", "1          29935905        29950488               1031             6   \n", "\n", "   Length of Sequences  | Seq_len - Trans_len | Match ?  \\\n", "0                 1329                        0     YES   \n", "1                 1031                        0     YES   \n", "\n", "                                           Sequences  \n", "0  CCCGCCATGATCGTGAGGCCTCCCCAGCGATGTGGAACTGCTGGAG...  \n", "1  ATAGAGGTGTTTGGGGTGTCAGCTCTGAAATGCTAAAAATAAAACC...  "]}, "execution_count": 47, "metadata": {}, "output_type": "execute_result"}], "source": ["df.head(2)"]}, {"cell_type": "code", "execution_count": 48, "metadata": {}, "outputs": [], "source": ["df.to_csv('/all_lincRNA_data/all_lincrna_transcript_data_with_sequences.csv')"]}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}