{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 165, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_ens_transcript_data = pd.read_csv('/dataset/Ensembl_numeric/all_lincRNA_data/all_lincRNA_sorted_chr/all_lincrna_transcript_data_with_sequences_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 166, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>Length of Sequences</th>\n", "      <th>| Seq_len - Trans_len |</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>4916</td>\n", "      <td>ENSG00000236423</td>\n", "      <td>ENST00000413332</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>3900372</td>\n", "      <td>3905532</td>\n", "      <td>634</td>\n", "      <td>3</td>\n", "      <td>634</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>11412</td>\n", "      <td>ENSG00000270035</td>\n", "      <td>ENST00000602973</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>7698303</td>\n", "      <td>7698872</td>\n", "      <td>570</td>\n", "      <td>1</td>\n", "      <td>570</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>11417</td>\n", "      <td>ENSG00000270066</td>\n", "      <td>ENST00000602755</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>109100193</td>\n", "      <td>109100619</td>\n", "      <td>427</td>\n", "      <td>1</td>\n", "      <td>427</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0   4916  ENSG00000236423  ENST00000413332  lincRNA       1               1   \n", "1  11412  ENSG00000270035  ENST00000602973  lincRNA      -1               1   \n", "2  11417  ENSG00000270066  ENST00000602755  lincRNA       1               1   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  \\\n", "0           3900372         3905532                634             3   \n", "1           7698303         7698872                570             1   \n", "2         109100193       109100619                427             1   \n", "\n", "   Length of Sequences  | Seq_len - Trans_len | Match ?  \\\n", "0                  634                        0     YES   \n", "1                  570                        0     YES   \n", "2                  427                        0     YES   \n", "\n", "                                           Sequences  \n", "0  GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...  \n", "1  GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...  \n", "2  GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...  "]}, "execution_count": 166, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.head(3)"]}, {"cell_type": "code", "execution_count": 167, "metadata": {}, "outputs": [], "source": ["ens_all_seq = df_ens_transcript_data['Sequences'].tolist()"]}, {"cell_type": "code", "execution_count": 168, "metadata": {}, "outputs": [], "source": ["# ens_all_seq[0]"]}, {"cell_type": "code", "execution_count": 169, "metadata": {}, "outputs": [{"data": {"text/plain": ["13666"]}, "execution_count": 169, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df_ens_transcript_data)"]}, {"cell_type": "code", "execution_count": 170, "metadata": {}, "outputs": [], "source": ["from Bio import Seq\n", "import regex as re\n", "from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 171, "metadata": {}, "outputs": [], "source": ["no_of_amino_acids = []\n", "len_longest_orf = []\n", "longest_orf_seq = []\n", "start_seq_index = []\n", "protein_seq_encoded = []"]}, {"cell_type": "code", "execution_count": 172, "metadata": {}, "outputs": [], "source": ["# no_of_amino_acids"]}, {"cell_type": "code", "execution_count": 173, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 5 \u00b5s, sys: 1e+03 ns, total: 6 \u00b5s\n", "Wall time: 10 \u00b5s\n"]}], "source": ["%%time\n", "def get_all_orf_info(input_seq):\n", "    startP = re.compile('ATG')\n", "    nuc = input_seq.replace('\\n','')\n", "    longest = (0,)\n", "    for m in startP.finditer(nuc, overlapped=True):\n", "        if len(Seq.Seq(nuc)[m.start():].translate(to_stop=True)) > longest[0]:\n", "            pro = Seq.Seq(nuc)[m.start():].translate(to_stop=True)\n", "            longest = (len(pro), \n", "                       m.start(), \n", "                       str(pro),\n", "                       nuc[m.start():m.start()+len(pro)*3+3])\n", "            return(longest)"]}, {"cell_type": "code", "execution_count": 174, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["  0%|          | 0/13666 [00:00<?, ?it/s]/site-packages/Bio/Seq.py:2609: BiopythonWarning: Partial codon, len(sequence) not a multiple of three. Explicitly trim the sequence or add trailing N before translation. This may become an error in future.\n", "  BiopythonWarning)\n", "100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 13666/13666 [00:05<00:00, 2394.18it/s]\n"]}], "source": ["for i in tqdm(range(len(ens_all_seq))):\n", "    longest = get_all_orf_info(ens_all_seq[i])\n", "    \n", "    if longest == None or longest[0] == 0:\n", "        no_of_amino_acids.append('No start codon found')\n", "        len_longest_orf.append('No start codon found')\n", "        start_seq_index.append('No start codon found')\n", "        protein_seq_encoded.append('No start codon found')\n", "        longest_orf_seq.append('No start codon found')\n", "        continue\n", "        \n", "    else:\n", "        \n", "        no_of_amino_acids.append(longest[0])\n", "        len_longest_orf.append((len(longest[3])-3))\n", "        start_seq_index.append(longest[1])\n", "        protein_seq_encoded.append(longest[2])\n", "        longest_orf_seq.append(longest[3])"]}, {"cell_type": "code", "execution_count": 175, "metadata": {}, "outputs": [], "source": ["stop_codon = []"]}, {"cell_type": "code", "execution_count": 176, "metadata": {}, "outputs": [{"data": {"text/plain": ["'TAG'"]}, "execution_count": 176, "metadata": {}, "output_type": "execute_result"}], "source": ["longest_orf_seq[0][-3:]"]}, {"cell_type": "code", "execution_count": 177, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 13666/13666 [00:00<00:00, 646420.05it/s]\n"]}], "source": ["for i in tqdm(range(len(longest_orf_seq))):\n", "    if longest_orf_seq[i] == 'No start codon found':\n", "        stop_codon.append('No start codon found')\n", "        continue\n", "        \n", "    else:\n", "        if longest_orf_seq[i][-3:] == 'TAA' or longest_orf_seq[i][-3:] == 'TAG' or longest_orf_seq[i][-3:] == 'TGA':\n", "            stop_codon.append('YES')\n", "        else:\n", "            stop_codon.append('NO')"]}, {"cell_type": "code", "execution_count": 178, "metadata": {}, "outputs": [{"data": {"text/plain": ["13666"]}, "execution_count": 178, "metadata": {}, "output_type": "execute_result"}], "source": ["len(stop_codon)"]}, {"cell_type": "code", "execution_count": 179, "metadata": {}, "outputs": [], "source": ["c = 0\n", "index = []\n", "for i in range(len(stop_codon)):\n", "    if stop_codon[i] == 'NO':\n", "        c = c + 1\n", "        index.append(i)\n", "        "]}, {"cell_type": "code", "execution_count": 180, "metadata": {}, "outputs": [{"data": {"text/plain": ["123"]}, "execution_count": 180, "metadata": {}, "output_type": "execute_result"}], "source": ["c"]}, {"cell_type": "code", "execution_count": 181, "metadata": {}, "outputs": [], "source": ["orf_multiple_of_3 = []"]}, {"cell_type": "code", "execution_count": 182, "metadata": {}, "outputs": [], "source": ["# longest_orf_seq"]}, {"cell_type": "code", "execution_count": 183, "metadata": {}, "outputs": [], "source": ["c = 0\n", "index = []\n", "for i in range(len(stop_codon)):\n", "    if len_longest_orf[i] == 'No start codon found':\n", "        orf_multiple_of_3.append('No start codon found')\n", "        continue\n", "        \n", "    else:\n", "        if len_longest_orf[i] % 3 == 0:\n", "            orf_multiple_of_3.append('YES')\n", "        else:\n", "            orf_multiple_of_3.append('NO')\n", "            c = c + 1\n", "            index.append(i)\n", "        "]}, {"cell_type": "code", "execution_count": 184, "metadata": {}, "outputs": [{"data": {"text/plain": ["94"]}, "execution_count": 184, "metadata": {}, "output_type": "execute_result"}], "source": ["c"]}, {"cell_type": "code", "execution_count": 185, "metadata": {}, "outputs": [], "source": ["# no_of_amino_acids = []\n", "# len_longest_orf = []\n", "# longest_orf_seq = []\n", "# start_seq_index = []\n", "# protein_seq_encoded = []\n", "# orf_multiple_of_3 = []\n", "# stop_codon = []"]}, {"cell_type": "code", "execution_count": 186, "metadata": {}, "outputs": [], "source": ["# orf_multiple_of_3"]}, {"cell_type": "code", "execution_count": 187, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data['Len of longest ORF'] = len_longest_orf\n", "df_ens_transcript_data['No. of Amino Acids'] = no_of_amino_acids\n", "df_ens_transcript_data['Longest ORF Seq'] = longest_orf_seq\n", "df_ens_transcript_data['ORF Start Index for longest Seq'] = start_seq_index\n", "df_ens_transcript_data['Protein Seq encoded'] = protein_seq_encoded\n", "df_ens_transcript_data['Stop codon present ?'] = stop_codon\n", "df_ens_transcript_data['ORF multiple of 3 ?'] = orf_multiple_of_3"]}, {"cell_type": "code", "execution_count": 188, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>...</th>\n", "      <th>| Seq_len - Trans_len |</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "      <th>Len of longest ORF</th>\n", "      <th>No. of Amino Acids</th>\n", "      <th>Longest ORF Seq</th>\n", "      <th>ORF Start Index for longest Seq</th>\n", "      <th>Protein Seq encoded</th>\n", "      <th>Stop codon present ?</th>\n", "      <th>ORF multiple of 3 ?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>4916</td>\n", "      <td>ENSG00000236423</td>\n", "      <td>ENST00000413332</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>3900372</td>\n", "      <td>3905532</td>\n", "      <td>634</td>\n", "      <td>3</td>\n", "      <td>...</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...</td>\n", "      <td>33</td>\n", "      <td>11</td>\n", "      <td>ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG</td>\n", "      <td>65</td>\n", "      <td>MFMRARPLAPL</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>11412</td>\n", "      <td>ENSG00000270035</td>\n", "      <td>ENST00000602973</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>7698303</td>\n", "      <td>7698872</td>\n", "      <td>570</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...</td>\n", "      <td>42</td>\n", "      <td>14</td>\n", "      <td>ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA</td>\n", "      <td>27</td>\n", "      <td>MVMPRAEDVRRQAV</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>11417</td>\n", "      <td>ENSG00000270066</td>\n", "      <td>ENST00000602755</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>109100193</td>\n", "      <td>109100619</td>\n", "      <td>427</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>0</td>\n", "      <td>YES</td>\n", "      <td>GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...</td>\n", "      <td>138</td>\n", "      <td>46</td>\n", "      <td>ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...</td>\n", "      <td>227</td>\n", "      <td>MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>3 rows \u00d7 21 columns</p>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0   4916  ENSG00000236423  ENST00000413332  lincRNA       1               1   \n", "1  11412  ENSG00000270035  ENST00000602973  lincRNA      -1               1   \n", "2  11417  ENSG00000270066  ENST00000602755  lincRNA       1               1   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  ...  \\\n", "0           3900372         3905532                634             3  ...   \n", "1           7698303         7698872                570             1  ...   \n", "2         109100193       109100619                427             1  ...   \n", "\n", "   | Seq_len - Trans_len |  Match ?  \\\n", "0                        0      YES   \n", "1                        0      YES   \n", "2                        0      YES   \n", "\n", "                                           Sequences Len of longest ORF  \\\n", "0  GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...                 33   \n", "1  GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...                 42   \n", "2  GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...                138   \n", "\n", "  No. of Amino Acids                                    Longest ORF Seq  \\\n", "0                 11               ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG   \n", "1                 14      ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA   \n", "2                 46  ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...   \n", "\n", "  ORF Start Index for longest Seq  \\\n", "0                              65   \n", "1                              27   \n", "2                             227   \n", "\n", "                              Protein Seq encoded Stop codon present ?  \\\n", "0                                     MFMRARPLAPL                  YES   \n", "1                                  MVMPRAEDVRRQAV                  YES   \n", "2  MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR                  YES   \n", "\n", "  ORF multiple of 3 ?  \n", "0                 YES  \n", "1                 YES  \n", "2                 YES  \n", "\n", "[3 rows x 21 columns]"]}, "execution_count": 188, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.head(3)"]}, {"cell_type": "code", "execution_count": 189, "metadata": {}, "outputs": [{"data": {"text/plain": ["SNO                                                                             3803\n", "Gene ID                                                              ENSG00000232453\n", "Transcript ID                                                        ENST00000427292\n", "Biotype                                                                      lincRNA\n", "Strand                                                                            -1\n", "Seq region Name                                                                    1\n", "Transcript Start                                                            58895845\n", "Transcript End                                                              58896703\n", "Transcript Length                                                                219\n", "No. of Exons                                                                       2\n", "Length of Sequences                                                              219\n", "| Seq_len - Trans_len |                                                            0\n", "Match ?                                                                          YES\n", "Sequences                          AAAACTGTAGCCCAGGGCATTGAAAACCTCTCGGAGAGGGCCAGGT...\n", "Len of longest ORF                                                               113\n", "No. of Amino Acids                                                                38\n", "Longest ORF Seq                    ATGCCTTTCTCTGAGTCACAGTCTGTATCTTCTAGAAAAGGCTGTC...\n", "ORF Start Index for longest Seq                                                  103\n", "Protein Seq encoded                           MPFSESQSVSSRKGCPSSPQAHWLLRYDIQTQCSWTWP\n", "Stop codon present ?                                                              NO\n", "ORF multiple of 3 ?                                                               NO\n", "Name: 304, dtype: object"]}, "execution_count": 189, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.iloc[304]"]}, {"cell_type": "code", "execution_count": 190, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data.to_csv('/feature_selection/ORF/Ensembl/all_lincrna_transcript_data_orf_info.csv')"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_ens_transcript_data = pd.read_csv('/1.lncRNA/feature_selection/ORF/Ensembl/all_lincrna_transcript_data_orf_info.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>...</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "      <th>GC %</th>\n", "      <th>Len of longest ORF</th>\n", "      <th>No. of Amino Acids</th>\n", "      <th>Longest ORF Seq</th>\n", "      <th>ORF Start Index for longest Seq</th>\n", "      <th>Protein Seq encoded</th>\n", "      <th>Stop codon present ?</th>\n", "      <th>ORF multiple of 3 ?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>4916</td>\n", "      <td>ENSG00000236423</td>\n", "      <td>ENST00000413332</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>3900372</td>\n", "      <td>3905532</td>\n", "      <td>634</td>\n", "      <td>3</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...</td>\n", "      <td>59.31</td>\n", "      <td>33</td>\n", "      <td>11</td>\n", "      <td>ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG</td>\n", "      <td>65</td>\n", "      <td>MFMRARPLAPL</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>11412</td>\n", "      <td>ENSG00000270035</td>\n", "      <td>ENST00000602973</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>7698303</td>\n", "      <td>7698872</td>\n", "      <td>570</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...</td>\n", "      <td>55.26</td>\n", "      <td>42</td>\n", "      <td>14</td>\n", "      <td>ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA</td>\n", "      <td>27</td>\n", "      <td>MVMPRAEDVRRQAV</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>11417</td>\n", "      <td>ENSG00000270066</td>\n", "      <td>ENST00000602755</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>109100193</td>\n", "      <td>109100619</td>\n", "      <td>427</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...</td>\n", "      <td>65.81</td>\n", "      <td>138</td>\n", "      <td>46</td>\n", "      <td>ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...</td>\n", "      <td>227</td>\n", "      <td>MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>3 rows \u00d7 22 columns</p>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0   4916  ENSG00000236423  ENST00000413332  lincRNA       1               1   \n", "1  11412  ENSG00000270035  ENST00000602973  lincRNA      -1               1   \n", "2  11417  ENSG00000270066  ENST00000602755  lincRNA       1               1   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  ...  \\\n", "0           3900372         3905532                634             3  ...   \n", "1           7698303         7698872                570             1  ...   \n", "2         109100193       109100619                427             1  ...   \n", "\n", "   Match ?                                          Sequences   GC %  \\\n", "0      YES  GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...  59.31   \n", "1      YES  GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...  55.26   \n", "2      YES  GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...  65.81   \n", "\n", "  Len of longest ORF  No. of Amino Acids  \\\n", "0                 33                  11   \n", "1                 42                  14   \n", "2                138                  46   \n", "\n", "                                     Longest ORF Seq  \\\n", "0               ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG   \n", "1      ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA   \n", "2  ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...   \n", "\n", "  ORF Start Index for longest Seq  \\\n", "0                              65   \n", "1                              27   \n", "2                             227   \n", "\n", "                              Protein Seq encoded Stop codon present ?  \\\n", "0                                     MFMRARPLAPL                  YES   \n", "1                                  MVMPRAEDVRRQAV                  YES   \n", "2  MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR                  YES   \n", "\n", "  ORF multiple of 3 ?  \n", "0                 YES  \n", "1                 YES  \n", "2                 YES  \n", "\n", "[3 rows x 22 columns]"]}, "execution_count": 2, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.head(3)"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 13666/13666 [00:00<00:00, 29221.83it/s]\n"]}], "source": ["c = 0\n", "index = []\n", "for i in tqdm(range(len(df_ens_transcript_data))):\n", "    if df_ens_transcript_data['Stop codon present ?'][i] == 'YES' and df_ens_transcript_data['ORF multiple of 3 ?'][i] == 'NO':\n", "        c = c + 1\n", "        index.append(i)"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [{"data": {"text/plain": ["7"]}, "execution_count": 5, "metadata": {}, "output_type": "execute_result"}], "source": ["c"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [{"data": {"text/plain": ["[776, 1856, 4745, 7076, 8135, 10276, 11922]"]}, "execution_count": 6, "metadata": {}, "output_type": "execute_result"}], "source": ["index"]}, {"cell_type": "code", "execution_count": 7, "metadata": {}, "outputs": [{"data": {"text/plain": ["SNO                                                                             1384\n", "Gene ID                                                              ENSG00000224687\n", "Transcript ID                                                        ENST00000452867\n", "Biotype                                                                      lincRNA\n", "Strand                                                                            -1\n", "Seq region Name                                                                    1\n", "Transcript Start                                                           178093150\n", "Transcript End                                                             178093572\n", "Transcript Length                                                                300\n", "No. of Exons                                                                       2\n", "Length of Sequences                                                              300\n", "| Seq_len - Trans_len |                                                            0\n", "Match ?                                                                          YES\n", "Sequences                          CTGCTGCTCAAAGTAAAGACCGTTTGAGAAGGCGGGGAGATCGAAG...\n", "Len of longest ORF                                                               166\n", "No. of Amino Acids                                                                56\n", "Longest ORF Seq                    ATGCACAGGGTCTGCGCCCTGCGCCCTGTCCGCGCTGAAGCTGAAC...\n", "ORF Start Index for longest Seq                                                  131\n", "Protein Seq encoded                MHRVCALRPVRAEAELVIGLQQHGEEVWCGWTDGPLRHMKYSKPSI...\n", "Stop codon present ?                                                             YES\n", "ORF multiple of 3 ?                                                               NO\n", "Name: 776, dtype: object"]}, "execution_count": 7, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.iloc[776]"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["from Bio.SeqUtils import GC123\n", "from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 11, "metadata": {}, "outputs": [], "source": ["gc_percentage = []\n", "nan_values = 0\n", "nan_index = []"]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 3 \u00b5s, sys: 1 \u00b5s, total: 4 \u00b5s\n", "Wall time: 5.72 \u00b5s\n"]}], "source": ["%%time\n", "def get_gc_percentage():\n", "    for i in tqdm(range(len(df_ens_transcript_data['Sequences']))):\n", "        \n", "        if type(df_ens_transcript_data['Sequences'][i]) is float:\n", "            gc_percentage.append('Nan')\n", "            df_ens_transcript_data['Sequences'][i] = 'Nan'\n", "            nan_values = nan_values + 1\n", "            nan_index.append(i)\n", "            \n", "        else: \n", "            gc_percentage.append(GC123(df_ens_transcript_data['Sequences'][i])[0])"]}, {"cell_type": "code", "execution_count": 13, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 13666/13666 [00:31<00:00, 429.19it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["CPU times: user 24.9 s, sys: 632 ms, total: 25.5 s\n", "Wall time: 31.8 s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "get_gc_percentage()"]}, {"cell_type": "code", "execution_count": 14, "metadata": {}, "outputs": [{"data": {"text/plain": ["0"]}, "execution_count": 14, "metadata": {}, "output_type": "execute_result"}], "source": ["nan_values"]}, {"cell_type": "code", "execution_count": 25, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 13666/13666 [00:00<00:00, 480500.28it/s]\n"]}], "source": ["for i in tqdm(range(len(gc_percentage))):\n", "    gc_percentage[i] = float(\"{0:.2f}\".format(gc_percentage[i]))"]}, {"cell_type": "code", "execution_count": 27, "metadata": {}, "outputs": [{"data": {"text/plain": ["[59.31, 55.26, 65.81, 51.52, 50.85]"]}, "execution_count": 27, "metadata": {}, "output_type": "execute_result"}], "source": ["gc_percentage[0:5]"]}, {"cell_type": "code", "execution_count": 29, "metadata": {}, "outputs": [], "source": ["a = list(df_ens_transcript_data.columns.values)"]}, {"cell_type": "code", "execution_count": 31, "metadata": {}, "outputs": [], "source": ["b = a[0:14]"]}, {"cell_type": "code", "execution_count": 33, "metadata": {}, "outputs": [], "source": ["b.append('GC %')"]}, {"cell_type": "code", "execution_count": 35, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data['GC %'] = gc_percentage"]}, {"cell_type": "code", "execution_count": 36, "metadata": {}, "outputs": [], "source": ["c = b + a[14:]"]}, {"cell_type": "code", "execution_count": 40, "metadata": {}, "outputs": [], "source": ["# c"]}, {"cell_type": "code", "execution_count": 38, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data = df_ens_transcript_data[c]"]}, {"cell_type": "code", "execution_count": 39, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>...</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "      <th>GC %</th>\n", "      <th>Len of longest ORF</th>\n", "      <th>No. of Amino Acids</th>\n", "      <th>Longest ORF Seq</th>\n", "      <th>ORF Start Index for longest Seq</th>\n", "      <th>Protein Seq encoded</th>\n", "      <th>Stop codon present ?</th>\n", "      <th>ORF multiple of 3 ?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>4916</td>\n", "      <td>ENSG00000236423</td>\n", "      <td>ENST00000413332</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>3900372</td>\n", "      <td>3905532</td>\n", "      <td>634</td>\n", "      <td>3</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...</td>\n", "      <td>59.31</td>\n", "      <td>33</td>\n", "      <td>11</td>\n", "      <td>ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG</td>\n", "      <td>65</td>\n", "      <td>MFMRARPLAPL</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>11412</td>\n", "      <td>ENSG00000270035</td>\n", "      <td>ENST00000602973</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>7698303</td>\n", "      <td>7698872</td>\n", "      <td>570</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...</td>\n", "      <td>55.26</td>\n", "      <td>42</td>\n", "      <td>14</td>\n", "      <td>ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA</td>\n", "      <td>27</td>\n", "      <td>MVMPRAEDVRRQAV</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>11417</td>\n", "      <td>ENSG00000270066</td>\n", "      <td>ENST00000602755</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>109100193</td>\n", "      <td>109100619</td>\n", "      <td>427</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...</td>\n", "      <td>65.81</td>\n", "      <td>138</td>\n", "      <td>46</td>\n", "      <td>ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...</td>\n", "      <td>227</td>\n", "      <td>MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>3 rows \u00d7 22 columns</p>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0   4916  ENSG00000236423  ENST00000413332  lincRNA       1               1   \n", "1  11412  ENSG00000270035  ENST00000602973  lincRNA      -1               1   \n", "2  11417  ENSG00000270066  ENST00000602755  lincRNA       1               1   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  ...  \\\n", "0           3900372         3905532                634             3  ...   \n", "1           7698303         7698872                570             1  ...   \n", "2         109100193       109100619                427             1  ...   \n", "\n", "   Match ?                                          Sequences   GC %  \\\n", "0      YES  GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...  59.31   \n", "1      YES  GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...  55.26   \n", "2      YES  GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...  65.81   \n", "\n", "  Len of longest ORF  No. of Amino Acids  \\\n", "0                 33                  11   \n", "1                 42                  14   \n", "2                138                  46   \n", "\n", "                                     Longest ORF Seq  \\\n", "0               ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG   \n", "1      ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA   \n", "2  ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...   \n", "\n", "  ORF Start Index for longest Seq  \\\n", "0                              65   \n", "1                              27   \n", "2                             227   \n", "\n", "                              Protein Seq encoded Stop codon present ?  \\\n", "0                                     MFMRARPLAPL                  YES   \n", "1                                  MVMPRAEDVRRQAV                  YES   \n", "2  MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR                  YES   \n", "\n", "  ORF multiple of 3 ?  \n", "0                 YES  \n", "1                 YES  \n", "2                 YES  \n", "\n", "[3 rows x 22 columns]"]}, "execution_count": 39, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.head(3)"]}, {"cell_type": "code", "execution_count": 41, "metadata": {}, "outputs": [{"data": {"text/plain": ["SNO                                                                             4916\n", "Gene ID                                                              ENSG00000236423\n", "Transcript ID                                                        ENST00000413332\n", "Biotype                                                                      lincRNA\n", "Strand                                                                             1\n", "Seq region Name                                                                    1\n", "Transcript Start                                                             3900372\n", "Transcript End                                                               3905532\n", "Transcript Length                                                                634\n", "No. of Exons                                                                       3\n", "Length of Sequences                                                              634\n", "| Seq_len - Trans_len |                                                            0\n", "Match ?                                                                          YES\n", "Sequences                          GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...\n", "GC %                                                                           59.31\n", "Len of longest ORF                                                                33\n", "No. of Amino Acids                                                                11\n", "Longest ORF Seq                                 ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG\n", "ORF Start Index for longest Seq                                                   65\n", "Protein Seq encoded                                                      MFMRARPLAPL\n", "Stop codon present ?                                                             YES\n", "ORF multiple of 3 ?                                                              YES\n", "Name: 0, dtype: object"]}, "execution_count": 41, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.iloc[0]"]}, {"cell_type": "code", "execution_count": 42, "metadata": {}, "outputs": [{"data": {"text/plain": ["22"]}, "execution_count": 42, "metadata": {}, "output_type": "execute_result"}], "source": ["len(c)"]}, {"cell_type": "code", "execution_count": 43, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data.to_csv('/1.lncRNA/feature_selection/ORF/Ensembl/all_lincrna_transcript_data_orf_info.csv')"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}