{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 31, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_overlap = pd.read_csv('/feature_selection/ORF/Ens-Ref/ens_ref_ORF&overlap_cleaned(ens).csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 32, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ENS lincRNA gene ID</th>\n", "      <th>Ref lncRNA gene ID</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>ENS-Ref Overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>Overlap Length</th>\n", "      <th>No. ENS lincrna Exons which overlap RefSeq exons</th>\n", "      <th>Max ORF length(ENS)</th>\n", "      <th>Max No. of amino acids(ENS)</th>\n", "      <th>Max Transcript Index for ORF(ENS)</th>\n", "      <th>GC % of trans row having Max ORF(ENS)</th>\n", "      <th>Max ORF Sequence(ENS)</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000236648</td>\n", "      <td>gene-LOC107984028</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>4</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>17270</td>\n", "      <td>3</td>\n", "      <td>90</td>\n", "      <td>30</td>\n", "      <td>153</td>\n", "      <td>51.97</td>\n", "      <td>ATGGCTGCAAATGGCTCCTCCGTGCAGGTGAGGACATGGAAGCACA...</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>gene-LOC101928628</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>1720</td>\n", "      <td>2</td>\n", "      <td>36</td>\n", "      <td>12</td>\n", "      <td>818</td>\n", "      <td>40.92</td>\n", "      <td>ATGTATTCCTCTCTCTCTCTCACCTATTCCCATGTTTAA</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["  ENS lincRNA gene ID Ref lncRNA gene ID Chr region  \\\n", "0     ENSG00000236648  gene-LOC107984028          1   \n", "1     ENSG00000230704  gene-LOC101928628          1   \n", "\n", "   No. of transcripts in lincrna  No. of transcripts in lncrna  \\\n", "0                              1                             4   \n", "1                              1                             1   \n", "\n", "  ENS-Ref Overlap ?  same_opp_overlap Overlap Length  \\\n", "0               Yes                 1          17270   \n", "1               Yes                 1           1720   \n", "\n", "   No. ENS lincrna Exons which overlap RefSeq exons  Max ORF length(ENS)  \\\n", "0                                                 3                   90   \n", "1                                                 2                   36   \n", "\n", "   Max No. of amino acids(ENS)  Max Transcript Index for ORF(ENS)  \\\n", "0                           30                                153   \n", "1                           12                                818   \n", "\n", "   GC % of trans row having Max ORF(ENS)  \\\n", "0                                  51.97   \n", "1                                  40.92   \n", "\n", "                               Max ORF Sequence(ENS)  \n", "0  ATGGCTGCAAATGGCTCCTCCGTGCAGGTGAGGACATGGAAGCACA...  \n", "1            ATGTATTCCTCTCTCTCTCTCACCTATTCCCATGTTTAA  "]}, "execution_count": 32, "metadata": {}, "output_type": "execute_result"}], "source": ["df_overlap.head(2)"]}, {"cell_type": "code", "execution_count": 33, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [00:00<00:00, 19075.18it/s]\n"]}], "source": ["from tqdm import tqdm\n", "l_id = []\n", "r_id = []\n", "for i in tqdm(range(len(df_overlap['ENS lincRNA gene ID']))):\n", "    l_id.append(df_overlap['ENS lincRNA gene ID'][i])\n", "    r_id.append(df_overlap['Ref lncRNA gene ID'][i])"]}, {"cell_type": "code", "execution_count": 34, "metadata": {}, "outputs": [], "source": ["set_lincrna_id = set(l_id)\n", "set_ref_id = set(r_id)"]}, {"cell_type": "code", "execution_count": 35, "metadata": {}, "outputs": [], "source": ["df_ref_transcript_data = pd.read_csv('/feature_selection/ORF/RefSeq/all_lncrna_transcript_data_orf_info.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 36, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 25715/25715 [00:01<00:00, 18699.81it/s]\n"]}], "source": ["drop_index = []\n", "drop_no = 0\n", "for i in tqdm(range(len(df_ref_transcript_data))):\n", "    if df_ref_transcript_data['Len of longest ORF'][i] == 'No start codon found' or df_ref_transcript_data['Len of longest ORF'][i] == 'Nan':\n", "        drop_index.append(i)\n", "        drop_no = drop_no + 1"]}, {"cell_type": "code", "execution_count": 37, "metadata": {}, "outputs": [], "source": ["df_ref_transcript_data.drop(drop_index, axis=0,inplace=True)\n", "df_ref_transcript_data.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 38, "metadata": {}, "outputs": [{"data": {"text/plain": ["25689"]}, "execution_count": 38, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df_ref_transcript_data)"]}, {"cell_type": "code", "execution_count": 39, "metadata": {}, "outputs": [], "source": ["df_ref_transcript_data[\"Len of longest ORF\"] = pd.to_numeric(df_ref_transcript_data[\"Len of longest ORF\"])"]}, {"cell_type": "code", "execution_count": 40, "metadata": {}, "outputs": [], "source": ["df_ref_transcript_data[\"GC %\"] = pd.to_numeric(df_ref_transcript_data[\"GC %\"])"]}, {"cell_type": "code", "execution_count": 41, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Index</th>\n", "      <th>RefSeq Gene ID</th>\n", "      <th>RefSeq Transcript ID</th>\n", "      <th>Dbxref</th>\n", "      <th>Parent</th>\n", "      <th>gbkey</th>\n", "      <th>Type</th>\n", "      <th>Seq_id</th>\n", "      <th>Strand</th>\n", "      <th>...</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "      <th>GC %</th>\n", "      <th>Len of longest ORF</th>\n", "      <th>No. of Amino Acids</th>\n", "      <th>Longest ORF Seq</th>\n", "      <th>ORF Start Index for longest Seq</th>\n", "      <th>Protein Seq encoded</th>\n", "      <th>Stop codon present ?</th>\n", "      <th>ORF multiple of 3 ?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>27</td>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>rna-XR_001737835.1</td>\n", "      <td>GeneID:107985730,Genbank:XR_001737835.1,HGNC:H...</td>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>ncRNA</td>\n", "      <td>lnc_RNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>TCGTATGAGTCCAGTTTTGTGAAGTGCCTAGAATAGTCAAAATTCA...</td>\n", "      <td>51.30</td>\n", "      <td>15</td>\n", "      <td>5</td>\n", "      <td>ATGAGTCCAGTTTTGTGA</td>\n", "      <td>4</td>\n", "      <td>MSPVL</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>1508</td>\n", "      <td>251564</td>\n", "      <td>gene-LOC107985451</td>\n", "      <td>rna-XR_001737807.1</td>\n", "      <td>GeneID:107985451,Genbank:XR_001737807.1</td>\n", "      <td>gene-LOC107985451</td>\n", "      <td>ncRNA</td>\n", "      <td>lnc_RNA</td>\n", "      <td>1</td>\n", "      <td>-1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GCCATTAAGGGATGAGCGTAAGCAGTGACTGACAGGCAGGCTCTGC...</td>\n", "      <td>51.74</td>\n", "      <td>18</td>\n", "      <td>6</td>\n", "      <td>ATGAGCGTAAGCAGTGACTGA</td>\n", "      <td>11</td>\n", "      <td>MSVSSD</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>1507</td>\n", "      <td>251556</td>\n", "      <td>gene-LOC100505795</td>\n", "      <td>rna-NR_038072.1</td>\n", "      <td>GeneID:100505795,Genbank:NR_038072.1</td>\n", "      <td>gene-LOC100505795</td>\n", "      <td>ncRNA</td>\n", "      <td>lnc_RNA</td>\n", "      <td>1</td>\n", "      <td>-1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>CTTATTAAAAATAGAATATTGCCTACATCACATAAAGGGCAAACTC...</td>\n", "      <td>44.27</td>\n", "      <td>123</td>\n", "      <td>41</td>\n", "      <td>ATGGGCACACTTGGTGATGTGCAACTCAGTGGGGCCTACAGGGTCC...</td>\n", "      <td>118</td>\n", "      <td>MGTLGDVQLSGAYRVPADIDITKEFEPGLRNRVSGCFKKAV</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>3 rows \u00d7 26 columns</p>\n", "</div>"], "text/plain": ["    SNO   Index     RefSeq Gene ID RefSeq Transcript ID  \\\n", "0     1      27   gene-MIR1302-2HG   rna-XR_001737835.1   \n", "1  1508  251564  gene-LOC107985451   rna-XR_001737807.1   \n", "2  1507  251556  gene-LOC100505795      rna-NR_038072.1   \n", "\n", "                                              Dbxref             Parent  \\\n", "0  GeneID:107985730,Genbank:XR_001737835.1,HGNC:H...   gene-MIR1302-2HG   \n", "1            GeneID:107985451,Genbank:XR_001737807.1  gene-LOC107985451   \n", "2               GeneID:100505795,Genbank:NR_038072.1  gene-LOC100505795   \n", "\n", "   gbkey     Type Seq_id  Strand  ...  Match ?  \\\n", "0  ncRNA  lnc_RNA      1       1  ...      YES   \n", "1  ncRNA  lnc_RNA      1      -1  ...      YES   \n", "2  ncRNA  lnc_RNA      1      -1  ...      YES   \n", "\n", "                                           Sequences   GC %  \\\n", "0  TCGTATGAGTCCAGTTTTGTGAAGTGCCTAGAATAGTCAAAATTCA...  51.30   \n", "1  GCCATTAAGGGATGAGCGTAAGCAGTGACTGACAGGCAGGCTCTGC...  51.74   \n", "2  CTTATTAAAAATAGAATATTGCCTACATCACATAAAGGGCAAACTC...  44.27   \n", "\n", "   Len of longest ORF  No. of Amino Acids  \\\n", "0                  15                   5   \n", "1                  18                   6   \n", "2                 123                  41   \n", "\n", "                                     Longest ORF Seq  \\\n", "0                                 ATGAGTCCAGTTTTGTGA   \n", "1                              ATGAGCGTAAGCAGTGACTGA   \n", "2  ATGGGCACACTTGGTGATGTGCAACTCAGTGGGGCCTACAGGGTCC...   \n", "\n", "  ORF Start Index for longest Seq                        Protein Seq encoded  \\\n", "0                               4                                      MSPVL   \n", "1                              11                                     MSVSSD   \n", "2                             118  MGTLGDVQLSGAYRVPADIDITKEFEPGLRNRVSGCFKKAV   \n", "\n", "   Stop codon present ?  ORF multiple of 3 ?  \n", "0                   YES                  YES  \n", "1                   YES                  YES  \n", "2                   YES                  YES  \n", "\n", "[3 rows x 26 columns]"]}, "execution_count": 41, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_transcript_data.head(3)"]}, {"cell_type": "code", "execution_count": 42, "metadata": {}, "outputs": [], "source": ["cleaned_max_orf = []\n", "cleaned_amino_acid = []\n", "cleaned_max_index = []\n", "cleaned_gc_percentage = []\n", "cleaned_max_orf_seq = []"]}, {"cell_type": "code", "execution_count": 43, "metadata": {}, "outputs": [], "source": ["def get_gene_orf():\n", "    for i in tqdm(range(len(df_overlap))):\n", "        a = df_ref_transcript_data.index[df_ref_transcript_data['RefSeq Gene ID'] == df_overlap['Ref lncRNA gene ID'][i]]\n", "        max_orf = 0\n", "        max_gc = 0\n", "        index_max_all = 0\n", "        for i in range(len(a)):\n", "            if df_ref_transcript_data.iloc[a[i]]['Len of longest ORF'] > max_orf:\n", "                max_orf = df_ref_transcript_data.iloc[a[i]]['Len of longest ORF']\n", "                max_gc = df_ref_transcript_data.iloc[a[i]]['GC %']\n", "                index_max_all = a[i]\n", "            elif df_ref_transcript_data.iloc[a[i]]['Len of longest ORF'] == max_orf:\n", "                if df_ref_transcript_data.iloc[a[i]]['GC %'] > max_gc:\n", "                    max_gc = df_ref_transcript_data.iloc[a[i]]['GC %']\n", "                    index_max_all = a[i]\n", "        \n", "        cleaned_max_index.append(index_max_all)\n", "        cleaned_max_orf.append(max_orf)\n", "        cleaned_amino_acid.append(df_ref_transcript_data.iloc[index_max_all]['No. of Amino Acids'])\n", "        cleaned_gc_percentage.append(max_gc)\n", "        cleaned_max_orf_seq.append(df_ref_transcript_data.iloc[index_max_all]['Longest ORF Seq'])"]}, {"cell_type": "code", "execution_count": 44, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [01:02<00:00, 73.19it/s] "]}, {"name": "stdout", "output_type": "stream", "text": ["CPU times: user 47 s, sys: 1.71 s, total: 48.7 s\n", "Wall time: 1min 2s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "get_gene_orf()"]}, {"cell_type": "code", "execution_count": 45, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["[231, 9]\n", "[49.57, 39.04]\n", "['77', '3']\n"]}], "source": ["print(cleaned_max_orf[0:2])\n", "print(cleaned_gc_percentage[0:2])\n", "print(cleaned_amino_acid[0:2])"]}, {"cell_type": "code", "execution_count": 46, "metadata": {}, "outputs": [], "source": ["df_overlap['Max ORF length(REF)'] = cleaned_max_orf\n", "df_overlap['Max No. of amino acids(REF)'] = cleaned_amino_acid\n", "df_overlap['Max Transcript Index for ORF(REF)'] = cleaned_max_index\n", "df_overlap['GC % of trans row having Max ORF(REF)'] = cleaned_gc_percentage\n", "df_overlap['Max ORF Sequence(REF)'] = cleaned_max_orf_seq"]}, {"cell_type": "code", "execution_count": 47, "metadata": {}, "outputs": [], "source": ["df_overlap.to_csv('/feature_selection/ORF/Ens-Ref/ens_ref_ORF&overlap_cleaned(ens&ref).csv')"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}