{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 114, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_overlap = pd.read_csv('/feature_selection/overlap/Ens-Ref/ens_ref_gene_overlap_with_ov_len_and_no_exons_same_strand.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 115, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ENS lincRNA gene ID</th>\n", "      <th>Ref lncRNA gene ID</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>ENS-Ref Overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>Overlap Length</th>\n", "      <th>No. ENS lincrna Exons which overlap RefSeq exons</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000236648</td>\n", "      <td>gene-LOC107984028</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>4</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>17270</td>\n", "      <td>3</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230714</td>\n", "      <td>gene-LOC105372922</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>0</td>\n", "      <td>-</td>\n", "      <td>-</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["  ENS lincRNA gene ID Ref lncRNA gene ID Chr region  \\\n", "0     ENSG00000236648  gene-LOC107984028          1   \n", "1     ENSG00000230714  gene-LOC105372922          1   \n", "\n", "   No. of transcripts in lincrna  No. of transcripts in lncrna  \\\n", "0                              1                             4   \n", "1                              1                             1   \n", "\n", "  ENS-Ref Overlap ?  same_opp_overlap Overlap Length  \\\n", "0               Yes                 1          17270   \n", "1               Yes                 0              -   \n", "\n", "  No. ENS lincrna Exons which overlap RefSeq exons  \n", "0                                                3  \n", "1                                                -  "]}, "execution_count": 115, "metadata": {}, "output_type": "execute_result"}], "source": ["df_overlap.head(2)"]}, {"cell_type": "code", "execution_count": 116, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 6252/6252 [00:00<00:00, 39034.50it/s]\n"]}], "source": ["drop_index = []\n", "drop_no = 0\n", "for i in tqdm(range(len(df_overlap))):\n", "    if df_overlap['same_opp_overlap'][i] == 0:\n", "        drop_index.append(i)\n", "        drop_no = drop_no + 1"]}, {"cell_type": "code", "execution_count": 117, "metadata": {}, "outputs": [], "source": ["df_overlap.drop(drop_index, axis=0,inplace=True)\n", "df_overlap.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 118, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ENS lincRNA gene ID</th>\n", "      <th>Ref lncRNA gene ID</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>ENS-Ref Overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>Overlap Length</th>\n", "      <th>No. ENS lincrna Exons which overlap RefSeq exons</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000236648</td>\n", "      <td>gene-LOC107984028</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>4</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>17270</td>\n", "      <td>3</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>gene-LOC101928628</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>1720</td>\n", "      <td>2</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["  ENS lincRNA gene ID Ref lncRNA gene ID Chr region  \\\n", "0     ENSG00000236648  gene-LOC107984028          1   \n", "1     ENSG00000230704  gene-LOC101928628          1   \n", "\n", "   No. of transcripts in lincrna  No. of transcripts in lncrna  \\\n", "0                              1                             4   \n", "1                              1                             1   \n", "\n", "  ENS-Ref Overlap ?  same_opp_overlap Overlap Length  \\\n", "0               Yes                 1          17270   \n", "1               Yes                 1           1720   \n", "\n", "  No. ENS lincrna Exons which overlap RefSeq exons  \n", "0                                                3  \n", "1                                                2  "]}, "execution_count": 118, "metadata": {}, "output_type": "execute_result"}], "source": ["df_overlap.head(2)"]}, {"cell_type": "code", "execution_count": 119, "metadata": {}, "outputs": [], "source": ["df_overlap.to_csv('/feature_selection/overlap/Ens-Ref/ens_ref_gene_overlap_cleaned.csv')"]}, {"cell_type": "code", "execution_count": 120, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [00:00<00:00, 18110.64it/s]\n"]}], "source": ["from tqdm import tqdm\n", "l_id = []\n", "r_id = []\n", "for i in tqdm(range(len(df_overlap['ENS lincRNA gene ID']))):\n", "    l_id.append(df_overlap['ENS lincRNA gene ID'][i])\n", "    r_id.append(df_overlap['Ref lncRNA gene ID'][i])"]}, {"cell_type": "code", "execution_count": 121, "metadata": {}, "outputs": [], "source": ["set_lincrna_id = set(l_id)\n", "set_ref_id = set(r_id)"]}, {"cell_type": "code", "execution_count": 122, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data = pd.read_csv('/feature_selection/ORF/Ensembl/all_lincrna_transcript_data_orf_info.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 123, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 13666/13666 [00:00<00:00, 37216.98it/s]\n"]}], "source": ["drop_index = []\n", "drop_no = 0\n", "for i in tqdm(range(len(df_ens_transcript_data))):\n", "    if df_ens_transcript_data['Len of longest ORF'][i] == 'No start codon found':\n", "        drop_index.append(i)\n", "        drop_no = drop_no + 1"]}, {"cell_type": "code", "execution_count": 124, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data.drop(drop_index, axis=0,inplace=True)\n", "df_ens_transcript_data.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 125, "metadata": {}, "outputs": [{"data": {"text/plain": ["13637"]}, "execution_count": 125, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df_ens_transcript_data)"]}, {"cell_type": "code", "execution_count": 126, "metadata": {}, "outputs": [], "source": ["df_ens_transcript_data[\"Len of longest ORF\"] = pd.to_numeric(df_ens_transcript_data[\"Len of longest ORF\"])"]}, {"cell_type": "code", "execution_count": 127, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Transcript ID</th>\n", "      <th>Biotype</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>Transcript Start</th>\n", "      <th>Transcript End</th>\n", "      <th>Transcript Length</th>\n", "      <th>No. of Exons</th>\n", "      <th>...</th>\n", "      <th>Match ?</th>\n", "      <th>Sequences</th>\n", "      <th>GC %</th>\n", "      <th>Len of longest ORF</th>\n", "      <th>No. of Amino Acids</th>\n", "      <th>Longest ORF Seq</th>\n", "      <th>ORF Start Index for longest Seq</th>\n", "      <th>Protein Seq encoded</th>\n", "      <th>Stop codon present ?</th>\n", "      <th>ORF multiple of 3 ?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>4916</td>\n", "      <td>ENSG00000236423</td>\n", "      <td>ENST00000413332</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>3900372</td>\n", "      <td>3905532</td>\n", "      <td>634</td>\n", "      <td>3</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...</td>\n", "      <td>59.31</td>\n", "      <td>33</td>\n", "      <td>11</td>\n", "      <td>ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG</td>\n", "      <td>65</td>\n", "      <td>MFMRARPLAPL</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>11412</td>\n", "      <td>ENSG00000270035</td>\n", "      <td>ENST00000602973</td>\n", "      <td>lincRNA</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>7698303</td>\n", "      <td>7698872</td>\n", "      <td>570</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...</td>\n", "      <td>55.26</td>\n", "      <td>42</td>\n", "      <td>14</td>\n", "      <td>ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA</td>\n", "      <td>27</td>\n", "      <td>MVMPRAEDVRRQAV</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>11417</td>\n", "      <td>ENSG00000270066</td>\n", "      <td>ENST00000602755</td>\n", "      <td>lincRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>109100193</td>\n", "      <td>109100619</td>\n", "      <td>427</td>\n", "      <td>1</td>\n", "      <td>...</td>\n", "      <td>YES</td>\n", "      <td>GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...</td>\n", "      <td>65.81</td>\n", "      <td>138</td>\n", "      <td>46</td>\n", "      <td>ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...</td>\n", "      <td>227</td>\n", "      <td>MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR</td>\n", "      <td>YES</td>\n", "      <td>YES</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>3 rows \u00d7 22 columns</p>\n", "</div>"], "text/plain": ["     SNO          Gene ID    Transcript ID  Biotype  Strand Seq region Name  \\\n", "0   4916  ENSG00000236423  ENST00000413332  lincRNA       1               1   \n", "1  11412  ENSG00000270035  ENST00000602973  lincRNA      -1               1   \n", "2  11417  ENSG00000270066  ENST00000602755  lincRNA       1               1   \n", "\n", "   Transcript Start  Transcript End  Transcript Length  No. of Exons  ...  \\\n", "0           3900372         3905532                634             3  ...   \n", "1           7698303         7698872                570             1  ...   \n", "2         109100193       109100619                427             1  ...   \n", "\n", "   Match ?                                          Sequences   GC %  \\\n", "0      YES  GCGAAGGTCACAGGCCTTGGGCCAGTGGTGGGACACTGGAGCAGGA...  59.31   \n", "1      YES  GGCACAGGGGAGGCCCTGAGTCACACGATGGTGATGCCCAGGGCAG...  55.26   \n", "2      YES  GTTTTAGGGAGGGAGAGCGGCCTGGGTCCTGGGTGTTGTGTGCGGA...  65.81   \n", "\n", "  Len of longest ORF  No. of Amino Acids  \\\n", "0                 33                  11   \n", "1                 42                  14   \n", "2                138                  46   \n", "\n", "                                     Longest ORF Seq  \\\n", "0               ATGTTCATGAGGGCGCGTCCCTTGGCACCCCTGTAG   \n", "1      ATGGTGATGCCCAGGGCAGAAGATGTCAGAAGACAGGCAGTTTGA   \n", "2  ATGAAAAGCTCTCCTCCTGCGAAGCCCCCTCGGGGCGCTGTGCAGC...   \n", "\n", "  ORF Start Index for longest Seq  \\\n", "0                              65   \n", "1                              27   \n", "2                             227   \n", "\n", "                              Protein Seq encoded Stop codon present ?  \\\n", "0                                     MFMRARPLAPL                  YES   \n", "1                                  MVMPRAEDVRRQAV                  YES   \n", "2  MKSSPPAKPPRGAVQRGPLGGGHAVVPEVPELALRGPALRSDELIR                  YES   \n", "\n", "  ORF multiple of 3 ?  \n", "0                 YES  \n", "1                 YES  \n", "2                 YES  \n", "\n", "[3 rows x 22 columns]"]}, "execution_count": 127, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_transcript_data.head(3)"]}, {"cell_type": "code", "execution_count": 140, "metadata": {}, "outputs": [], "source": ["cleaned_max_orf = []\n", "cleaned_amino_acid = []\n", "cleaned_max_index = []\n", "cleaned_gc_percentage = []\n", "cleaned_max_orf_seq = []"]}, {"cell_type": "code", "execution_count": 141, "metadata": {}, "outputs": [], "source": ["def get_gene_orf():\n", "    for i in tqdm(range(len(df_overlap))):\n", "        a = df_ens_transcript_data.index[df_ens_transcript_data['Gene ID'] == df_overlap['ENS lincRNA gene ID'][i]]\n", "        max_orf = 0\n", "        max_gc = 0\n", "        index_max_all = 0\n", "        for i in range(len(a)):\n", "            if df_ens_transcript_data.iloc[a[i]]['Len of longest ORF'] > max_orf:\n", "                max_orf = df_ens_transcript_data.iloc[a[i]]['Len of longest ORF']\n", "                max_gc = df_ens_transcript_data.iloc[a[i]]['GC %']\n", "                index_max_all = a[i]\n", "            elif df_ens_transcript_data.iloc[a[i]]['Len of longest ORF'] == max_orf:\n", "                if df_ens_transcript_data.iloc[a[i]]['GC %'] > max_gc:\n", "                    max_gc = df_ens_transcript_data.iloc[a[i]]['GC %']\n", "                    index_max_all = a[i]\n", "        \n", "        cleaned_max_index.append(index_max_all)\n", "        cleaned_max_orf.append(max_orf)\n", "        cleaned_amino_acid.append(df_ens_transcript_data.iloc[index_max_all]['No. of Amino Acids'])\n", "        cleaned_gc_percentage.append(max_gc)\n", "        cleaned_max_orf_seq.append(df_ens_transcript_data.iloc[index_max_all]['Longest ORF Seq'])"]}, {"cell_type": "code", "execution_count": 142, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [01:09<00:00, 66.04it/s] "]}, {"name": "stdout", "output_type": "stream", "text": ["CPU times: user 48.9 s, sys: 1.78 s, total: 50.7 s\n", "Wall time: 1min 9s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "get_gene_orf()"]}, {"cell_type": "code", "execution_count": 147, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["201\n", "43.26\n", "67\n"]}], "source": ["print(cleaned_max_orf[5])\n", "print(cleaned_gc_percentage[5])\n", "print(cleaned_amino_acid[5])"]}, {"cell_type": "code", "execution_count": 157, "metadata": {}, "outputs": [], "source": ["df_overlap['Max ORF length(ENS)'] = cleaned_max_orf\n", "df_overlap['Max No. of amino acids(ENS)'] = cleaned_amino_acid\n", "df_overlap['Max Transcript Index for ORF(ENS)'] = cleaned_max_index\n", "df_overlap['GC % of trans row having Max ORF(ENS)'] = cleaned_gc_percentage\n", "df_overlap['Max ORF Sequence(ENS)'] = cleaned_max_orf_seq"]}, {"cell_type": "code", "execution_count": 159, "metadata": {}, "outputs": [], "source": ["df_overlap.to_csv('/feature_selection/ORF/Ens-Ref/ens_ref_ORF&overlap_cleaned(ens).csv')"]}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}