{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_ens_linc_genes = pd.read_csv('/dataset/Ensembl_numeric/all_lincRNA_data/all_lincRNA_sorted_chr/all_lincrna_gene_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Gene ID</th>\n", "      <th>Display Name</th>\n", "      <th>Biotype</th>\n", "      <th>Start</th>\n", "      <th>End</th>\n", "      <th>Strand</th>\n", "      <th>Seq region Name</th>\n", "      <th>No. of Transcripts</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>6092</td>\n", "      <td>ENSG00000271387</td>\n", "      <td>AL445228.2</td>\n", "      <td>lincRNA</td>\n", "      <td>184385753</td>\n", "      <td>184386704</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>2329</td>\n", "      <td>ENSG00000236648</td>\n", "      <td>AL136529.1</td>\n", "      <td>lincRNA</td>\n", "      <td>17717687</td>\n", "      <td>17749978</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>589</td>\n", "      <td>ENSG00000225359</td>\n", "      <td>AL358393.1</td>\n", "      <td>lincRNA</td>\n", "      <td>181190471</td>\n", "      <td>181191149</td>\n", "      <td>-1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["    SNO          Gene ID Display Name  Biotype      Start        End  Strand  \\\n", "0  6092  ENSG00000271387   AL445228.2  lincRNA  184385753  184386704      -1   \n", "1  2329  ENSG00000236648   AL136529.1  lincRNA   17717687   17749978       1   \n", "2   589  ENSG00000225359   AL358393.1  lincRNA  181190471  181191149      -1   \n", "\n", "  Seq region Name  No. of Transcripts  \n", "0               1                   1  \n", "1               1                   1  \n", "2               1                   1  "]}, "execution_count": 2, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_linc_genes.head(3)"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [], "source": ["df_ens_linc_index = df_ens_linc_genes.set_index(['Gene ID'])"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [], "source": ["df_feature_file = pd.read_csv('/feature_selection/ORF/Ens-Ref/ens_ref_ORF&overlap_cleaned(ens&ref).csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [{"data": {"text/plain": ["4557"]}, "execution_count": 5, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df_feature_file)"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ENS lincRNA gene ID</th>\n", "      <th>Ref lncRNA gene ID</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>ENS-Ref Overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>Overlap Length</th>\n", "      <th>No. ENS lincrna Exons which overlap RefSeq exons</th>\n", "      <th>Max ORF length(ENS)</th>\n", "      <th>Max No. of amino acids(ENS)</th>\n", "      <th>Max Transcript Index for ORF(ENS)</th>\n", "      <th>GC % of trans row having Max ORF(ENS)</th>\n", "      <th>Max ORF Sequence(ENS)</th>\n", "      <th>Max ORF length(REF)</th>\n", "      <th>Max No. of amino acids(REF)</th>\n", "      <th>Max Transcript Index for ORF(REF)</th>\n", "      <th>GC % of trans row having Max ORF(REF)</th>\n", "      <th>Max ORF Sequence(REF)</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000236648</td>\n", "      <td>gene-LOC107984028</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>4</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>17270</td>\n", "      <td>3</td>\n", "      <td>90</td>\n", "      <td>30</td>\n", "      <td>153</td>\n", "      <td>51.97</td>\n", "      <td>ATGGCTGCAAATGGCTCCTCCGTGCAGGTGAGGACATGGAAGCACA...</td>\n", "      <td>231</td>\n", "      <td>77</td>\n", "      <td>1610</td>\n", "      <td>49.57</td>\n", "      <td>ATGTCAGCGTGGCTGCCCATGGGATCCCAGCCTGGTTTGCCTTTAA...</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>gene-LOC101928628</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>1720</td>\n", "      <td>2</td>\n", "      <td>36</td>\n", "      <td>12</td>\n", "      <td>818</td>\n", "      <td>40.92</td>\n", "      <td>ATGTATTCCTCTCTCTCTCTCACCTATTCCCATGTTTAA</td>\n", "      <td>9</td>\n", "      <td>3</td>\n", "      <td>37</td>\n", "      <td>39.04</td>\n", "      <td>ATGTATCAATAG</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>ENSG00000230699</td>\n", "      <td>gene-LOC284600</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>3348</td>\n", "      <td>1</td>\n", "      <td>18</td>\n", "      <td>6</td>\n", "      <td>819</td>\n", "      <td>59.68</td>\n", "      <td>ATGGCAAGTCAGTGGATTTAG</td>\n", "      <td>27</td>\n", "      <td>9</td>\n", "      <td>1686</td>\n", "      <td>59.80</td>\n", "      <td>ATGGCTGTGATTTCAGAAAAGCGGACGTAA</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["  ENS lincRNA gene ID Ref lncRNA gene ID Chr region  \\\n", "0     ENSG00000236648  gene-LOC107984028          1   \n", "1     ENSG00000230704  gene-LOC101928628          1   \n", "2     ENSG00000230699     gene-LOC284600          1   \n", "\n", "   No. of transcripts in lincrna  No. of transcripts in lncrna  \\\n", "0                              1                             4   \n", "1                              1                             1   \n", "2                              1                             1   \n", "\n", "  ENS-Ref Overlap ?  same_opp_overlap Overlap Length  \\\n", "0               Yes                 1          17270   \n", "1               Yes                 1           1720   \n", "2               Yes                 1           3348   \n", "\n", "   No. ENS lincrna Exons which overlap RefSeq exons  Max ORF length(ENS)  \\\n", "0                                                 3                   90   \n", "1                                                 2                   36   \n", "2                                                 1                   18   \n", "\n", "   Max No. of amino acids(ENS)  Max Transcript Index for ORF(ENS)  \\\n", "0                           30                                153   \n", "1                           12                                818   \n", "2                            6                                819   \n", "\n", "   GC % of trans row having Max ORF(ENS)  \\\n", "0                                  51.97   \n", "1                                  40.92   \n", "2                                  59.68   \n", "\n", "                               Max ORF Sequence(ENS)  Max ORF length(REF)  \\\n", "0  ATGGCTGCAAATGGCTCCTCCGTGCAGGTGAGGACATGGAAGCACA...                  231   \n", "1            ATGTATTCCTCTCTCTCTCTCACCTATTCCCATGTTTAA                    9   \n", "2                              ATGGCAAGTCAGTGGATTTAG                   27   \n", "\n", "   Max No. of amino acids(REF)  Max Transcript Index for ORF(REF)  \\\n", "0                           77                               1610   \n", "1                            3                                 37   \n", "2                            9                               1686   \n", "\n", "   GC % of trans row having Max ORF(REF)  \\\n", "0                                  49.57   \n", "1                                  39.04   \n", "2                                  59.80   \n", "\n", "                               Max ORF Sequence(REF)  \n", "0  ATGTCAGCGTGGCTGCCCATGGGATCCCAGCCTGGTTTGCCTTTAA...  \n", "1                                       ATGTATCAATAG  \n", "2                     ATGGCTGTGATTTCAGAAAAGCGGACGTAA  "]}, "execution_count": 6, "metadata": {}, "output_type": "execute_result"}], "source": ["df_feature_file.head(3)"]}, {"cell_type": "code", "execution_count": 8, "metadata": {}, "outputs": [], "source": ["ens_linc_gene_ids = df_feature_file['ENS lincRNA gene ID'].tolist()"]}, {"cell_type": "code", "execution_count": 12, "metadata": {}, "outputs": [{"data": {"text/plain": ["'AL13'"]}, "execution_count": 12, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_linc_index.loc['ENSG00000236648']['Display Name'][0:4]"]}, {"cell_type": "code", "execution_count": 13, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 14, "metadata": {}, "outputs": [], "source": ["ens_linc_hgnc = []"]}, {"cell_type": "code", "execution_count": 15, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 6 \u00b5s, sys: 1 \u00b5s, total: 7 \u00b5s\n", "Wall time: 11.9 \u00b5s\n"]}], "source": ["%%time\n", "def is_ens_linc_hgnc():\n", "    for i in tqdm(range(len(ens_linc_gene_ids))):\n", "        if df_ens_linc_index.loc[ens_linc_gene_ids[i]]['Display Name'][0:4] == 'LINC':\n", "            ens_linc_hgnc.append(1)\n", "        else:\n", "            ens_linc_hgnc.append(0)"]}, {"cell_type": "code", "execution_count": 16, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [00:03<00:00, 1139.57it/s]"]}, {"name": "stdout", "output_type": "stream", "text": ["CPU times: user 3.1 s, sys: 111 ms, total: 3.21 s\n", "Wall time: 4.02 s\n"]}, {"name": "stderr", "output_type": "stream", "text": ["\n"]}], "source": ["%%time\n", "is_ens_linc_hgnc()"]}, {"cell_type": "markdown", "metadata": {}, "source": ["## Ref lnc HGNC count done below"]}, {"cell_type": "code", "execution_count": 18, "metadata": {}, "outputs": [], "source": ["df_ref_lnc_genes = pd.read_csv('/dataset/refseq_numeric/all_type=lncRNA_data/all_lncrna_sorted_chr/all_lncrna_gene_data_sorted_chr.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 19, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>SNO</th>\n", "      <th>Index</th>\n", "      <th>RefSeq Gene ID</th>\n", "      <th>Dbxref</th>\n", "      <th>Name</th>\n", "      <th>gbkey</th>\n", "      <th>Biotype</th>\n", "      <th>Type</th>\n", "      <th>Seq_id</th>\n", "      <th>Strand</th>\n", "      <th>Start</th>\n", "      <th>End</th>\n", "      <th>No. of Transcripts</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>1</td>\n", "      <td>26</td>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>GeneID:107985730,HGNC:HGNC:52482</td>\n", "      <td>MIR1302-2HG</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>29926</td>\n", "      <td>31295</td>\n", "      <td>1</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>877</td>\n", "      <td>256858</td>\n", "      <td>gene-LINC01142</td>\n", "      <td>GeneID:284688,HGNC:HGNC:49456</td>\n", "      <td>LINC01142</td>\n", "      <td>Gene</td>\n", "      <td>lncRNA</td>\n", "      <td>gene</td>\n", "      <td>1</td>\n", "      <td>-1</td>\n", "      <td>170271405</td>\n", "      <td>170284208</td>\n", "      <td>1</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["   SNO   Index    RefSeq Gene ID                            Dbxref  \\\n", "0    1      26  gene-MIR1302-2HG  GeneID:107985730,HGNC:HGNC:52482   \n", "1  877  256858    gene-LINC01142     GeneID:284688,HGNC:HGNC:49456   \n", "\n", "          Name gbkey Biotype  Type Seq_id  Strand      Start        End  \\\n", "0  MIR1302-2HG  Gene  lncRNA  gene      1       1      29926      31295   \n", "1    LINC01142  Gene  lncRNA  gene      1      -1  170271405  170284208   \n", "\n", "   No. of Transcripts  \n", "0                   1  \n", "1                   1  "]}, "execution_count": 19, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_lnc_genes.head(2)"]}, {"cell_type": "code", "execution_count": 20, "metadata": {}, "outputs": [], "source": ["df_ref_lnc_index = df_ref_lnc_genes.set_index(['RefSeq Gene ID'])"]}, {"cell_type": "code", "execution_count": 21, "metadata": {}, "outputs": [], "source": ["ref_lnc_gene_ids = df_feature_file['Ref lncRNA gene ID'].tolist()"]}, {"cell_type": "code", "execution_count": 22, "metadata": {}, "outputs": [{"data": {"text/plain": ["'MIR1'"]}, "execution_count": 22, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_lnc_index.loc['gene-MIR1302-2HG']['Name'][0:4]"]}, {"cell_type": "code", "execution_count": 24, "metadata": {}, "outputs": [], "source": ["ref_lnc_hgnc = []"]}, {"cell_type": "code", "execution_count": 25, "metadata": {}, "outputs": [{"name": "stdout", "output_type": "stream", "text": ["CPU times: user 7 \u00b5s, sys: 2 \u00b5s, total: 9 \u00b5s\n", "Wall time: 14.1 \u00b5s\n"]}], "source": ["%%time\n", "def is_ref_lnc_hgnc():\n", "    for i in tqdm(range(len(ref_lnc_gene_ids))):\n", "        if df_ref_lnc_index.loc[ref_lnc_gene_ids[i]]['Name'][0:4] == 'LINC':\n", "            ref_lnc_hgnc.append(1)\n", "        else:\n", "            ref_lnc_hgnc.append(0)"]}, {"cell_type": "code", "execution_count": 26, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [00:04<00:00, 1070.18it/s]\n"]}], "source": ["is_ref_lnc_hgnc()"]}, {"cell_type": "code", "execution_count": 28, "metadata": {}, "outputs": [{"data": {"text/plain": ["4557"]}, "execution_count": 28, "metadata": {}, "output_type": "execute_result"}], "source": ["len(ref_lnc_hgnc)"]}, {"cell_type": "code", "execution_count": 29, "metadata": {}, "outputs": [{"data": {"text/plain": ["4557"]}, "execution_count": 29, "metadata": {}, "output_type": "execute_result"}], "source": ["len(ens_linc_hgnc)"]}, {"cell_type": "code", "execution_count": 30, "metadata": {}, "outputs": [], "source": ["df_feature_file['HGNC ID refseq lncrna?'] = ref_lnc_hgnc\n", "df_feature_file['HGNC ID ensembl lincrna?'] = ens_linc_hgnc"]}, {"cell_type": "code", "execution_count": 31, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ENS lincRNA gene ID</th>\n", "      <th>Ref lncRNA gene ID</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>ENS-Ref Overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>Overlap Length</th>\n", "      <th>No. ENS lincrna Exons which overlap RefSeq exons</th>\n", "      <th>Max ORF length(ENS)</th>\n", "      <th>...</th>\n", "      <th>Max Transcript Index for ORF(ENS)</th>\n", "      <th>GC % of trans row having Max ORF(ENS)</th>\n", "      <th>Max ORF Sequence(ENS)</th>\n", "      <th>Max ORF length(REF)</th>\n", "      <th>Max No. of amino acids(REF)</th>\n", "      <th>Max Transcript Index for ORF(REF)</th>\n", "      <th>GC % of trans row having Max ORF(REF)</th>\n", "      <th>Max ORF Sequence(REF)</th>\n", "      <th>HGNC ID refseq lncrna?</th>\n", "      <th>HGNC ID ensembl lincrna?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000236648</td>\n", "      <td>gene-LOC107984028</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>4</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>17270</td>\n", "      <td>3</td>\n", "      <td>90</td>\n", "      <td>...</td>\n", "      <td>153</td>\n", "      <td>51.97</td>\n", "      <td>ATGGCTGCAAATGGCTCCTCCGTGCAGGTGAGGACATGGAAGCACA...</td>\n", "      <td>231</td>\n", "      <td>77</td>\n", "      <td>1610</td>\n", "      <td>49.57</td>\n", "      <td>ATGTCAGCGTGGCTGCCCATGGGATCCCAGCCTGGTTTGCCTTTAA...</td>\n", "      <td>0</td>\n", "      <td>0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>gene-LOC101928628</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>1720</td>\n", "      <td>2</td>\n", "      <td>36</td>\n", "      <td>...</td>\n", "      <td>818</td>\n", "      <td>40.92</td>\n", "      <td>ATGTATTCCTCTCTCTCTCTCACCTATTCCCATGTTTAA</td>\n", "      <td>9</td>\n", "      <td>3</td>\n", "      <td>37</td>\n", "      <td>39.04</td>\n", "      <td>ATGTATCAATAG</td>\n", "      <td>0</td>\n", "      <td>0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>2</th>\n", "      <td>ENSG00000230699</td>\n", "      <td>gene-LOC284600</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>3348</td>\n", "      <td>1</td>\n", "      <td>18</td>\n", "      <td>...</td>\n", "      <td>819</td>\n", "      <td>59.68</td>\n", "      <td>ATGGCAAGTCAGTGGATTTAG</td>\n", "      <td>27</td>\n", "      <td>9</td>\n", "      <td>1686</td>\n", "      <td>59.80</td>\n", "      <td>ATGGCTGTGATTTCAGAAAAGCGGACGTAA</td>\n", "      <td>0</td>\n", "      <td>0</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>3 rows \u00d7 21 columns</p>\n", "</div>"], "text/plain": ["  ENS lincRNA gene ID Ref lncRNA gene ID Chr region  \\\n", "0     ENSG00000236648  gene-LOC107984028          1   \n", "1     ENSG00000230704  gene-LOC101928628          1   \n", "2     ENSG00000230699     gene-LOC284600          1   \n", "\n", "   No. of transcripts in lincrna  No. of transcripts in lncrna  \\\n", "0                              1                             4   \n", "1                              1                             1   \n", "2                              1                             1   \n", "\n", "  ENS-Ref Overlap ?  same_opp_overlap Overlap Length  \\\n", "0               Yes                 1          17270   \n", "1               Yes                 1           1720   \n", "2               Yes                 1           3348   \n", "\n", "   No. ENS lincrna Exons which overlap RefSeq exons  Max ORF length(ENS)  ...  \\\n", "0                                                 3                   90  ...   \n", "1                                                 2                   36  ...   \n", "2                                                 1                   18  ...   \n", "\n", "   Max Transcript Index for ORF(ENS)  GC % of trans row having Max ORF(ENS)  \\\n", "0                                153                                  51.97   \n", "1                                818                                  40.92   \n", "2                                819                                  59.68   \n", "\n", "                               Max ORF Sequence(ENS) Max ORF length(REF)  \\\n", "0  ATGGCTGCAAATGGCTCCTCCGTGCAGGTGAGGACATGGAAGCACA...                 231   \n", "1            ATGTATTCCTCTCTCTCTCTCACCTATTCCCATGTTTAA                   9   \n", "2                              ATGGCAAGTCAGTGGATTTAG                  27   \n", "\n", "   Max No. of amino acids(REF)  Max Transcript Index for ORF(REF)  \\\n", "0                           77                               1610   \n", "1                            3                                 37   \n", "2                            9                               1686   \n", "\n", "   GC % of trans row having Max ORF(REF)  \\\n", "0                                  49.57   \n", "1                                  39.04   \n", "2                                  59.80   \n", "\n", "                               Max ORF Sequence(REF) HGNC ID refseq lncrna?  \\\n", "0  ATGTCAGCGTGGCTGCCCATGGGATCCCAGCCTGGTTTGCCTTTAA...                      0   \n", "1                                       ATGTATCAATAG                      0   \n", "2                     ATGGCTGTGATTTCAGAAAAGCGGACGTAA                      0   \n", "\n", "   HGNC ID ensembl lincrna?  \n", "0                         0  \n", "1                         0  \n", "2                         0  \n", "\n", "[3 rows x 21 columns]"]}, "execution_count": 31, "metadata": {}, "output_type": "execute_result"}], "source": ["df_feature_file.head(3)"]}, {"cell_type": "code", "execution_count": 32, "metadata": {}, "outputs": [], "source": ["df_feature_file.to_csv('/feature_selection/check_hgnc_1/cleaned_feature_file_v1.csv')"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}