{"cells": [{"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["####################################################################################################\n", "\n", "#    Copyright 2019 Srijan Verma and EMBL-European Bioinformatics Institute\n", "\n", "#    Licensed under the Apache License, Version 2.0 (the \"License\");\n", "#    you may not use this file except in compliance with the License.\n", "#    You may obtain a copy of the License at\n", "\n", "#        http://www.apache.org/licenses/LICENSE-2.0\n", "\n", "#    Unless required by applicable law or agreed to in writing, software\n", "#    distributed under the License is distributed on an \"AS IS\" BASIS,\n", "#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n", "#    See the License for the specific language governing permissions and\n", "#    limitations under the License.\n", "\n", "####################################################################################################"]}, {"cell_type": "code", "execution_count": 1, "metadata": {}, "outputs": [], "source": ["import pandas as pd\n", "df_feature_file = pd.read_csv('/feature_selection/seq_align_score_v8/seq_alig_score_ens-ref_v8.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 9, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>ENS lincRNA gene ID</th>\n", "      <th>Ref lncRNA gene ID</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>ENS-Ref Overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>Overlap Length</th>\n", "      <th>No. ENS lincrna Exons which overlap RefSeq exons</th>\n", "      <th>Max ORF length(ENS)</th>\n", "      <th>...</th>\n", "      <th>Inflated overlap 1kb ens(linc)-ens(protein_coding)</th>\n", "      <th>is ref LOC ?</th>\n", "      <th>Inflated overlap 1kb ref(lnc)-ref(protein_coding)</th>\n", "      <th>Inflated overlap 0.5kb ens(linc)-ens(linc)</th>\n", "      <th>Inflated overlap 0.5kb ref(lnc)-ref(lnc)</th>\n", "      <th>Max exon count</th>\n", "      <th>Max exon length</th>\n", "      <th>Ens ID for seq alignment score</th>\n", "      <th>Ref ID for seq alignment score</th>\n", "      <th>Seq Alignment Score</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000236648</td>\n", "      <td>gene-LOC107984028</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>4</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>17270</td>\n", "      <td>3</td>\n", "      <td>90</td>\n", "      <td>...</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>4</td>\n", "      <td>308</td>\n", "      <td>ENST00000430540</td>\n", "      <td>rna-XR_002958657.1</td>\n", "      <td>560.0</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>gene-LOC101928628</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>1720</td>\n", "      <td>2</td>\n", "      <td>36</td>\n", "      <td>...</td>\n", "      <td>0</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>2</td>\n", "      <td>677</td>\n", "      <td>ENST00000454271</td>\n", "      <td>rna-XR_241147.3</td>\n", "      <td>677.0</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "<p>2 rows \u00d7 33 columns</p>\n", "</div>"], "text/plain": ["  ENS lincRNA gene ID Ref lncRNA gene ID Chr region  \\\n", "0     ENSG00000236648  gene-LOC107984028          1   \n", "1     ENSG00000230704  gene-LOC101928628          1   \n", "\n", "   No. of transcripts in lincrna  No. of transcripts in lncrna  \\\n", "0                              1                             4   \n", "1                              1                             1   \n", "\n", "  ENS-Ref Overlap ?  same_opp_overlap Overlap Length  \\\n", "0               Yes                 1          17270   \n", "1               Yes                 1           1720   \n", "\n", "   No. ENS lincrna Exons which overlap RefSeq exons  Max ORF length(ENS)  ...  \\\n", "0                                                 3                   90  ...   \n", "1                                                 2                   36  ...   \n", "\n", "   Inflated overlap 1kb ens(linc)-ens(protein_coding)  is ref LOC ?  \\\n", "0                                                  1              1   \n", "1                                                  0              1   \n", "\n", "   Inflated overlap 1kb ref(lnc)-ref(protein_coding)  \\\n", "0                                                  1   \n", "1                                                  1   \n", "\n", "  Inflated overlap 0.5kb ens(linc)-ens(linc)  \\\n", "0                                          1   \n", "1                                          1   \n", "\n", "   Inflated overlap 0.5kb ref(lnc)-ref(lnc)  Max exon count  Max exon length  \\\n", "0                                         1               4              308   \n", "1                                         1               2              677   \n", "\n", "   Ens ID for seq alignment score Ref ID for seq alignment score  \\\n", "0                 ENST00000430540             rna-XR_002958657.1   \n", "1                 ENST00000454271                rna-XR_241147.3   \n", "\n", "   Seq Alignment Score  \n", "0                560.0  \n", "1                677.0  \n", "\n", "[2 rows x 33 columns]"]}, "execution_count": 9, "metadata": {}, "output_type": "execute_result"}], "source": ["df_feature_file.head(2)"]}, {"cell_type": "code", "execution_count": 2, "metadata": {}, "outputs": [], "source": ["df_ens_ens_overlap = pd.read_csv('/feature_selection/overlap/Ensembl/all_gene_overlap.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 3, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>lincRNA ID</th>\n", "      <th>all gene ID</th>\n", "      <th>All Gene biotype</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lincrna</th>\n", "      <th>No. of transcripts in all gene</th>\n", "      <th>Ensembl-Ensembl overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>Ensembl-linc_Ensembl overlap?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>ENSG00000000460</td>\n", "      <td>protein_coding</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>9</td>\n", "      <td>Yes</td>\n", "      <td>0</td>\n", "      <td>No</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>ENSG00000230704</td>\n", "      <td>ENSG00000007908</td>\n", "      <td>protein_coding</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>7</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>No</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["        lincRNA ID      all gene ID All Gene biotype Chr region  \\\n", "0  ENSG00000230704  ENSG00000000460   protein_coding          1   \n", "1  ENSG00000230704  ENSG00000007908   protein_coding          1   \n", "\n", "   No. of transcripts in lincrna  No. of transcripts in all gene  \\\n", "0                              1                               9   \n", "1                              1                               7   \n", "\n", "  Ensembl-Ensembl overlap ?  same_opp_overlap Ensembl-linc_Ensembl overlap?  \n", "0                       Yes                 0                            No  \n", "1                       Yes                 1                            No  "]}, "execution_count": 3, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ens_ens_overlap.head(2)"]}, {"cell_type": "code", "execution_count": 4, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 5, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 5509/5509 [00:00<00:00, 29545.09it/s]\n"]}], "source": ["drop_index = []\n", "for i in tqdm(range(len(df_ens_ens_overlap))):\n", "    if df_ens_ens_overlap['All Gene biotype'][i] == 'protein_coding':\n", "        continue\n", "    else:\n", "        drop_index.append(i)"]}, {"cell_type": "code", "execution_count": 6, "metadata": {}, "outputs": [], "source": ["df_ens_ens_overlap.drop(drop_index, axis=0,inplace=True)\n", "df_ens_ens_overlap.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 7, "metadata": {}, "outputs": [{"data": {"text/plain": ["1003"]}, "execution_count": 7, "metadata": {}, "output_type": "execute_result"}], "source": ["len(df_ens_ens_overlap)"]}, {"cell_type": "code", "execution_count": 8, "metadata": {}, "outputs": [], "source": ["set_ens_linc_ids = set(df_ens_ens_overlap['lincRNA ID'].tolist())"]}, {"cell_type": "code", "execution_count": 10, "metadata": {}, "outputs": [], "source": ["protein_ens = []"]}, {"cell_type": "code", "execution_count": 14, "metadata": {}, "outputs": [], "source": ["from tqdm import tqdm"]}, {"cell_type": "code", "execution_count": 15, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [00:00<00:00, 34012.05it/s]\n"]}], "source": ["for i in tqdm(range(len(df_feature_file))):\n", "    if df_feature_file['ENS lincRNA gene ID'][i] in set_ens_linc_ids:\n", "        protein_ens.append(0)\n", "    else:\n", "        protein_ens.append(1)"]}, {"cell_type": "code", "execution_count": 17, "metadata": {}, "outputs": [{"data": {"text/plain": ["4557"]}, "execution_count": 17, "metadata": {}, "output_type": "execute_result"}], "source": ["len(protein_ens)"]}, {"cell_type": "code", "execution_count": 18, "metadata": {}, "outputs": [], "source": ["df_ref_ref_overlap = pd.read_csv('/feature_selection/overlap/Refseq/all_gene_overlap.csv', index_col=0)"]}, {"cell_type": "code", "execution_count": 19, "metadata": {}, "outputs": [{"data": {"text/html": ["<div>\n", "<style scoped>\n", "    .dataframe tbody tr th:only-of-type {\n", "        vertical-align: middle;\n", "    }\n", "\n", "    .dataframe tbody tr th {\n", "        vertical-align: top;\n", "    }\n", "\n", "    .dataframe thead th {\n", "        text-align: right;\n", "    }\n", "</style>\n", "<table border=\"1\" class=\"dataframe\">\n", "  <thead>\n", "    <tr style=\"text-align: right;\">\n", "      <th></th>\n", "      <th>lncRNA ID</th>\n", "      <th>all gene ID</th>\n", "      <th>Gene Biotype</th>\n", "      <th>Chr region</th>\n", "      <th>No. of transcripts in lncrna</th>\n", "      <th>No. of transcripts in all gene</th>\n", "      <th>RefSeq-RefSeq overlap ?</th>\n", "      <th>same_opp_overlap</th>\n", "      <th>RefSeq-lnc_RefSeq overlap?</th>\n", "    </tr>\n", "  </thead>\n", "  <tbody>\n", "    <tr>\n", "      <th>0</th>\n", "      <td>gene-MIR1302-2HG</td>\n", "      <td>gene-MIR1302-2</td>\n", "      <td>miRNA</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>2</td>\n", "      <td>Yes</td>\n", "      <td>1</td>\n", "      <td>No</td>\n", "    </tr>\n", "    <tr>\n", "      <th>1</th>\n", "      <td>gene-LINC01681</td>\n", "      <td>gene-ISCUP1</td>\n", "      <td>pseudogene</td>\n", "      <td>1</td>\n", "      <td>1</td>\n", "      <td>0</td>\n", "      <td>Yes</td>\n", "      <td>0</td>\n", "      <td>No</td>\n", "    </tr>\n", "  </tbody>\n", "</table>\n", "</div>"], "text/plain": ["          lncRNA ID     all gene ID Gene Biotype Chr region  \\\n", "0  gene-MIR1302-2HG  gene-MIR1302-2        miRNA          1   \n", "1    gene-LINC01681     gene-ISCUP1   pseudogene          1   \n", "\n", "   No. of transcripts in lncrna  No. of transcripts in all gene  \\\n", "0                             1                               2   \n", "1                             1                               0   \n", "\n", "  RefSeq-RefSeq overlap ?  same_opp_overlap RefSeq-lnc_RefSeq overlap?  \n", "0                     Yes                 1                         No  \n", "1                     Yes                 0                         No  "]}, "execution_count": 19, "metadata": {}, "output_type": "execute_result"}], "source": ["df_ref_ref_overlap.head(2)"]}, {"cell_type": "code", "execution_count": 20, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 11278/11278 [00:00<00:00, 36024.24it/s]\n"]}], "source": ["drop_index = []\n", "for i in tqdm(range(len(df_ref_ref_overlap))):\n", "    if df_ref_ref_overlap['Gene Biotype'][i] == 'protein_coding':\n", "        continue\n", "    else:\n", "        drop_index.append(i)"]}, {"cell_type": "code", "execution_count": 21, "metadata": {}, "outputs": [], "source": ["df_ref_ref_overlap.drop(drop_index, axis=0,inplace=True)\n", "df_ref_ref_overlap.reset_index(drop=True, inplace=True)"]}, {"cell_type": "code", "execution_count": 22, "metadata": {}, "outputs": [], "source": ["set_ref_lnc_ids = set(df_ref_ref_overlap['lncRNA ID'].tolist())"]}, {"cell_type": "code", "execution_count": 24, "metadata": {}, "outputs": [], "source": ["protein_ref = []"]}, {"cell_type": "code", "execution_count": 25, "metadata": {}, "outputs": [{"name": "stderr", "output_type": "stream", "text": ["100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 4557/4557 [00:00<00:00, 36021.44it/s]\n"]}], "source": ["for i in tqdm(range(len(df_feature_file))):\n", "    if df_feature_file['Ref lncRNA gene ID'][i] in set_ref_lnc_ids:\n", "        protein_ref.append(0)\n", "    else:\n", "        protein_ref.append(1) # Means no overlap"]}, {"cell_type": "code", "execution_count": 26, "metadata": {}, "outputs": [], "source": ["df_feature_file['ens(linc)-ens(protein-coding) overlap ?'] = protein_ens\n", "df_feature_file['ref(lnc)-ref(protein-coding) overlap ?'] = protein_ref"]}, {"cell_type": "code", "execution_count": 27, "metadata": {}, "outputs": [], "source": ["df_feature_file.to_csv('/feature_selection/protein_overlap_v9/protein_overlap_v9.csv')"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": []}], "metadata": {"kernelspec": {"display_name": "Python [conda env:gsoc_ensembl]", "language": "python", "name": "conda-env-gsoc_ensembl-py"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.6.5"}}, "nbformat": 4, "nbformat_minor": 2}